
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<link rel="stylesheet" type="text/css" href="css/GENEDBQuery.css"/>
<link rel="stylesheet" type="text/css" href="css/IndividualResult.css"/>
<link type="text/css" href="css/GENEDBQuery_old.css" rel="stylesheet" />
<link type="text/css" href="css/styleFoot.css" rel="stylesheet" />
<title>IMGT/GENE-DB Consultation</TITLE>
</head>
<body id="GENE-DB">
<table style="width:100%;">
<tr>
<td align="left" nowrap valign="middle" width="70%">
<font size="+3" color="#000099"><b>THANK YOU</b><br/>
<b>for using <a href="/genedb/doc">IMGT/GENE-DB</a></b></font>
</td>
<td class="notshow" align=right valign=top width="30%"><a href="/"><img SRC="images/logo_IMGT2.png" ALT="WELCOME !" class="mainlogo"/></a>
</td>
</tr>
<tr>
<td class="logolegendleft">IMGT&reg;, the international ImMunoGeneTics information system&reg;</td>
<td class="logolegend">http://www.imgt.org</td>
</tr>
</table>
<hr/>
<img src="images/logoGene-DB.png" ALT="Gene-DB logo" class="mainlogo" />

<table class="citation">
<tr>
<td class="citation">
<b>Citing IMGT/GENE-DB</b>: Giudicelli V., Chaume D. and Lefranc M.-P.. Nucleic Acids Res., 33: D256 - D261 (2005).
PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/15608191/">15608191</a>
<a href="/PDF/NAR/33_D256-D261_2005.pdf"><img src="/images/pdficon.png" alt="pdf" /></a>
</td>
</tr>
</table>
<p class="programversion">IMGT/GENE-DB program version: <a href="/IMGTgenedbdoc/programversions.html">3.1.35</a> (3 March 2022)<br/>
<a href="/IMGTgenedbdoc/dataupdates.html">IMGT/GENE-DB data updates </a>
</p>

<h2 class="subtitle"><a href="/IMGTScientificChart/SequenceDescription/IMGTreferencesequences.html">IMGT/GENE-DB reference sequences</a> in FASTA format:</h2>
<h2 class="subtitle">Amino acids  sequences with gaps according to the <a href="/IMGTScientificChart/Numbering/IMGTnumberingCR.html">IMGT unique numbering</a><br /> for <span class="casimir">F+ORF+in-frame P with IMGT Gaps</span> alleles including orphons</h2>
<hr>
<pre>
The FASTA header contains 15 fields separated by '|':

1. IMGT/LIGM-DB accession number(s)
2. IMGT gene and allele name
3. species
4. IMGT allele functionality
5. exon(s), region name(s), or extracted label(s)
6. start and end positions in the IMGT/LIGM-DB accession number(s)
7. number of nucleotides in the IMGT/LIGM-DB accession number(s)
8. codon start, or 'NR' (not relevant) for non coding labels
9. +n: number of nucleotides (nt) added in 5' compared to the corresponding label extracted from IMGT/LIGM-DB
10. +n or -n: number of nucleotides (nt) added or removed in 3' compared to the corresponding label extracted from IMGT/LIGM-DB
11. +n, -n, and/or nS: number of added, deleted, and/or substituted nucleotides to correct sequencing errors, or 'not corrected' if non corrected sequencing errors
12. number of amino acids (AA): this field indicates that the sequence is in amino acids
13. number of characters in the sequence: nt (or AA)+IMGT gaps=total
14. partial (if it is)
15. reverse complementary (if it is)
</pre>
<hr/>
<b>Number of results = 113</b><br />
<pre>
>AE000658|TRAV1-1*01|Homo sapiens|F|V-REGION|128090..128364|275 nt|1| | | |91 AA|91+16=107| | |
GQSLEQ.PSEVTAVEGAIVQINCTYQTSG......FYGLSWYQQHDGGAPTFLSYNAL..
..DGLEET.....GRFSSFLSRSDSYGYLLLQELQMKDSASYFCAVR
>X04939|TRAV1-1*02|Homo sapiens|(F)|V-REGION|52..320|269 nt|1| | | |89 AA|89+16=105| | |
GQSLEQ.PSEVTAVEGAIVQINCTYQTSG......FYGLSWYQQHDGGAPTFLSYNGL..
..DGLEET.....GRFSSFLSRSDSYGYLLLQELQMKDSASYFCA
>AE000658|TRAV1-2*01|Homo sapiens|F|V-REGION|149176..149450|275 nt|1| | | |91 AA|91+16=107| | |
GQNIDQ.PTEMTATEGAIVQINCTYQTSG......FNGLFWYQQHAGEAPTFLSYNVL..
..DGLEEK.....GRFSSFLSRSKGYSYLLLKELQMKDSASYLCAVR
>U32544|TRAV1-2*02|Homo sapiens|[F]|V-REGION|302..477|176 nt|1| | | |58 AA|58+16=74|partial in 3'| |
GQNIDQ.PTEMTATEGAIVQINCTYQTSG......FNGLFWYQQHAGEAPTFLSYNVL..
..DGLEEK.....G
>IMGT000024|TRAV1-2*03|Homo sapiens|F|V-REGION|31409..31683|275 nt|1| | | |91 AA|91+16=107| | |
GQNIDQ.PTEMTATEGAIVQINCTYQTSG......FNGLFWYQQHAGEAPTFLSYNVL..
..DGLEEK.....GRFSSFLSRSKGYSYLLLKELQMKDSASYLCAVR
>AE000659|TRAV10*01|Homo sapiens|F|V-REGION|81133..81412|280 nt|1| | | |93 AA|93+14=107| | |
KNQVEQSPQSLIILEGKNCTLQCNYTVSP......FSNLRWYKQDTGRGPVSLTIMTFS.
..ENTKSN.....GRYTATLDADTKQSSLHITASQLSDSASYICVVS
>IMGT000024|TRAV10*02|Homo sapiens|F|V-REGION|213893..214172|280 nt|1| | | |93 AA|93+14=107| | |
KNQVEQSPQSLIILEGKNCTLQCNYTVSP......FSNLRWYKQDTGRGPVSLTIMTFS.
..ENTKSN.....GRYTATLDADTKQSSLHITASQLSDSASYICVVS
>AE000659|TRAV11*01|Homo sapiens|P|V-REGION|85121..85397|277 nt|1| | ||92 AA|92+14=106| | |
LHTLEQSPSFLNIQEGMHAVLNCTYQERT......LFNFHWFRQDPGRRLVSLTLIQSS.
..QKEQGD.....KYFKELLGKEKFYSVWNIAASHLGDSATYFCAL
>AE000659|TRAV12-1*01|Homo sapiens|F|V-REGION|96856..97129|274 nt|1| | | |91 AA|91+16=107| | |
RKEVEQDPGPFNVPEGATVAFNCTYSNSA......SQSFFWYRQDCRKEPKLLMSVYS..
..SGN.ED.....GRFTAQLNRASQYISLLIRDSKLSDSATYLCVVN
>M17657|TRAV12-1*02|Homo sapiens|(F)|V-REGION|67..340|274 nt|1| | ||91 AA|91+16=107| | |
RKEVEQDPGPFNVPEGATVAFNCTYSNSA......SQSFFWYRQDCRKEPKLLMSVYS..
..SGN.ED.....GRFTAHVNRASQYISLLIRDSKLSDSATYLCVVN
>AE000659|TRAV12-2*01|Homo sapiens|F|V-REGION|143579..143855|277 nt|1| | | |92 AA|92+15=107| | |
QKEVEQNSGPLSVPEGAIASLNCTYSDRG......SQSFFWYRQYSGKSPELIMFIYS..
..NGDKED.....GRFTAQLNKASQYVSLLIRDSQPSDSATYLCAVN
>M81774|TRAV12-2*02|Homo sapiens|(F)|V-REGION|67..339|273 nt|1| | ||91 AA|91+15=106| | |
QKEVEQNSGPLSVPEGAIASLNCTYSDRG......SQSFFWYRQYSGKSPELIMSIYS..
..NGDKED.....GRFTAQLNKASQYVSLLIRDSQPSDSATYLCAV
>X04946|TRAV12-2*03|Homo sapiens|(F)|V-REGION|1..252|252 nt|1| | | |84 AA|84+23=107|partial in 5'| |
........GPLSVPEGAIASLNCTYSDRV......SQSFFWYRQYSGKSPELIMSIYS..
..NGDKED.....GRFTAQLNKASQYVSLLIRDSQPSDSATYLCAVN
>AE000659|TRAV12-3*01|Homo sapiens|F|V-REGION|221187..221463|277 nt|1| | | |92 AA|92+15=107| | |
QKEVEQDPGPLSVPEGAIVSLNCTYSNSA......FQYFMWYRQYSRKGPELLMYTYS..
..SGNKED.....GRFTAQVDKSSKYISLFIRDSQPSDSATYLCAMS
>M17656|TRAV12-3*02|Homo sapiens|(F)|V-REGION|67..343|277 nt|1| | | |92 AA|92+15=107| | |
QKEVEQDPGPLSVPEGAIVSLNCTYSNSA......FQYFMWYRQYSRIGPELLMYTYS..
..SGNKED.....GRFTAQVDKSSKYISLFIRDSQPSDSATYLCAMS
>AE000659|TRAV13-1*01|Homo sapiens|F|V-REGION|124440..124719|280 nt|1| | | |93 AA|93+14=107| | |
GENVEQHPSTLSVQEGDSAVIKCTYSDSA......SNYFPWYKQELGKGPQLIIDIRSN.
..VGEKKD.....QRIAVTLNKTAKHFSLHITETQPEDSAVYFCAAS
>X04954|TRAV13-1*02|Homo sapiens|(F)|V-REGION|58..337|280 nt|1| | ||93 AA|93+14=107| | |
GENVEQHPSTLSVQEGDSAVIKCTYSDSA......SNYFPWYKQELGKRPQLIIDIRSN.
..VGEKKD.....QRIAVTLNKTAKHFSLHITETQPEDSAVYFCAAS
>L11162|TRAV13-1*03|Homo sapiens|[F]|V-REGION|222..455|234 nt|1| | ||78 AA|78+14=92|partial in 3'| |
GENVEQHPSTLSVQEGDSAVIKCTYSDSA......SNYFPWYKQELGKRPQLIIDIRSN.
..VGEKKD.....QRIAVTLNKTAKHFSLQIT
>AE000659|TRAV13-2*01|Homo sapiens|F|V-REGION|173825..174104|280 nt|1| | | |93 AA|93+14=107| | |
GESVGLHLPTLSVQEGDNSIINCAYSNSA......SDYFIWYKQESGKGPQFIIDIRSN.
..MDKRQG.....QRVTVLLNKTVKHLSLQIAATQPGDSAVYFCAEN
>M17658|TRAV13-2*02|Homo sapiens|(F)|V-REGION|48..324|277 nt|1| | | |92 AA|92+14=106| | |
GESVGLHLPTLSVQEGDNSIINCAYSNSA......SDYFIWYKQESGKGPQFIIDIRSN.
..MDKRQG.....QRVTVLLNKTVKHLSLQIAATQPGDSAVYFCAE
>M21626|TRAV14/DV4*01|Homo sapiens|F|V-REGION|226..515|290 nt|1| | | |96 AA|96+12=108| | |
AQKITQTQPGMFVQEKEAVTLDCTYDTSDP.....SYGLFWYKQPSSGEMIFLIYQGSY.
.DQQNATE.....GRYSLNFQKARKSANLVISASQLGDSAMYFCAMRE
>AE000659|TRAV14/DV4*02|Homo sapiens|F|V-REGION|179711..180000|290 nt|1| | | |96 AA|96+12=108| | |
AQKITQTQPGMFVQEKEAVTLDCTYDTSDQ.....SYGLFWYKQPSSGEMIFLIYQGSY.
.DEQNATE.....GRYSLNFQKARKSANLVISASQLGDSAMYFCAMRE
>M21624|TRAV14/DV4*03|Homo sapiens|(F)|V-REGION|131..412|282 nt|1| | | |94 AA|94+12=106| | |
AQKITQTQPGMFVQEKEAVTLDCTYDTSDP.....SYGLFWYKQPSSGEMIFLIYQGSY.
.DQQNATE.....GRYSLNFQKARKSANLVISASQLGDSAMYFCAM
>L09758|TRAV14/DV4*04|Homo sapiens|[F]|V-REGION|1..271|271 nt|1| | ||90 AA|90+13=103|partial in 5' and in 3' | |
.QKITQTQPGMFVQEKEAVTLDCTYDTSDQ.....SYGLFWYKQPSSGEMIFLIYQGSY.
.DEQNATE.....GRYSLNFQKARKSANLVISASQLGDSAMYF
>AE000659|TRAV16*01|Homo sapiens|F|V-REGION|246079..246350|272 nt|1| | | |90 AA|90+17=107| | |
AQRVTQPEKLLSVFKGAPVELKCNYSYSG......SPELFWYVQYSRQRLQLLLRHI...
...SRESI.....KGFTADLNKGETSFHLKKPFAQEEDSAMYYCALS
>AE000660|TRAV17*01|Homo sapiens|F|V-REGION|2163..2439|277 nt|1| | | |92 AA|92+15=107| | |
SQQGEEDPQALSIQEGENATMNCSYKTSI.......NNLQWYRQNSGRGLVHLILIRSN.
..EREKHS.....GRLRVTLDTSKKSSSLLITASRAADTASYFCATD
>AE000660|TRAV18*01|Homo sapiens|F|V-REGION|7667..7944|278 nt|1| | | |92 AA|92+15=107| | |
GDSVTQTEGPVTLPERAALTLNCTYQSSY......STFLFWYVQYLNKEPELLLKSSE..
..NQETDS.....RGFQASPIKSDSSFHLEKPSVQLSDSAVYYCALR
>AE000660|TRAV19*01|Homo sapiens|F|V-REGION|12163..12452|290 nt|1| | | |96 AA|96+12=108| | |
AQKVTQAQTEISVVEKEDVTLDCVYETRDT.....TYYLFWYKQPPSGELVFLIRRNSF.
.DEQNEIS.....GRYSWNFQKSTSSFNFTITASQVVDSAVYFCALSE
>AE000658|TRAV2*01|Homo sapiens|F|V-REGION|218448..218710|263 nt|1| | | |87 AA|87+20=107| | |
KDQVFQ.PSTVASSEGAVVEIFCNHSVSN......AYNFFWYLHFPGCAPRLLVKGS...
...KPSQQ.....GRYNMTYER..FSSSLLILQVREADAAVYYCAVE
>M17659|TRAV2*02|Homo sapiens|(F)|V-REGION|53..316|264 nt|1| | | |88 AA|88+20=108| | |
KDQVFQ.PSTVASSEGAVVEIFCNHSVSN......AYNFFWHLHFPGCAPRLLVKGS...
...KPSQQ.....GRYNMTYER..FSSSLLILQVREADAAVYYCAVAW
>AE000660|TRAV20*01|Homo sapiens|F|V-REGION|45158..45431|274 nt|1| | | |91 AA|91+16=107| | |
EDQVTQSPEALRLQEGESSSLNCSYTVSG......LRGLFWYRQDPGKGPEFLFTLYSA.
..GEEKEK.....ERLKATLTK..KESFLHITAPKPEDSATYLCAVQ
>IMGT000024|TRAV20*02|Homo sapiens|F|V-REGION|428977..429250|274 nt|1| | | |91 AA|91+16=107| | |
EDQVTQSPEALRLQEGESSSLNCSYTVSG......LRGLFWYRQDPGKGPEFLFTLYSA.
..GEEKEK.....ERLKATLTK..KESFLHITAPKPEDSATYLCAVQ
>S60789|TRAV20*03|Homo sapiens|(F)|V-REGION|79..342|264 nt|1| | ||88 AA|88+16=104|partial in 3'| |
EDQVTQSPEALRLQEGESRSLNCSYTVSG......LRGLFWYRQDPGKGPEFLFTLYSA.
..GEEKEK.....ERLKATLTK..KESFLHITAPKPEDSATYLC
>X70305|TRAV20*04|Homo sapiens|(F)|V-REGION|133..399|267 nt|1| | ||89 AA|89+16=105|partial in 3'| |
EDQVTQSPEALRLQEGESSSLNCSCTVSG......LRGLFWYRQDPGKGPEFLFTLYSA.
..GEEKEK.....ERLKATLTK..KESFLHITAPKPEDSATYLCA
>AE000660|TRAV21*01|Homo sapiens|F|V-REGION|57069..57347|279 nt|1| | | |93 AA|93+14=107| | |
KQEVTQIPAALSVPEGENLVLNCSFTDSA......IYNLQWFRQDPGKGLTSLLLIQSS.
..QREQTS.....GRLNASLDKSSGRSTLYIAASQPGDSATYLCAVR
>X58736|TRAV21*02|Homo sapiens|(F)|V-REGION|128..400|273 nt|1| | | |91 AA|91+14=105|partial in 3'| |
KQEVTQIPAALSVPEGENLVLNCSFTDSA......IYNLQWFRQDPGKGLTSLLLIQSS.
..QREQTS.....GRLNASLDKSSGRSTLYIAASQPGDSATYLCA
>AE000660|TRAV22*01|Homo sapiens|F|V-REGION|75233..75503|271 nt|1| | | |90 AA|90+17=107| | |
GIQVEQSPPDLILQEGANSTLRCNFSDSV.......NNLQWFHQNPWGQLINLFYIPS..
...GTKQN.....GRLSATTVATERYSLLYISSSQTTDSGVYFCAVE
>AE000660|TRAV23/DV6*01|Homo sapiens|F|V-REGION|90991..91270|280 nt|1| | ||93 AA|93+14=107| | |
QQQVKQSPQSLIVQKGGISIINCAYENTA......FDYFPWYQQFPGKGPALLIAIRPD.
..VSEKKE.....GRFTISFNKSAKQFSLHIMDSQPGDSATYFCAAS
>M17660|TRAV23/DV6*02|Homo sapiens|(F)|V-REGION|85..364|280 nt|1| | ||93 AA|93+14=107| | |
QQQVKQSPQSLIVQKGGIPIINCAYENTA......FDYFPWYQQFPGKGPALLIAIRPD.
..VSEKKE.....GRFTISFNKSAKQFSLHIMDSQPGDSATYFCAAS
>M97704|TRAV23/DV6*03|Homo sapiens|(F)|V-REGION|145..424|280 nt|1| | ||93 AA|93+14=107| | |
QQQVKQSPQSLIVQKGGISIINCAYENTA......FDYFPWYQQFPGKGPALLIAIRPD.
..VSEKKE.....GRFTISFNKSAKQFSLHIMDSQPGDSATYFCAAS
>Y10411|TRAV23/DV6*04|Homo sapiens|[F]|V-REGION|1..267|267 nt|1| | | |89 AA|89+15=104|partial in 5' and in 3' | |
.QQVKQSPQSLIVQKGGISIINCAYENTA......FDYFPWYQQFPGKGPALLIAIRPD.
..VSEKKE.....GRFTISFNKSAKQFSLHIMDSQPGDSATYFC
>IMGT000024|TRAV23/DV6*05|Homo sapiens|F|V-REGION|474779..475058|280 nt|1| | | |93 AA|93+14=107| | |
QQQVKQSPQSLIVQKGGISIINCAYENTA......FDYFPWYQQFPGKGPALLIAIRPD.
..VSEKKE.....GRFTISFNKSAKQFSSHIMDSQPGDSATYFCAAS
>AE000660|TRAV24*01|Homo sapiens|F|V-REGION|109879..110155|277 nt|1| | | |92 AA|92+14=106| | |
ILNVEQSPQSLHVQEGDSTNFTCSFPSSN......FYALHWYRWETAKSPEALFVMTLN.
..GDEKKK.....GRISATLNTKEGYSYLYIKGSQPEDSATYLCAF
>M17661|TRAV24*02|Homo sapiens|(F)|V-REGION|64..340|277 nt|1| | | |92 AA|92+14=106| | |
ILNVEQGPQSLHVQEGDSTNFTCSFPSSN......FYALHWYRWETAKTPEALFVMTLN.
..GDEKKK.....GRISATLNTKEGYSYLYIKGSQPEDSATYLCAF
>AE000660|TRAV25*01|Homo sapiens|F|V-REGION|116744..117016|273 nt|1| | | |91 AA|91+15=106| | |
GQQVMQIPQYQHVQEGEDFTTYCNSSTTL.......SNIQWYKQRPGGHPVFLIQLVKS.
..GEVKKQ.....KRLTFQFGEAKKNSSLHITATQTTDVGTYFCAG
>AE000660|TRAV26-1*01|Homo sapiens|F|V-REGION|127999..128275|277 nt|1| | | |92 AA|92+16=108| | |
DAKTTQ.PPSMDCAEGRAANLPCNHSTISG.....NEYVYWYRQIHSQGPQYIIHGLK..
...NNETN.....EMASLIITEDRKSSTLILPHATLRDTAVYYCIVRV
>IMGT000024|TRAV26-1*02|Homo sapiens|F|V-REGION|512106..512382|277 nt|1| | | |92 AA|92+16=108| | |
DAKTTQ.PTSMDCAEGRAANLPCNHSTISG.....NEYVYWYRQIHSQGPQYIIHGLK..
...NNETN.....EMASLIITEDRKSSTLILPHATLRDTAVYYCIVRV
>L06886|TRAV26-1*03|Homo sapiens|(F)|V-REGION|52..318|267 nt|1| | ||89 AA|89+16=105|partial in 3'| |
DAKTTQ.PPSMDCAEGRAANLPCNHSTISG.....NEYVYWYRQIHSQGPQNIIHGLK..
...NNETN.....EMASLIITEDRKSSTLILPHATLRDTAVYYCI
>AE000660|TRAV26-2*01|Homo sapiens|F|V-REGION|207019..207294|276 nt|1| | | |92 AA|92+16=108| | |
DAKTTQ.PNSMESNEEEPVHLPCNHSTISG.....TDYIHWYRQLPSQGPEYVIHGLT..
...SNVNN.....RMASLAIAEDRKSSTLILHRATLRDAAVYYCILRD
>L11160|TRAV26-2*02|Homo sapiens|[F]|V-REGION|509..725|217 nt|1| | ||72 AA|72+16=88|partial in 3'| |
DAKTTQ.PNSMESNEEEPVHLPCNHSTISG.....TDYIHWYRQLPSQGPEYVIHGLT..
...SNVNN.....RMACVAIAEDRKSST
>AE000660|TRAV27*01|Homo sapiens|F|V-REGION|152346..152619|274 nt|1| | | |91 AA|91+15=106| | |
TQLLEQSPQFLSIQEGENLTVYCNSSSVF.......SSLQWYRQEPGEGPVLLVTVVTG.
..GEVKKL.....KRLTFQFGDARKDSSLHITAAQPGDTGLYLCAG
>X04957|TRAV27*02|Homo sapiens|(F)|V-REGION|55..326|272 nt|1| | | |90 AA|90+15=105| | |
TQLLEQSPQFLSIQEGENLTVYCNSSSVF.......SSLQWYRQEPGEGPVLLVTVVTG.
..GEVKKL.....KRLTFQFGDARKDSSLHITAAQPGDTGHYLCA
>IMGT000024|TRAV27*03|Homo sapiens|F|V-REGION|536457..536730|274 nt|1| | | |91 AA|91+15=106| | |
TQLLEQSPQFLSIQEGENLTVYCNSSSVF.......SSLQWYRQEPGEGPVLLVTVVTG.
..GEVKKL.....KRLTFQFGDARKDSSLHITAAQTGDTGLYLCAG
>AE000660|TRAV29/DV5*01|Homo sapiens|F|V-REGION|167519..167798|280 nt|1| | | |93 AA|93+14=107| | |
DQQVKQNSPSLSVQEGRISILNCDYTNSM......FDYFLWYKKYPAEGPTFLISISSI.
..KDKNED.....GRFTVFLNKSAKHLSLHIVPSQPGDSAVYFCAAS
>S81645|TRAV29/DV5*02|Homo sapiens|F|V-REGION|174..452|279 nt|1| | | |93 AA|93+14=107| | |
DQQVKQNSPSLSVQEGRISILNCDYTNSM......FDYFLWYKKYPAEGPTFLISISSI.
..KDKNED.....GRFTVFLNKSAKHLSLDIVPSQPGDSAVYFCAAS
>IMGT000024|TRAV29/DV5*04|Homo sapiens|F|V-REGION|551688..551967|280 nt|1| | | |93 AA|93+14=107| | |
DQQVKQNSPSLSVQEGRISILNCDYTNSM......FDYFLWYKKYPAEGPTFLISISSI.
..KDKNED.....GRFTVFLNKSAKHLSLHIVPSQPGDSAVYFCAAS
>AE000658|TRAV3*01|Homo sapiens|F|V-REGION|229927..230212|286 nt|1| | | |95 AA|95+13=108| | |
AQSVAQPEDQVNVAEGNPLTVKCTYSVSG......NPYLFWYVQYPNRGLQFLLKYITG.
.DNLVKGS.....YGFEAEFNKSQTSFHLKKPSALVSDSALYFCAVRD
>AE000660|TRAV30*01|Homo sapiens|F|V-REGION|172643..172916|274 nt|1| | | |91 AA|91+16=107| | |
QQPV.QSPQAVILREGEDAVINCSSSKAL.......YSVHWYRQKHGEAPVFLMILLKG.
..GEQKGH.....EKISASFNEKKQQSSLYLTASQLSYSGTYFCGTE
>X58768|TRAV30*02|Homo sapiens|(F)|V-REGION|95..361|267 nt|1| | | |89 AA|89+16=105|partial in 3'| |
QQPV.QSPQAVILREGEDAVTNCSSSKAL.......YSVHWYRQKHGEAPVFLMILLKG.
..GEQMRR.....EKISASFNEKKQQSSLYLTASQLSYSGTYFCG
>L06883|TRAV30*03|Homo sapiens|(F)|V-REGION|64..330|267 nt|1| | | |89 AA|89+16=105|partial in 3'| |
QQPV.QSPQAVILREGEDAVINCSSSKAL.......YSVHWYRQKHGEAPVFLMILLKG.
..GEQKGH.....EKISASFNEKKRQSSLYLTASQLSYSGTYFCG
>U32537|TRAV30*04|Homo sapiens|[F]|V-REGION|256..485|230 nt|1| | | |76 AA|76+16=92|partial in 3'| |
QQPV.QSPQAVILREGEDAVINCSSSKAL.......YSVHWYRQKHGEAPVFLMILLKG.
..GEQKRH.....EKISASFNEKKQQSSLYLT
>IMGT000024|TRAV30*05|Homo sapiens|F|V-REGION|556812..557085|274 nt|1| | | |91 AA|91+16=107| | |
QQPV.QSPQAVILREGEDAVINCSSSKAL.......YSVHWYRQKHGEAPVFLMILLKG.
..GEQKGH.....DKISASFNEKKQQSSLYLTASQLSYSGTYFCGTE
>AE000660|TRAV34*01|Homo sapiens|F|V-REGION|211781..212057|277 nt|1| | | |92 AA|92+15=107| | |
SQELEQSPQSLIVQEGKNLTINCTSSKTL.......YGLYWYKQKYGEGLIFLMMLQKG.
..GEEKSH.....EKITAKLDEKKQQSSLHITASQPSHAGIYLCGAD
>AE000660|TRAV35*01|Homo sapiens|F|V-REGION|226128..226403|276 nt|1| | | |92 AA|92+15=107| | |
GQQLNQSPQSMFIQEGEDVSMNCTSSSIF.......NTWLWYKQEPGEGPVLLIALYKA.
..GELTSN.....GRLTAQFGITRKDSFLNISASIPSDVGIYFCAGQ
>X58738|TRAV35*02|Homo sapiens|(F)|V-REGION|131..400|270 nt|1| | ||90 AA|90+15=105|partial in 3'| |
GQQLNQSPQSMFIQEGEDVSMNCTSSSIF.......NTWLWYKQDPGEGPVLLIALYKA.
..GELTSN.....GRLTAQFGITRKDSFLNISASIPSDVGIYFCA
>IMGT000024|TRAV35*03|Homo sapiens|P|V-REGION|610297..610572|276 nt|1| | | |92 AA|92+15=107| | |
GQQLNQSPQSMFTQEGEDVSMNCTSSSIF.......NT*LWYKQDPGEGPVLLIALYKA.
..GELTSN.....GRLTAQFGITRKDSFLNISASIPSDVGIYFCAGQ
>AE000660|TRAV36/DV7*01|Homo sapiens|F|V-REGION|230905..231181|277 nt|1| | | |92 AA|92+15=107| | |
EDKVVQSPLSLVVHEGDTVTLNCSYEVTN......FRSLLWYKQEKKAP.TFLFMLTSS.
..GIEKKS.....GRLSSILDKKELSSILNITATQTGDSAIYLCAVE
>X61070|TRAV36/DV7*02|Homo sapiens|(F)|V-REGION|7..280|274 nt|1| | | |91 AA|91+15=106| | |
EDKVVQSPQSLVVHEGDTVTLNCSYEMTN......FRSLQWYKQEKKAP.TFLFMLTSS.
..GIEKKS.....GRLSSILDKKELFSILNITATQTGDSAVYLCAV
>X58767|TRAV36/DV7*03|Homo sapiens|(F)|V-REGION|109..378|270 nt|1| | ||90 AA|90+15=105|partial in 3'| |
EDKVVQSPLSLVVHEGDTVTPNCSYEVTN......FRSLLWYKQEKKAP.TFLFMLTSS.
..GIEKKS.....GRLSSILDKKELFSILNITATQTGDSAVYLCA
>Z46643|TRAV36/DV7*04|Homo sapiens|(F)|V-REGION|87..357|271 nt|1| | | |90 AA|90+15=105| | |
EDKVVQSPLSLVVHEGDTVTLNCSYEVTN......FRSLLWYKQEKKAP.TFLFMLTSS.
..GIEKKS.....GRLSSILDKKELFSILNITATQTGDSAVYLCA
>IMGT000024|TRAV36/DV7*05|Homo sapiens|F|V-REGION|615075..615351|277 nt|1| | | |92 AA|92+15=107| | |
EDKVVQSPLSLVVHEGDTVTLNCSYEVTN......FRSLLWYKQEKKAP.TFLFMLTSS.
..GIEKKS.....GRLSSILDKKELFSILNITATQTGDSAIYLCAVE
>AE000661|TRAV38-1*01|Homo sapiens|F|V-REGION|25123..25412|290 nt|1| | | |96 AA|96+12=108| | |
AQTVTQSQPEMSVQEAETVTLSCTYDTSEN.....NYYLFWYKQPPSRQMILVIRQEAY.
.KQQNATE.....NRFSVNFQKAAKSFSLKISDSQLGDTAMYFCAFMK
>M64355|TRAV38-1*02|Homo sapiens|(F)|V-REGION|94..373|280 nt|1| | | |93 AA|93+12=105| | |
AQTVTQSQPEMSVQEAETVTLSCTYDTSEN.....DYYLFWYKQPPSRQMILVIRQEAY.
.KQQNATE.....NRFSVNFQKAAKSFSLKISDSQLGDTAMYFCA
>M95394|TRAV38-1*03|Homo sapiens|(F)|V-REGION|35..317|283 nt|1| | | |94 AA|94+12=106| | |
AQTVTQSQPEMSVQEAETVTLSCTYDTSES.....NYYLFWYKQPPSRQMILVIRQEAY.
.KQQNATE.....NRFSVNFQKAAKSFSLKISDSQLGDTAMYFCAF
>L06880|TRAV38-1*04|Homo sapiens|(F)|V-REGION|61..339|279 nt|1| | ||93 AA|93+12=105|partial in 3'| |
AQTVTQSQPEMSVQEAETVTLSCTYDTSEN.....NYYLFWYKQPPSRQMILVIRQEAY.
.KQQNATE.....NRFSVNFQKAAKSFSLKISDSQLGDTAMYFCA
>AE000661|TRAV38-2/DV8*01|Homo sapiens|F|V-REGION|34309..34597|289 nt|1| | | |96 AA|96+12=108| | |
AQTVTQSQPEMSVQEAETVTLSCTYDTSES.....DYYLFWYKQPPSRQMILVIRQEAY.
.KQQNATE.....NRFSVNFQKAAKSFSLKISDSQLGDAAMYFCAYRS
>AE000661|TRAV39*01|Homo sapiens|F|V-REGION|57128..57404|277 nt|1| | | |92 AA|92+15=107| | |
ELKVEQNPLFLSMQEGKNYTIYCNYSTTS.......DRLYWYRQDPGKSLESLFVLLSN.
..GAVKQE.....GRLMASLDTKARLSTLHITAAVHDLSATYFCAVD
>AE000658|TRAV4*01|Homo sapiens|F|V-REGION|242632..242908|277 nt|1| | | |92 AA|92+16=108| | |
LAKTTQ.PISMDSYEGQEVNITCSHNNIAT.....NDYITWYQQFPSQGPRFIIQGYK..
...TKVTN.....EVASLFIPADRKSSTLSLPRVSLSDTAVYYCLVGD
>X73521|TRAV40*01|Homo sapiens|F|V-REGION|237..499|263 nt|1| | | |87 AA|87+20=107| | |
SNSVKQT.GQITVSEGASVTMNCTYTSTG......YPTLFWYVEYPSKPLQLLQRET...
...MENSK.....NFGGGNIKD..KNSPIVKYSVQVSDSAVYYCLLG
>AE000661|TRAV41*01|Homo sapiens|F|V-REGION|73820..74089|270 nt|1| | | |90 AA|90+17=107| | |
KNEVEQSPQNLTAQEGEFITINCSYSVGI.......SALHWLQQHPGGGIVSLFMLSS..
...GKKKH.....GRLIATINIQEKHSSLHITASHPRDSAVYICAVR
>AE000659|TRAV5*01|Homo sapiens|F|V-REGION|4886..5162|277 nt|1| | | |92 AA|92+15=107| | |
GEDVEQS.LFLSVREGDSSVINCTYTDSS......STYLYWYKQEPGAGLQLLTYIFSN.
..MDMKQD.....QRLTVLLNKKDKHLSLRIADTQTGDSAIYFCAES
>AE000659|TRAV6*01|Homo sapiens|F|V-REGION|24214..24493|280 nt|1| | | |93 AA|93+14=107| | |
SQKIEQNSEALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLKQSLFHITASQPADSATYLCALD
>X58747|TRAV6*02|Homo sapiens|(F)|V-REGION|128..400|273 nt|1| | ||91 AA|91+14=105|partial in 3'| |
SQKIEQNSEALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLKQSLFHITASQPADSATYLCA
>Z49060|TRAV6*03|Homo sapiens|[F]|V-REGION|1..249|249 nt|1| | ||83 AA|83+22=105|partial in 5' and in 3' | |
........EALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLKQSLFHITASQPADSATYLCA
>Y10409|TRAV6*04|Homo sapiens|[F]|V-REGION|1..249|249 nt|1| | ||83 AA|83+22=105|partial in 5' and in 3' | |
........EALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLKQSLFHVTASQPADSATYLCA
>Y10410|TRAV6*05|Homo sapiens|[F]|V-REGION|1..249|249 nt|1| | ||83 AA|83+22=105|partial in 5' and in 3' | |
........EALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLKQSLFHITASQPADSATYLCA
>U32542|TRAV6*06|Homo sapiens|[F]|V-REGION|174..390|217 nt|1| | ||72 AA|72+14=86|partial in 3'| |
SQKIEQNSEALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLNQ
>IMGT000024|TRAV6*07|Homo sapiens|F|V-REGION|156898..157177|280 nt|1| | | |93 AA|93+14=107| | |
SQKIEQNSEALNIQEGKTATLTCNYTNYS......PAYLQWYRQDPGRGPVFLLLIREN.
..EKEKRK.....ERLKVTFDTTLKQSLFHITASQPADSATYLCALD
>AE000659|TRAV7*01|Homo sapiens|F|V-REGION|38620..38893|274 nt|1| | | |91 AA|91+16=107| | |
ENQVEHSPHFLGPQQGDVASMSCTYSVSR......FNNLQWYRQNTGMGPKHLLSMYSA.
..GYEKQK.....GRLNATLLK..NGSSLYITAVQPEDSATYFCAVD
>AE000659|TRAV8-1*01|Homo sapiens|F|V-REGION|52948..53231|284 nt|1| | | |94 AA|94+13=107| | |
AQSVSQHNHHVILSEAASLELGCNYSYGG......TVNLFWYVQYPGQHLQLLLKYFSG.
.DPLVKGI.....KGFEAEFIKSKFSFNLRKPSVQWSDTAEYFCAVN
>U32520|TRAV8-1*02|Homo sapiens|[F]|V-REGION|122..371|250 nt|1| | ||83 AA|83+13=96|partial in 3'| |
AQSVSQHNHHVILSEAASLELGCNYSYGG......TVNLFWYVQYPGQHLQLLLKYFSG.
.DPLVKGI.....KGVEAEFIKSKFSFNLRKPSVQW
>AE000659|TRAV8-2*01|Homo sapiens|F|V-REGION|102293..102576|284 nt|1| | | |94 AA|94+13=107| | |
AQSVTQLDSHVSVSEGTPVLLRCNYSSSY......SPSLFWYVQHPNKGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCVVS
>M17650|TRAV8-2*02|Homo sapiens|(F)|V-REGION|16..295|280 nt|1| | ||93 AA|93+13=106| | |
AQSVTQLSSHVSVSEGTPVLLRCNYSSSY......SPSLFWYVQHPNKGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCVV
>IMGT000024|TRAV8-2*03|Homo sapiens|F|V-REGION|235035..235318|284 nt|1| | | |94 AA|94+13=107| | |
AQSVTQLDSHVSVSEGTPVLLRCNYSSSY......SPSLFWYVQHPNKGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCVVS
>AE000659|TRAV8-3*01|Homo sapiens|F|V-REGION|108073..108356|284 nt|1| | | |94 AA|94+13=107| | |
AQSVTQPDIHITVSEGASLELRCNYSYGA......TPYLFWYVQSPGQGLQLLLKYFSG.
.DTLVQGI.....KGFEAEFKRSQSSFNLRKPSVHWSDAAEYFCAVG
>M35617|TRAV8-3*02|Homo sapiens|(F)|V-REGION|177..458|282 nt|1| | | |94 AA|94+13=107| | |
AQSVTQPDIHITVSEGASLELRCNYSYGA......TPYLFWYVQSPGQGLQLLLKYFSG.
.DTLVQGI.....KGFEAEFKRSQSSFNLRKPSVHWSDAAEYFCAVV
>L06885|TRAV8-3*03|Homo sapiens|(F)|V-REGION|25..300|276 nt|1| | ||92 AA|92+13=105|partial in 3'| |
AQSVTQPDIHITVSEGASLELRCNYSYGA......TPYLFWYVQSPGQGLQLLLKYFSG.
.DTLVQGI.....KGFEAEFKRSQSSFNLRKPSVHWSDASEYFCA
>AE000659|TRAV8-4*01|Homo sapiens|F|V-REGION|150100..150383|284 nt|1| | | |94 AA|94+13=107| | |
AQSVTQLGSHVSVSEGALVLLRCNYSSSV......PPYLFWYVQYPNQGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCAVS
>M12423|TRAV8-4*02|Homo sapiens|(F)|V-REGION|227..510|284 nt|1| | ||94 AA|94+13=107| | |
AQSVTQLGSHVSVSEGALVLLRCNYSSSV......PPYLFWYVQYPNQGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCAVS
>D13077|TRAV8-4*03|Homo sapiens|(F)|V-REGION|58..333|276 nt|1| | ||92 AA|92+13=105|partial in 3'| |
AQSVTQLGSHVSVSEGALVLLRCNYSSSV......PPYLFWYVQYPNQGLQLLLKYTTG.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCA
>M12959|TRAV8-4*04|Homo sapiens|(F)|V-REGION|194..477|284 nt|1| | ||94 AA|94+13=107| | |
AQSVTQLGSHVSVSERALVLLRCNYSSSV......PPYLFWYVQYPNQGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCAVS
>X63455|TRAV8-4*05|Homo sapiens|(F)|V-REGION|147..430|284 nt|1| | ||94 AA|94+13=107| | |
AQSVTQLGSHVSVSEGALVLLRCNYSSSV......PPYLFWYVQYPNQGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMSDAAEYFCAVS
>K02777|TRAV8-4*06|Homo sapiens|(F)|V-REGION|1..218|218 nt|1| | ||72 AA|72+35=107|partial in 5'| |
......................GATHYCC......PPILFWYVQYPNQGLQLLLKYTSA.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPAAHMSDAAEYFCAVS
>M17665|TRAV8-4*07|Homo sapiens|(F)|V-REGION|1..197|197 nt|1| | | |65 AA|65+41=106|partial in 5'| |
..........................VE........PYLFWYVQYPNQGLQLLLKYTTG.
.ATLVKGI.....NGFEAEFKKSETSFHLTKPSAHMTDPAEYFCAV
>X02850|TRAV8-6*01|Homo sapiens|F|V-REGION|202..485|284 nt|1| | | |94 AA|94+13=107| | |
AQSVTQLDSQVPVFEEAPVELRCNYSSSV......SVYLFWYVQYPNQGLQLLLKYLSG.
.STLVESI.....NGFEAEFNKSQTSFHLRKPSVHISDTAEYFCAVS
>AE000659|TRAV8-6*02|Homo sapiens|F|V-REGION|234250..234533|284 nt|1| | | |94 AA|94+13=107| | |
AQSVTQLDSQVPVFEEAPVELRCNYSSSV......SVYLFWYVQYPNQGLQLLLKYLSG.
.STLVKGI.....NGFEAEFNKSQTSFHLRKPSVHISDTAEYFCAVS
>AE000660|TRAV8-7*01|Homo sapiens|P|V-REGION|136739..137022|284 nt|1| | ||94 AA|94+13=107| | |
TQSVTQLDGHITVSEEAPLELKCNYSYSG......VPSLFWYVQYSSQSLQLLLKDLTE.
.ATQVKGI.....RGFEAEFKKSETSFYLRKPSTHVSDAAEYFCAVG
>U76492|TRAV8-7*02|Homo sapiens|P|V-REGION|193..476|284 nt|1| | | |94 AA|94+13=107| | |
TQSVTQLDGHITVSEEAPLELKCNYSYSG......VPSLFWYVQYSSQSLQLLLKDLTK.
.ATQVKGI.....RGFEAEFKKSETSFYLRKPSTHVSDAAEYFCAVG
>AE000659|TRAV9-1*01|Homo sapiens|F|V-REGION|67042..67322|281 nt|1| | | |93 AA|93+14=107| | |
GDSVVQTEGQVLPSEGDSLIVNCSYETTQ......YPSLFWYVQYPGEGPQLHLKAMKA.
..NDKGRN.....KGFEAMYRKETTSFHLEKDSVQESDSAVYFCALS
>AE000659|TRAV9-2*01|Homo sapiens|F|V-REGION|196741..197021|281 nt|1| | | |93 AA|93+14=107| | |
GNSVTQMEGPVTLSEEAFLTINCTYTATG......YPSLFWYVQYPGEGLQLLLKATKA.
..DDKGSN.....KGFEATYRKETTSFHLEKGSVQVSDSAVYFCALS
>IMGT000024|TRAV9-2*02|Homo sapiens|F|V-REGION|329474..329754|281 nt|1| | | |93 AA|93+14=107| | |
GDSVTQMEGPVTLSEEAFLTINCTYTATG......YPSLFWYVQYPGEGLQLLLKATKA.
..DDKGSN.....KGFEATYRKETTSFHLEKGSVQVSDSAVYFCALS
>L06881|TRAV9-2*03|Homo sapiens|(F)|V-REGION|58..330|273 nt|1| | ||91 AA|91+14=105|partial in 3'| |
GDSVTQMEGPVTLSEEAFLTINCTYTATG......YPSLFWYVQYPGEGLQLLLKATKA.
..DDKGSN.....KGFEATYRKETTSFHLEKGSVQVSDSAVYFCA
>L06882|TRAV9-2*04|Homo sapiens|(F)|V-REGION|58..330|273 nt|1| | | |91 AA|91+14=105| | |
GNSVTQMEGPVTLSEEAFLTINCTYTATG......YPSLFWYVQYPGEGLQLLLKATKA.
..DDKGSN.....KGFEATYRKETTSFHLEKGSVQVSDSAVYFCA

</pre>
<hr />





<a href="/genedb/doc">-> IMGT/GENE-DB Documentation</a><br />
<a href="/genedb/">-> IMGT/GENE-DB Query page</a><br />
<a href="/genedb/directlinks">-> IMGT/GENE-DB direct links</a><br />
<a href="/genedb/stats">-> IMGT/GENE-DB statistics</a><br />
<a href="/IMGTgenedbdoc/programversions.html">-> IMGT/GENE-DB program versions</a><br />
<a href="/IMGTgenedbdoc/dataupdates.html">-> IMGT/GENE-DB data updates</a>

<hr />

<p class="menu">
<a href="/">IMGT Home page</a>
| <a href="/IMGTrepertoire/">IMGT Repertoire (IG and TR)</a>
| <a href="/IMGTrepertoireMH/">IMGT Repertoire (MH)</a>
| <a href="/IMGTrepertoireRPI/">IMGT Repertoire (RPI)</a>
| <a href="/IMGTindex/">IMGT Index</a>
| <a href="/IMGTScientificChart/">IMGT Scientific chart</a>
| <a href="/IMGTeducation/">IMGT Education</a>
</p>



<span class="logosFoot">
<a href="https://www.cnrs.fr"><img src="/images/cnrs.png" alt="logo CNRS" style="vertical-align:middle;" /></a>
<a href="https://www.umontpellier.fr/"><img src="/images/logos/universite_mtp.png" alt="Universit&#233; de Montpellier" style="width:60px;vertical-align:middle;" /></a>
<a href="https://ec.europa.eu/info/research-and-innovation_en"><img src="/images/UE.png" alt="logo UE" style="vertical-align:middle;" /></a>
</span>



<p>&#169;  Copyright 1995-2022 IMGT&reg;, the international ImMunoGeneTics information system&reg;
			| <a href="/about/termsofuse.php">Terms of use</a> | <a href="/about/aboutus.php">About us</a> | <a href="mailto:contact-imgt@igh.cnrs.fr">Contact us</a> | <a href="/about/CitingIMGT.php">Citing IMGT</a></p>

<!-- #echo var=$LAST_MODIFIED -->

</body>
</html>

</body>
</html>