
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<link rel="stylesheet" type="text/css" href="css/GENEDBQuery.css"/>
<link rel="stylesheet" type="text/css" href="css/IndividualResult.css"/>
<link type="text/css" href="css/GENEDBQuery_old.css" rel="stylesheet" />
<link type="text/css" href="css/styleFoot.css" rel="stylesheet" />
<title>IMGT/GENE-DB Consultation</TITLE>
</head>
<body id="GENE-DB">
<table style="width:100%;">
<tr>
<td align="left" nowrap valign="middle" width="70%">
<font size="+3" color="#000099"><b>THANK YOU</b><br/>
<b>for using <a href="/genedb/doc">IMGT/GENE-DB</a></b></font>
</td>
<td class="notshow" align=right valign=top width="30%"><a href="/"><img SRC="images/logo_IMGT2.png" ALT="WELCOME !" class="mainlogo"/></a>
</td>
</tr>
<tr>
<td class="logolegendleft">IMGT&reg;, the international ImMunoGeneTics information system&reg;</td>
<td class="logolegend">http://www.imgt.org</td>
</tr>
</table>
<hr/>
<img src="images/logoGene-DB.png" ALT="Gene-DB logo" class="mainlogo" />

<table class="citation">
<tr>
<td class="citation">
<b>Citing IMGT/GENE-DB</b>: Giudicelli V., Chaume D. and Lefranc M.-P.. Nucleic Acids Res., 33: D256 - D261 (2005).
PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/15608191/">15608191</a>
<a href="/PDF/NAR/33_D256-D261_2005.pdf"><img src="/images/pdficon.png" alt="pdf" /></a>
</td>
</tr>
</table>
<p class="programversion">IMGT/GENE-DB program version: <a href="/IMGTgenedbdoc/programversions.html">3.1.35</a> (3 March 2022)<br/>
<a href="/IMGTgenedbdoc/dataupdates.html">IMGT/GENE-DB data updates </a>
</p>

<h2 class="subtitle"><a href="/IMGTScientificChart/SequenceDescription/IMGTreferencesequences.html">IMGT/GENE-DB reference sequences</a> in FASTA format:</h2>
<h2 class="subtitle">Amino acids  sequences with gaps according to the <a href="/IMGTScientificChart/Numbering/IMGTnumberingCR.html">IMGT unique numbering</a><br /> for <span class="casimir">F+ORF+in-frame P with IMGT Gaps</span> alleles including orphons</h2>
<hr>
<pre>
The FASTA header contains 15 fields separated by '|':

1. IMGT/LIGM-DB accession number(s)
2. IMGT gene and allele name
3. species
4. IMGT allele functionality
5. exon(s), region name(s), or extracted label(s)
6. start and end positions in the IMGT/LIGM-DB accession number(s)
7. number of nucleotides in the IMGT/LIGM-DB accession number(s)
8. codon start, or 'NR' (not relevant) for non coding labels
9. +n: number of nucleotides (nt) added in 5' compared to the corresponding label extracted from IMGT/LIGM-DB
10. +n or -n: number of nucleotides (nt) added or removed in 3' compared to the corresponding label extracted from IMGT/LIGM-DB
11. +n, -n, and/or nS: number of added, deleted, and/or substituted nucleotides to correct sequencing errors, or 'not corrected' if non corrected sequencing errors
12. number of amino acids (AA): this field indicates that the sequence is in amino acids
13. number of characters in the sequence: nt (or AA)+IMGT gaps=total
14. partial (if it is)
15. reverse complementary (if it is)
</pre>
<hr/>
<b>Number of results = 166</b><br />
<pre>
>IMGT000068|IGKV1-108*02|Mus musculus_C57BL/6J|P|V-REGION|766495..766724|230 nt|1| | ||76 AA|76+35=111|partial in 5'| |
........................SSKSLVHS.NGNSYLDWHLQKPGQSLQLLIYEV...
....SKRNSGVP.DRFSGSG..SGTDFTLKISRVEPEDLGVYYCFQGTHLP
>D00080|IGKV1-110*01|Mus musculus_BALB/c|F|V-REGION|780..1081|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLSLPVSLGDQASISCRSSQSLVHS.NGNTYLHWYLQKPGQSPKLLIYKV...
....SNRFSGVP.DRFSGSG..SGTDFTLKISRVEAEDLGVYFCSQSTHVP
>M28132|IGKV1-110*02|Mus musculus_NZB/BINJ|F|V-REGION|775..1076|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLSLPVSLGDQASISCRSSQSLVHS.NGNTYLYWYLQKPGQSPKLLIYRV...
....SNRFSGVP.DRFSGSG..SGTDFTLKISRVEAEDLGVYFCFQGTHVP
>AJ231208|IGKV1-115*01|Mus musculus_C3H|P|V-REGION|732..1033|302 nt|1| | | |100 AA|100+11=111| | |
DVLLTQTPLFLPVSLGDQASISCSSSQSLVHS.NGNSYLEWHLQKSGQSLQLLIYEV...
....SKRHSGVP.DRFSGSG..SGTDFTLKISRVEPEDLGVYYCFQGTHLP
>D00081|IGKV1-117*01|Mus musculus_BALB/c|F|V-REGION|781..1082|302 nt|1| | | |100 AA|100+11=111| | |
DVLMTQTPLSLPVSLGDQASISCRSSQSIVHS.NGNTYLEWYLQKPGQSPKLLIYKV...
....SNRFSGVP.DRFSGSG..SGTDFTLKISRVEAEDLGVYYCFQGSHVP
>M28134|IGKV1-117*02|Mus musculus_CE/J|F|V-REGION|787..1088|302 nt|1| | ||100 AA|100+11=111| | |
DVVMTQTPLSLPVSLGDQASISCRSSQSIVHS.NGNTYLEWYLQKPGQSPKLLIYKV...
....SNRLSGVP.DRFSGSG..SGTDFTLKISRVEAEDLGVYYCFQGSHVP
>D00082|IGKV1-122*01|Mus musculus_BALB/c|F|V-REGION|528..829|302 nt|1| | | |100 AA|100+11=111| | |
DAVMTQTPLSLPVSLGDQASISCRSSQSLENS.NGNTYLNWYLQKPGQSPQLLIYRV...
....SNRFSGVL.DRFSGSG..SGTDFTLKISRVEAEDLGVYFCLQVTHVP
>AJ231197|IGKV1-131*01|Mus musculus_C57BL/6|ORF|V-REGION|572..873|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLTLSVTIGQPASISCKSSQSLLHS.NGKTYLNWLQQRPGQAPKRLFYLV...
....SKLDPGIP.DSFSGSG..SETDFTLKISRVEAEDLGVYYCLQGTHFP
>AJ231198|IGKV1-132*01|Mus musculus_C57BL/6|F|V-REGION|524..825|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLSLSVTIGQPASISCKSSQSLLYS.NGKTYLNWLQQRPGQAPKHLMYQV...
....SKLDPGIP.DRFSGSG..SETDFTLKISRVEAEDLGVYYCLQGTYYP
>Z72382|IGKV1-133*01|Mus musculus_129/Sv|F|V-REGION|599..900|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLTLSVTIGQPASISCKSSQSLLYS.NGKTYLNWLLQRPGQSPKRLIYLV...
....SKLDSGVP.DRFTGSG..SGTDFTLKISRVEAEDLGVYYCVQGTHFP
>Z72384|IGKV1-135*01|Mus musculus_129/Sv|F|V-REGION|3597..3898|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLTLSVTIGQPASISCKSSQSLLDS.DGKTYLNWLLQRPGQSPKRLIYLV...
....SKLDSGVP.DRFTGSG..SGTDFTLKISRVEAEDLGVYYCWQGTHFP
>AJ231200|IGKV1-35*01|Mus musculus_C3H|ORF|V-REGION|540..841|302 nt|1| | | |100 AA|100+11=111| | |
DVVMTQTPLTLSATIGQSASVSCRSSQSLLHS.NGNTYLNWFVQRPGPSPQLLIYGV...
....FEQESGVP.DRFSGSG..SGTDFTLKISRVEAEDLGVYYCMQATYEP
>AJ231206|IGKV1-88*01|Mus musculus_C57BL/6|F|V-REGION|576..877|302 nt|1| | | |100 AA|100+11=111| | |
DVVVTQTPLSLPVSFGDQVSISCRSSQSLANS.YGNTYLSWYLHKPGQSPQLLIYGI...
....SNRFSGVP.DRFSGSG..SGTDFTLKISTIKPEDLGMYYCLQGTHQP
>AJ231207|IGKV1-99*01|Mus musculus_C3H|F|V-REGION|637..938|302 nt|1| | | |100 AA|100+11=111| | |
DVVLTQTPLSLPVNIGDQASISCKSTKSLLNS.DGFTYLDWYLQKPGQSPQLLIYLV...
....SNRFSGVP.DRFSGSG..SGTDFTLKISRVEAEDLGVYYCFQSNYLP
>M54906|IGKV10-94*01|Mus musculus_A/J|F|V-REGION|459..745|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCSASQGI......SNYLNWYQQKPDGTVKLLIYYT...
....SSLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKLP
>M54908|IGKV10-94*02|Mus musculus_PERU|F|V-REGION|459..745|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCSASQGI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSNLP
>M54904|IGKV10-94*03|Mus musculus_AKR|F|V-REGION|459..745|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKLP
>AF441453|IGKV10-94*04|Mus musculus domesticus|F|V-REGION|146..432|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKLP
>AF441460|IGKV10-94*05|Mus musculus_O20/A|F|V-REGION|146..432|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKRP
>AF441455|IGKV10-94*06|Mus musculus domesticus|F|V-REGION|149..435|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQDSKLP
>AF441452|IGKV10-94*07|Mus musculus castaneus|F|V-REGION|147..433|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQRPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKLP
>AF441457|IGKV10-94*08|Mus musculus molossinus_MOLF/Ei|F|V-REGION|149..435|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKLP
>AF029261|IGKV10-95*01|Mus musculus_BALB/c|F|V-REGION|328..614|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASEDI......STYLNWYQQKPDGTVKLLIYYT...
....SGLHSGVP.SRFSGSG..SGADYSLTISNLEPEDIATYYCQQYSKLP
>M15520|IGKV10-96*01|Mus musculus_A/J|F|V-REGION|452..737|286 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQGNTLP
>M54907|IGKV10-96*02|Mus musculus_PERU|F|V-REGION|456..742|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQGSTLP
>M54903|IGKV10-96*03|Mus musculus_AKR|F|V-REGION|453..739|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQDSKHP
>AF441444|IGKV10-96*04|Mus musculus musculus|F|V-REGION|146..432|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQDSKHP
>AF441445|IGKV10-96*05|Mus musculus domesticus|P|V-REGION|146..432|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWY*QKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQDSKLP
>AF441443|IGKV10-96*06|Mus musculus castaneus_CAST/Ei|ORF|V-REGION|146..432|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNGYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQDSKHP
>IMGT000068|IGKV11-106*02|Mus musculus_C57BL/6J|F|V-REGION|794074..794360|287 nt|1| | | |95 AA|95+16=111| | |
DVLMTQSPSSLSASLGERVSLTCQASQGI......SNNLNWYQQTPGKAPRLLIYDA...
....SKLEDGVP.SRFSGTG..YRTDFNFTISSLEEEDVATYFCLQHRYLP
>AJ231256|IGKV11-125*01|Mus musculus_129/Sv|F|V-REGION|323..609|287 nt|1| | | |95 AA|95+16=111| | |
DVQMIQSPSSLSASLGDIVTMTCQASQGT......SINLNWFQQKPGKAPKLLIYGA...
....SNLEDGVP.SRFSGSR..YGTDFTLTISSLEDEDMATYFCLQHSYLP
>IMGT000068|IGKV12-104-2*01|Mus musculus_C57BL/6J|F|V-REGION|844378..844663|286 nt|1| | | |95 AA|95+16=111| | |
DIRMIQTPASLSGSLGESVTITCQASQDI......GKSLLWYQQKTGNPPKILIYTT...
....SNLADGIS.SRVSGSG..SGTQFFLKFSSLKPEDTATYYCCQGYWLF
>AJ235951|IGKV12-38*01|Mus musculus_C57BL/6|F|V-REGION|319..605|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLAASVGETVTITCRASENI......YYSLAWYQQKQGKSPQLLIYNA...
....NSLEDGVP.SRFSGSG..SGTQYSMKINSMQPEDTATYFCKQAYDVP
>AJ235952|IGKV12-40*01|Mus musculus_C57BL/6|P|V-REGION|343..629|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSASVGETVTITCQASENI......ASDLAWY*QKQGKSPQLLVYDA...
....RNLADGVP.SRFSGSG..SGTHYSLNIHSLQSEDVARYYCQHYYGTP
>AJ235953|IGKV12-41*01|Mus musculus_C57BL/6|F|V-REGION|299..585|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSASVGETVTITCRASGNI......HNYLAWYQQKQGKSPQLLVYNA...
....KTLADGVP.SRFSGSG..SGTQYSLKINSLQPEDFGSYYCQHFWSTP
>V00778|IGKV12-41*02|Mus musculus_BALB/c|F|V-REGION|559..845|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSASVGETVTITCRASGNI......HNYLAWYQQKQGKSPQLLVYNA...
....KTLADGVP.SRFSGSG..SGTQYSLKINSLQPEDFGSYYCQHFWSTP
>AJ235955|IGKV12-44*01|Mus musculus_C3H|F|V-REGION|308..594|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSASVGETVTITCRASENI......YSYLAWYQQKQGKSPQLLVYNA...
....KTLAEGVP.SRFSGSG..SGTQFSLKINSLQPEDFGSYYCQHHYGTP
>AJ235956|IGKV12-46*01|Mus musculus_C57BL/6|F|V-REGION|317..603|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSVSVGETVTITCRASENI......YSNLAWYQQKQGKSPQLLVYAA...
....TNLADGVP.SRFSGSG..SGTQYSLKINSLQSEDFGSYYCQHFWGTP
>AJ235959|IGKV12-47*01|Mus musculus_C57BL/6|ORF|V-REGION|290..564|275 nt|1| | | |91 AA|91+16=107| | |
DIQVTQSPASLSAPVGESVSITCKASEEI......YSALNWYQQKPGKSPQLLIYYA...
....TSLGDDVP.SRFSGSK..SGTQYSLKISSLQPEDLATYYCEQS
>IMGT000068|IGKV12-49*02|Mus musculus_C57BL/6J|P|V-REGION|2170924..2171145|222 nt|1| | | |74 AA|74+16=90|partial in 3'|rev-compl|
DIQLTHSPASLSLSVRESIIVRCLKSEKI......YSELHWYQQKQGKSSQLLIYST...
....TSLADDVP.SRFSGSG..SGTQYSLS
>AJ235950|IGKV12-89*01|Mus musculus_C57BL/6|F|V-REGION|412..698|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSASVGETVTITCGASENI......YGALNWYQRKQGKSPQLLIYGA...
....TNLADGMS.SRFSGSG..SGRQYSLKISSLHPDDVATYYCQNVLSTP
>AJ235949|IGKV12-98*01|Mus musculus_C3H|F|V-REGION|307..593|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASQSASLGESVTITCLASQTI......GTWLAWYQQKPGKSPQLLIYAA...
....TSLADGVP.SRFSGSG..SGTKFSFKISSLQAEDFVSYYCQQLYSTP
>J00546|IGKV12-e*01|Mus musculus_BALB/c|F|V-REGION|12..298|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPASLSVSVGETVTITCRASENI......YSNLAWLFSRNRENPPSLVYAA...
....TNLADGVP.SRFSGSG..SGTQYSLKINSQQPEDFGSYYCQHFWSAP
>IMGT000068|IGKV13-50-1*01|Mus musculus_C57BL/6J|P|V-REGION|2147702..2147996|295 nt|1| | | |98 AA|98+17=115| | |
DISMTQSPS.LSASLGDRVILTCRPYEGI......NDWLA*CQ*KLGNVHKLLIYHP...
....ISAESGVS.SRFSGSG..FGKDFTFTIINLQPEEIATYYCQQYFSILFCCL
>AC158673|IGKV13-57-2*01|Mus musculus_C57BL/6J|P|V-REGION|100695..100989|295 nt|1| | ||98 AA|98+17=115| | |
DISMTQSPS.LSASLGDRVILTCRPYKGI......NDWLA*CQ*KLRNVHKLLIYHP...
....ISVE*GVS.SRISGSG..FGKDFTFTISNL*PEDIATYYCQQYFSILFCCL
>AJ231271|IGKV13-76*01|Mus musculus_C57BL/6|P|V-REGION|791..1077|287 nt|1| | | |95 AA|95+16=111| | |
DIQMT*PSSSLSSSLGDRVTITCRPDEGI......NDW*AWYQWKPGNVPKLLIYHC...
....TSVESGVS.SRFSGSE..FGKDFTLAVSKLQCENIDTYYCQQCFSIP
>IMGT000068|IGKV13-77-1*01|Mus musculus_C57BL/6J|P|V-REGION|1540577..1540863|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSVSLGDRVTISCRPDVSI......NDW*P*FQ*KPGNAHKHLIYHS...
....TSVESGIP.SRFSGSE..FGKDFTLTVNKLKCENIATYYCQQYFFIT
>AJ231276|IGKV13-82*01|Mus musculus_C57BL/6|P|V-REGION|362..647|286 nt|1| | | |95 AA|95+16=111| | |
DIQMA*SSSSLYSSLGDRVTITCRPDEGI......NDW*AWYQQKPGNVPKFLIYHS...
....TSVESGVS.SRFSGSE..YGKDFTLAVSNLQHENIATYYCEQYLVYL
>AJ231273|IGKV13-84*01|Mus musculus_C57BL/6|F|V-REGION|332..618|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSSSSFSVSLGDRVTITCKASEDI......YNRLAWYQQKPGNAPRLLISGA...
....TSLETGVP.SRFSGSG..SGKDYTLSITSLQTEDVATYYCQQYWSTP
>AJ231274|IGKV13-85*01|Mus musculus_C57BL/6|F|V-REGION|329..615|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSSSYLSVSLGGRVTITCKASDHI......NNWLAWYQQKPGNAPRLLISGA...
....TSLETGVP.SRFSGSG..SGKDYTLSITSLQTEDVATYYCQQYWSTP
>M35154|IGKV13-85*02|Mus musculus_BALB/c|F|V-REGION|177..432|256 nt|1| | | |85 AA|85+16=101|partial in 3'| |
DIQMTQSSSYLSVSLGGRVTITCKASEHI......NSWLAWYQQKPGNAPRLLISGA...
....TSLETGVP.SRFSGSG..SGKDYTLSITSLQTEDVAT
>M35155|IGKV13-85*03|Mus musculus_AKR|F|V-REGION|177..432|256 nt|1| | | |85 AA|85+16=101|partial in 3'| |
DIQMTQSSSSLSVSLGDRVTITCKASEHI......NSWLAWYQQKPGNAPRLLISGA...
....TSLETGVP.SRFSGSA..SGKDYTLSITSLQTEDVAT
>AJ231243|IGKV14-100*01|Mus musculus_C3H|F|V-REGION|273..559|287 nt|1| | | |95 AA|95+16=111| | |
DILMTQSPSSMSVSLGDTVSITCHASQGI......SSNIGWLQQKPGKSFKGLIYHG...
....TNLEDGVP.SRFSGSG..SGADYSLTISSLESEDFADYYCVQYAQFP
>V01563|IGKV14-111*01|Mus musculus_BALB/c|F|V-REGION|744..1030|287 nt|1| | | |95 AA|95+16=111| | |
DIKMTQSPSSMYASLGERVTITCKASQDI......NSYLSWFQQKPGKSPKTLIYRA...
....NRLVDGVP.SRFSGSG..SGQDYSLTISSLEYEDMGIYYCLQYDEFP
>AY591690|IGKV14-126*01|Mus musculus_C57BL/6J|F|V-REGION|2180..2466|287 nt|1| | | |95 AA|95+16=111| | |
DIKMTQSPSSMYASLGERVTITCKASQDI......KSYLSWYQQKPWKSPKTLIYYA...
....TSLADGVP.SRFSGSG..SGQDYSLTISSLESDDTATYYCLQHGESP
>AJ231241|IGKV14-130*01|Mus musculus_C57BL/6|F|V-REGION|407..693|287 nt|1| | | |95 AA|95+16=111| | |
EIQMTQSPSSMSASLGDRITITCQATQDI......VKNLNWYQQKPGKPPSFLIYYA...
....TELAEGVP.SRFSGSG..SGSDYSLTISNLESEDFADYYCLQFYEFP
>IMGT000068|IGKV14-87-2*01|Mus musculus_C57BL/6J|F|V-REGION|1334543..1334829|287 nt|1| | | |95 AA|95+16=111| |rev-compl|
DIQMTQSSSFLSASLGDHLTINCRASKDI......NKYFAWVQQKPRKAPRMLIHFA...
....STLLPGVP.EKFSGSG..SGTDFSLTIRNIESEDIAMYYCLQYSEHP
>AJ231268|IGKV15-101*01|Mus musculus_C3H|P|V-REGION|437..723|287 nt|1| | | |95 AA|95+16=111| | |
NTQMNQTPSTLSASSGEQVIINC*ASENI......NSWSSWHQQIPGNAPQILIYKA...
....STLHTWVP.SRFSGS*..SGADYSLISSSLQPEDIDTYYCVQTKSFP
>AJ231270|IGKV15-102*01|Mus musculus_C3H|P|V-REGION|316..602|287 nt|1| | | |95 AA|95+16=111| | |
DT*MNQSPCTLSASIRE*IVINYHTSEKV......NTWLSCNQ*KLGNYPKLLIYNI...
....SNLHTGVP.SRLSGSR..SGTDYSLFISSLQLEDIATYYCVQSSSLS
>AJ231269|IGKV15-103*01|Mus musculus_C3H|ORF|V-REGION|547..833|287 nt|1| | | |95 AA|95+16=111| | |
DIQMNQSPSSLSASLGDTITITCHASQNI......NVWLSWYQQKPGNIPKLLIYKA...
....SNLHTGVP.SRFSGSG..SGTGFTLTISSLQPEDIATYYCQQGQSYP
>IMGT000068|IGKV15-97*02|Mus musculus_C57BL/6J|P|V-REGION|1045925..1046197|273 nt|1| | | |91 AA|91+20=111| | |
DIQMTQSPSTLSASLGDRVTITCHASQNI......KSWLAWCQQKPGNSPKLLIYEA...
....TKLQIG.....FSGSG..YGTDFTISISSLQSEDIVIYHCL*HETC*
>AJ235936|IGKV16-104*01|Mus musculus_C3H|F|V-REGION|345..632|288 nt|1| | | |96 AA|96+16=112| | |
DVQITQSPSYLAASPGETITINCRASKSI......SKYLAWYQEKPGKTNKLLIYSG...
....STLQSGIP.SRFSGSG..SGTDFTLTISSLEPEDFAMYYCQQHNEYPY
>AJ231258|IGKV17-121*01|Mus musculus_C57BL/6|F|V-REGION|525..811|287 nt|1| | | |95 AA|95+16=111| | |
ETTVTQSPASLSMAIGEKVTIRCITSTDI......DDDMNWYQQKPGEPPKLLISEG...
....NTLRPGVP.SRFSSSG..YGTDFVFTIENMLSEDVADYYCLQSDNLP
>AJ231259|IGKV17-127*01|Mus musculus_C57BL/6|F|V-REGION|520..806|287 nt|1| | | |95 AA|95+16=111| | |
ETTVTQSPASLSVATGEKVTIRCITSTDI......DDDMNWYQQKPGEPPKLLISEG...
....NTLRPGVP.SRFSSSG..YGTDFVFTIENTLSEDVADYYCLQSDNMP
>AJ235929|IGKV17/OR16-3*01|Mus musculus_C57BL/6|ORF|V-REGION|431..717|287 nt|1| | | |95 AA|95+16=111| | |
ETTVTQSPTSLSMATGEKVTIRCITSTDI......DDDMNWCQQKPGEPPNLLISEG...
....NTLRPGVP.SRFSSSG..YGTDFLLTIENTLSEDVADYYYLQSDNMP
>AJ235932|IGKV17/OR19-2*01|Mus musculus_C57BL/6|ORF|V-REGION|293..579|287 nt|1| | ||95 AA|95+16=111| | |
ETTVTQSPTSLSMATGEKVTIRCITSTDI......DDDMNGCQQKPGDPSNLLISEG...
....NTLRPGVP.SRFSSSG..YGTDFLLTIENTLSEDVADYYYLQSDIMP
>AJ235966|IGKV18-36*01|Mus musculus_C3H|F|V-REGION|352..638|287 nt|1| | | |95 AA|95+16=111| | |
TGETTQAPASLSFSLGETATLSCRSSESV......GSYLAWYQQKAEQVPRLLIHSA...
....STRAGGVP.VRFSGTG..SGTDFTLTISSLEPEDAAVYYCQPFKSWS
>AJ235935|IGKV19-93*01|Mus musculus_C57BL/6|F|V-REGION|292..578|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSASLGGKVTITCKASQDI......NKYIAWYQHKPGKGPRLLIHYT...
....STLQPGIP.SRFSGSG..SGRDYSFSISNLEPEDIATYYCLQYDNLL
>S65298|IGKV19-93*02|Mus musculus|F|V-REGION|290..576|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSASLGGKVTITCKASQDI......NKYIAWYQHKPGKGPRLLIHYT...
....STLQPGIP.SRFSGSG..SGRDYSFSISNLEPEDIATYYCLQYDNLL
>AJ132683|IGKV2-109*01|Mus musculus_C3H|F|V-REGION|569..870|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAFSNPVTLGTSASISCRSSKSLLHS.NGITYLYWYLQKPGQSPQLLIYQM...
....SNLASGVP.DRFSSSG..SGTDFTLRISRVEAEDVGVYYCAQNLELP
>K02418|IGKV2-109*02|Mus musculus_BALB/c|F|V-REGION|558..859|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAFSNPVTLGTSASISCRSSKSLLHS.NGITYLYWYLQKPGQSPQLLIYQM...
....SNLASGVP.DRFSSSG..SGTDFTLRISRVEAEDVGVYYCAQNLELP
>M80409|IGKV2-109*03|Mus musculus castaneus|F|V-REGION|673..974|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAFSNPVTLGTSASISCSSSKSLLHS.NGITYLYWYLQRPGQSPQLLIYRM...
....SNLASGVP.DRFSGSG..SGTDFTLRISRVEAEDVGVYYCAQMLERP
>M80410|IGKV2-109*04|Mus musculus castaneus|F|V-REGION|386..687|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAFSNPVTLGTSASISCRSSKSLLHS.DGITYLYWYLQRPGQSPQLLIYRM...
....SNLASGVP.DRFSGSG..SGTDFTLRISRVEAEDVGVYYCAQMLEFP
>J00562|IGKV2-112*01|Mus musculus_BALB/c|F|V-REGION|548..847|300 nt|1| | | |100 AA|100+11=111| | |
DIVITQDELSNPVTSGESVSISCRSSKSLLYK.DGKTYLNWFLQRPGQSPQLLIYLM...
....STRASGVS.DRFSGSG..SGTDFTLEISRVKAEDVGVYYCQQLVEYP
>M80407|IGKV2-112*02|Mus musculus castaneus|F|V-REGION|617..917|301 nt|1| | ||100 AA|100+11=111| | |
DIVITQDELSNPVTSGESVSISCRSSKSLLYK.DGKTYLNWFLQRPGQSPQLLVYWM...
....STRASGVS.DRFSGSG..SGTDFTLEISRVKAEDVGVYYCQQVVEYP
>IMGT000068|IGKV2-116*01|Mus musculus_C57BL/6J|F|V-REGION|606681..606982|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAFSNPVTLGTSASISCRSSKNLLHS.NGITYLYWYLQRPGQSPQLLIYRV...
....SNLASGVP.NRFSGSE..SGTDFTLRISRVEAEDVGVYYCAQLLELP
>AJ231263|IGKV2-137*01|Mus musculus_C3H|F|V-REGION|444..745|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAPSVPVTPGESVSISCRSSKSLLHS.NGNTYLYWFLQRPGQSPQLLIYRM...
....SNLASGVP.DRFSGSG..SGTAFTLRISRVEAEDVGVYYCMQHLEYP
>K02417|IGKV2-a*01|Mus musculus_BALB/c|F|V-REGION|349..650|302 nt|1| | | |100 AA|100+11=111| | |
DIVMTQAAFSNPVTLGTSASISCRSSKSLLHS.SGNTYLYWFLQKPGQSPQLLIYYI...
....SNLASGVP.DRFSGSG..SGTDFTLRISRVEAEDVGVYYCMQGLEYP
>AC122918|IGKV20-101-2*01|Mus musculus_C57BL/6J|F|V-REGION|132212..132493|282 nt|1| | | |94 AA|94+17=111| | |
NIQVIQSPF.LSASVGERVTISCKTHQHI......NSSIAWYQQKVGKAPILLIRDA...
....SFSLTDTP.SRFTGNG..FGTDFTLSISSMQSQDGATYFCQQHFHYY
>X16955|IGKV3-1*01|Mus musculus_BALB/c|F|V-REGION|499..797|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCRASESVEY..YGTSLMQWYQQKPGQPPKLLIYAA...
....SNVESGVP.ARFSGSG..SGTDFSLNIHPVEEDDIAMYFCQQSRKVP
>K02160|IGKV3-10*01|Mus musculus_BALB/c|F|V-REGION|497..795|299 nt|1| | | |99 AA|99+12=111| | |
NIVLTQSPASLAVSLGQRATISCRASESVDS..YGNSFMHWYQQKPGQPPKLLIYLA...
....SNLESGVP.ARFSGSG..SRTDFTLTIDPVEADDAATYYCQQNNEDP
>K02159|IGKV3-12*01|Mus musculus_BALB/c|F|V-REGION|362..660|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCRASKSVST..SGYSYMHWYQQKPGQPPKLLIYLA...
....SNLESGVP.ARFSGSG..SGTDFTLNIHPVEEEDAATYYCQHSRELP
>X16954|IGKV3-2*01|Mus musculus_BALB/c|F|V-REGION|520..818|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCRASESVDN..YGISFMNWFQQKPGQPPKLLIYAA...
....SNQGSGVP.ARFSGSG..SGTDFSLNIHPMEEDDTAMYFCQQSKEVP
>IMGT000068|IGKV3-25-1*01|Mus musculus_C57BL/6J|P|V-REGION|2655020..2655321|302 nt|1| | ||100 AA|100+11=111| | |
DIVLTQTSACFVVSLGQRDNISCKSCESVTDT.LCNNSMQWYQQNPEQLPIVLIYEA...
....YSVESGVY.VRFCGSW..CETDFKLTIYPVEDSNAASYYCQQSKEFF
>K02162|IGKV3-3*01|Mus musculus_BALB/c|F|V-REGION|328..626|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATIFCRASQSVDY..NGISYMHWFQQKPGQPPKLLIYAA...
....SNLESGIP.ARFSGSG..SGTDFTLNIHPVEEEDAATYYCQQSIEDP
>Y15968|IGKV3-4*01|Mus musculus_C57BL/6|F|V-REGION|420..718|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCKASQSVDY..DGDSYMNWYQQKPGQPPKLLIYAA...
....SNLESGIP.ARFSGSG..SGTDFTLNIHPVEEEDAATYYCQQSNEDP
>K02161|IGKV3-5*01|Mus musculus_BALB/c|F|V-REGION|497..795|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCRASESVDS..YGNSFMHWYQQKPGQPPKLLIYRA...
....SNLESGIP.ARFSGSG..SRTDFTLTINPVEADDVATYYCQQSNEDP
>K02158|IGKV3-7*01|Mus musculus_BALB/c|F|V-REGION|362..660|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCRASQSVST..SSYSYMHWYQQKPGQPPKLLIKYA...
....SNLESGVP.ARFSGSG..SGTDFTLNIHPVEEEDTATYYCQHSWEIP
>IMGT000068|IGKV3-7*02|Mus musculus_C57BL/6J|F|V-REGION|3062103..3062401|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCRASQSVST..SSYSYMHWYQQKPGQPPKLLIKYA...
....SNLESGVP.ARFSGSG..SGTDFTLNIHPVEEEDTATYYCQHSWEIP
>Y15971|IGKV3-8*01|Mus musculus_C57BL/6|P|V-REGION|284..582|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQAPPSLDVSQG*RATISCRTSKSVRT..SSYSYMHWYQQKPGQPPKLLNLCA...
....SNQVSRVP.ARFSGSG..SGTDFTLKIHPVEEEDAATYFCQQSNENP
>Y15972|IGKV3-9*01|Mus musculus_C57BL/6|F|V-REGION|420..718|299 nt|1| | | |99 AA|99+12=111| | |
DIVLTQSPASLAVSLGQRATISCQASESVSF..AGTSLMHWYQQKPGQPPKLLIYRA...
....SNLESGVP.ARFSGSG..SESDFTLTIDPVEEDDAAMYYCMQSMEDP
>AJ235938|IGKV4-50*01|Mus musculus_C57BL/6|F|V-REGION|347..632|286 nt|1| | | |95 AA|95+17=112| | |
ENVLTQSPAIMSASLGEKVTMSCRASSSV.......NYMYWYQQKSDASPKLWIYYT...
....SNLAPGVP.ARFSGSG..SGNSYSLTISSMEGEDAATYYCQQFTSSPS
>V01565|IGKV4-51*01|Mus musculus_BALB/c|F|V-REGION|601..892|292 nt|1| | | |97 AA|97+15=112| | |
ENVLTQSPAIMAASLGEKVTMTCSASSSVS.....SSYLHWYQQKSGTSPKLWIYGT...
....SNLASGVP.ARFSGSG..AGISYSLTISSMEAENDATYYCQQWSGYPF
>AJ239198|IGKV4-52*01|Mus musculus_C57BL/6|ORF|V-REGION|431..716|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAILSASPGEKVTMTCSASSSV.......SYMYRYQQKPGSSPKPWIYGT...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCQQYHSYPP
>AJ231231|IGKV4-53*01|Mus musculus_C57BL/6|F|V-REGION|380..671|292 nt|1| | | |97 AA|97+15=112| | |
EIVLTQSPALMAASPGEKVTITCSVSSSIS.....SSNLHWYQQKSETSPKPWIYGT...
....SNLASGVP.VRFSGSG..SGTSYSLTISSMEAEDAATYYCQQWSSYPL
>AJ231223|IGKV4-54*01|Mus musculus_C57BL/6|ORF|V-REGION|385..670|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAILSASPGEKVTMTCSASSSV.......SYMYRYQQKPGSSPKPWIYGT...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCQQYHSYPP
>AJ231225|IGKV4-55*01|Mus musculus_C57BL/6|F|V-REGION|454..739|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGEKVTMTCSASSSV.......SYMYWYQQKPGSSPRLLIYDT...
....SNLASGVP.VRFSGSG..SGTSYSLTISRMEAEDAATYYCQQWSSYPP
>AJ231220|IGKV4-56*01|Mus musculus_C57BL/6|P|V-REGION|368..653|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGQKVTITCSAISSV.......NYMHWYQQKPGSSPKLWIYAT...
....SKLALGVP.ACFSGSG..SGTSYSLTISSMVAEDATSYFCHQWSSYPP
>AJ231221|IGKV4-57*01|Mus musculus_C57BL/6|F|V-REGION|357..642|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGEKVTITCSASSSV.......SYMHWFQQKPGTSPKLWIYST...
....SNLASGVP.ARFSGSG..SGTSYSLTISRMEAEDAATYYCQQRSSYPP
>AC158673|IGKV4-57-1*01|Mus musculus_C57BL/6J|F|V-REGION|121283..121574|292 nt|1| | ||97 AA|97+15=112| | |
ENVLTQSPAIMSASPGEKVTMTCRASSSVS.....SSYLHWYQQKSGASPKLWIYST...
....SNLASGVP.ARFSGSG..SGTSYSLTISSVEAEDAATYYCQQYSGYPL
>K00884|IGKV4-58*01|Mus musculus_BALB/c|F|V-REGION|214..505|292 nt|1| | | |97 AA|97+15=112| | |
ENVLTQSPAIMAASLGQKVTMTCSASSSVS.....SSYLHWYQQKSGASPKPLIHRT...
....SNLASGVP.ARFSGSG..SGTSYSLTISSVEAEDDATYYCQQWSGYPF
>AJ231234|IGKV4-59*01|Mus musculus_C57BL/6|F|V-REGION|401..686|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGEKVTMTCSASSSV.......SYMHWYQQKSGTSPKRWIYDT...
....SKLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCQQWSSNPP
>AC156953|IGKV4-60*01|Mus musculus_C57BL/6J|P|V-REGION|107939..108230|292 nt|1| | | |97 AA|97+15=112| | |
EIVLTQSPALMAASPGEKVTITCSVSSSIS.....SSYLHWYQQKSGISPKPWIYGT...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCQQWSSSPP
>AJ231209|IGKV4-61*01|Mus musculus_C57BL/6|F|V-REGION|412..697|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGEKVTISCSASSSV.......SYMYWYQQKPGSSPKPWIYRT...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCQQYHSYPP
>AJ231210|IGKV4-62*01|Mus musculus_C57BL/6|ORF|V-REGION|385..670|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGEKVTMTCSASSSV.......SYMYWYQQKPGSSPRLWIYDT...
....SNLVSGVP.ARFSGSR..SGTSYSLTISSMEAEDAATYYCQQYSGYPS
>AJ231211|IGKV4-63*01|Mus musculus_C57BL/6|F|V-REGION|381..666|286 nt|1| | | |95 AA|95+17=112| | |
ENVLTQSPAIMSASPGEKVTMTCSASSSV.......SYMHWYQQKSSTSPKLWIYDT...
....SKLASGVP.GRFSGSG..SGNSYSLTISSMEAEDVATYYCFQGSGYPL
>IMGT000068|IGKV4-65*02|Mus musculus|P|V-REGION|1796837..1797128|292 nt|1| | | |97 AA|97+15=112| |rev-compl|
EIGLIQSSTIMAASPGEKVTITCSVSSSVS.....SSY*HWYQQKSGASPKLWIYGS...
....FNLASEVS.AHFSGSG..SGTSYSLTISSMEAEDAATYYCQQVTCSLP
>AJ231222|IGKV4-68*01|Mus musculus_C57BL/6|F|V-REGION|450..735|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPALMSASPGEKVTMTCSASSSV.......SYMYWYQQKPRSSPKPWIYLT...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCQQWSSNPP
>AJ235942|IGKV4-69*01|Mus musculus_C57BL/6|F|V-REGION|352..633|282 nt|1| | | |94 AA|94+17=111| | |
QILLTQSPAIMSASPGEKVTMTCSASSSV.......SYMHWYQQKPGSSPKPWIYDT...
....SNLASGFP.ARFSGSG..SGTSYSLIISSMEAEDAATYYCHQRSSYP
>AJ235943|IGKV4-70*01|Mus musculus_C57BL/6|F|V-REGION|351..632|282 nt|1| | | |94 AA|94+17=111| | |
QIVLTQSPAIMSASPGEKVTMTCSASSSI.......SYMHWYQQKPGTSPKRWIYDT...
....SKLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCHQRSSYP
>AJ231218|IGKV4-71*01|Mus musculus_C57BL/6|F|V-REGION|376..661|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAIMSASPGEKVTMTCSASSSV.......SYMHWYQQKPGSSPRLWIYLT...
....FNLASGVP.ARFSGSG..SGTSYSLSISSMEAEDAATYYCQQWSSNPP
>AJ231219|IGKV4-72*01|Mus musculus_C57BL/6|F|V-REGION|419..704|286 nt|1| | | |95 AA|95+17=112| | |
QIVLSQSPAILSASPGEKVTMTCRASSSV.......SYMHWYQQKPGSSPKPWIYAT...
....SNLASGVP.ARFSGSG..SGTSYSLTISRVEAEDAATYYCQQWSSNPP
>AJ231216|IGKV4-73*01|Mus musculus_C57BL/6|ORF|V-REGION|372..663|292 nt|1| | | |97 AA|97+15=112| | |
QIVLTQSPAIMSASPGERVTMTCSASSSVS.....SSYLYWYQQKSGSSPKLWIYSI...
....SNLASGVP.ARFSGSG..SGTSYSLTINSMEAEDAATYYCQQWSSNPP
>AJ231217|IGKV4-74*01|Mus musculus_C57BL/6|F|V-REGION|427..718|292 nt|1| | | |97 AA|97+15=112| | |
QIVLTQSPAIMSASLGERVTMTCTASSSVS.....SSYLHWYQQKPGSSPKLWIYST...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATYYCHQYHRSPP
>AJ231227|IGKV4-75*01|Mus musculus_C57BL/6|P|V-REGION|391..682|292 nt|1| | | |97 AA|97+15=112| | |
KNVLTQSPAIMATSPGENVTMTCSASSNVS.....SGKLHW*KQ*SGTSPKLWIYST...
....SNLASGVP.ARFSGSG..SGTSYSLTISSVEAEDDANYYCQKWSGYPP
>AJ235940|IGKV4-77*01|Mus musculus_C57BL/6|P|V-REGION|444..729|286 nt|1| | | |95 AA|95+17=112| | |
QIVLTQSPAILSASPGEKVTMTCSASSSV.......SYM*WFQQKPGSSPKLWIYSI...
....SNLASGVP.ARFSGSG..SGTSYSLTISSVKAEDAATYYCQQWSSSPP
>AJ231212|IGKV4-78*01|Mus musculus_C57BL/6|F|V-REGION|400..691|292 nt|1| | | |97 AA|97+15=112| | |
QIVLTQSPAIMSASPGEKVTMTCSARSSVS.....SSYLYWYQQKPGSSPKLWIYST...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAATFYCQQYSGYPS
>AJ231214|IGKV4-79*01|Mus musculus_C57BL/6|F|V-REGION|401..692|292 nt|1| | | |97 AA|97+15=112| | |
QIVLTQSPAIMSASPGEKVTLTCSASSSVS.....SSYLYWYQQKPGSSPKLWIYST...
....SNLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAASYFCHQWSSYPP
>AJ231213|IGKV4-80*01|Mus musculus_C57BL/6|F|V-REGION|450..731|282 nt|1| | | |94 AA|94+17=111| | |
QIVLTQSPAIMSASLGEEITLTCSASSSV.......SYMHWYQQKSGTSPKLLIYST...
....SNLASGVP.SRFSGSG..SGTFYSLTISSVEAEDAADYYCHQWSSYP
>AJ231215|IGKV4-81*01|Mus musculus_C57BL/6|F|V-REGION|376..667|292 nt|1| | | |97 AA|97+15=112| | |
ENVLTQSPAIMAASPGEKVTMTCSASSSVS.....SSNLHWYQQKSGTSTKFWIYRT...
....SNLASEVP.APFSGSG..SGTSYSLTISSVEAEDAATYYCQQWSGYPP
>AJ231230|IGKV4-83*01|Mus musculus_C57BL/6|P|V-REGION|551..836|286 nt|1| | | |95 AA|95+17=112| | |
GIVLTQSPTTMTAFPGENVTITCSASSSI.......NYIHWYQQKSGNTPKQ*IYKT...
....SDLPSGVP.TLFSGSG..SGTSYSLTISSVEAEDAATYYCQQWSGYQP
>X05555|IGKV4-86*01|Mus musculus_BALB/c|F|V-REGION|5..287|283 nt|1| | | |94 AA|94+17=111| | |
EIVLTQSPAITAASLGQKVTITCSASSSV.......SYMHWYQQKSGTSPKPWIYEI...
....SKLASGVP.ARFSGSG..SGTSYSLTISSMEAEDAAIYYCQQWNYPL
>AJ231224|IGKV4-90*01|Mus musculus_C57BL/6|F|V-REGION|360..644|285 nt|1| | | |95 AA|95+17=112| | |
EILLTQSPAIIAASPGEKVTITCSASSSV.......SYMNWYQQKPGSSPKIWIYGI...
....SNLASGVP.ARFSGSG..SGTSFSFTINSMEAEDVATYYCQQRSSYPP
>AJ231229|IGKV4-91*01|Mus musculus_C57BL/6|F|V-REGION|394..685|292 nt|1| | | |97 AA|97+15=112| | |
EIVLTQSPTTMAASPGEKITITCSASSSIS.....SNYLHWYQQKPGFSPKLLIYRT...
....SNLASGVP.ARFSGSG..SGTSYSLTIGTMEAEDVATYYCQQGSSIPR
>AJ231226|IGKV4-92*01|Mus musculus_C57BL/6|F|V-REGION|403..693|291 nt|1| | | |97 AA|97+15=112| | |
EMVLTQSPVSITASRGEKVTITCRASSSIS.....SNYLHWYQQKPGSSPKLLIYRT...
....SILASGVL.DSFSGSG..SESSYTLTISCMQDEVAATYYCQQGSSSPP
>AJ235963|IGKV5-37*01|Mus musculus_C57BL/6|F|V-REGION|524..810|287 nt|1| | | |95 AA|95+16=111| | |
DILLTQSPATLSVTPGETVSLSCRASQSI......YKNLHWYQQKSHRSPRLLIKYA...
....SDSISGIP.SRFTGSG..SGTDYTLSINSVKPEDEGIYYCLQGYSTP
>AJ235964|IGKV5-39*01|Mus musculus_C57BL/6|F|V-REGION|403..689|287 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSPATLSVTPGDRVSLSCRASQSI......SDYLHWYQQKSHESPRLLIKYA...
....SQSISGIP.SRFSGSG..SGSDFTLSINSVEPEDVGVYYCQNGHSFP
>AJ235973|IGKV5-43*01|Mus musculus_C3H|F|V-REGION|526..812|287 nt|1| | | |95 AA|95+16=111| | |
DIVLTQSPATLSVTPGDSVSLSCRASQSI......SNNLHWYQQKSHESPRLLIKYA...
....SQSISGIP.SRFSGSG..SGTDFTLSINSVETEDFGMYFCQQSNSWP
>AJ235974|IGKV5-45*01|Mus musculus_C57BL/6|F|V-REGION|599..885|287 nt|1| | | |95 AA|95+16=111| | |
DIVLTQSPATLSVTPGDRVSLSCRASQSI......SNYLHWYQQKSHESPRLLIKYA...
....SQSISGIP.SRFSGSG..SGTDFTLSINSVETEDFGMYFCQQSNSWP
>V01564|IGKV5-48*01|Mus musculus_BALB/c|F|V-REGION|320..606|287 nt|1| | | |95 AA|95+16=111| | |
DILLTQSPAILSVSPGERVSFSCRASQSI......GTSIHWYQQRTNGSPRLLIKYA...
....SESISGIP.SRFSGSG..SGTDFTLSINSVESEDIADYYCQQSNSWP
>J00569|IGKV6-13*01|Mus musculus|F|V-REGION|432..716|285 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSQKFMSTSVGDRVSITCKASQNV......GTAVAWYQQKPGQSPKLLIYSA...
....SNRYTGVP.DRFTGSG..SGTDFTLTISNMQSEDLADYFCQQYSSYP
>Y15975|IGKV6-14*01|Mus musculus_C57BL/6|F|V-REGION|353..639|287 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSQKFMSTSVGDRVSITCKASQNV......RTAVAWYQQKPGQSPKALIYLA...
....SNRHTGVP.DRFTGSG..SGTDFTLTISNVQSEDLADYFCLQHWNYP
>Y15976|IGKV6-15*01|Mus musculus_C57BL/6|F|V-REGION|349..635|287 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSQKFMSTSVGDRVSVTCKASQNV......GTNVAWYQQKPGQSPKALIYSA...
....SYRYSGVP.DRFTGSG..SGTDFTLTISNVQSEDLAEYFCQQYNSYP
>IMGT000068|IGKV6-15-1*01|Mus musculus_C57BL/6J|P|V-REGION|2860075..2860278|204 nt|1| | | |68 AA|68+16=84|partial in 3'|rev-compl|
DIVMTQSQKFMSTSVGERVIITCKASQNV......GTAVAWYQQKPGQSPKALIYWA...
....SNWYTGVP.DRFTGSG..SG
>Y15978|IGKV6-17*01|Mus musculus_C57BL/6|F|V-REGION|365..651|287 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSHKFMSTSVGDRVSITCKASQDV......STAVAWYQQKPGQSPKLLIYSA...
....SYRYTGVP.DRFTGSG..SGTDFTFTISSVQAEDLAVYYCQQHYSTP
>Y15981|IGKV6-20*01|Mus musculus_C57BL/6|F|V-REGION|473..759|287 nt|1| | | |95 AA|95+16=111| | |
NIVMTQSPKSMSMSVGERVTLSCKASENV......GTYVSWYQQKPEQSPKLLIYGA...
....SNRYTGVP.DRFTGSG..SATDFTLTISSVQAEDLADYHCGQSYSYP
>AJ235961|IGKV6-23*01|Mus musculus_C57BL/6|F|V-REGION|645..931|287 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSHKFMSTSVGDRVSITCKASQDV......GTAVAWYQQKPGQSPKLLIYWA...
....STRHTGVP.DRFTGSG..SGTDFTLTISNVQSEDLADYFCQQYSSYP
>AJ235962|IGKV6-25*01|Mus musculus_C57BL/6|F|V-REGION|359..645|287 nt|1| | | |95 AA|95+16=111| | |
DIVMTQSHKFMSTSVGDRVSITCKASQDV......STAVAWYQQKPGQSPKLLIYWA...
....STRHTGVP.DRFTGSG..SGTDYTLTISSVQAEDLALYYCQQHYSTP
>AJ235967|IGKV6-29*01|Mus musculus_C57BL/6|F|V-REGION|449..735|287 nt|1| | | |95 AA|95+16=111| | |
NIVMTQSPKSMSMSVGERVTLSCKASENV......GTYVSWYQQKPEQSPKLLIYGA...
....SNRYPGVP.DRFTGSG..SATDFTLTISSLQAEDLADYHCGQGYSYL
>AJ235968|IGKV6-32*01|Mus musculus_C57BL/6|F|V-REGION|389..675|287 nt|1| | | |95 AA|95+16=111| | |
SIVMTQTPKFLLVSAGDRVTITCKASQSV......SNDVAWYQQKPGQSPKLLIYYA...
....SNRYTGVP.DRFTGSG..YGTDFTFTISTVQAEDLAVYFCQQDYSSP
>M14360|IGKV6-32*02|Mus musculus_BALB/c|F|V-REGION|1190..1476|287 nt|1| | | |95 AA|95+16=111| | |
SIVMTQTPKFLLVSAGERVTITCKASQSV......SNDVAWYQQKPGQSPKLLIYYA...
....SNRYTGVP.DRFTGSG..YGTDFTFTISTVQAEDLAVYFCQQDYSSP
>M14361|IGKV6-b*01|Mus musculus_C58|F|V-REGION|1346..1632|287 nt|1| | | |95 AA|95+16=111| | |
SIVMTQTPKFLPVSAGDRVTMTCKASQSV......GNNVAWYQQKPGQSPKLLIYYA...
....SNRYTGVP.DRFTGSG..SGTDFTFTISSVQVEDLAVYFCQQHYSSP
>M24937|IGKV6-c*01|Mus musculus_SK|F|V-REGION|509..795|287 nt|1| | | |95 AA|95+16=111| | |
SIVMTQTPKFLPVTAEDRVTITCKASQSV......SNEVAWYQQKPGQSPKLLIYYA...
....SNRYTGVP.DRFTGSG..SGTDFTFTISSVQVEDLAVYFCQQHYSSP
>L36249|IGKV6-d*01|Mus musculus_PERA|F|V-REGION|509..795|287 nt|1| | | |95 AA|95+16=111| | |
SIVMTQSPKSLPVSAGDRVTMTCKASQSV......SNDVAWYQQKPGQSPKLLIYYA...
....SNRYTGVP.ERFTGSG..SGTDFTFTISGVQAEDLAVYFCQQHYTTP
>AF044198|IGKV7-33*01|Mus musculus_BALB/c|F|V-REGION|332..636|305 nt|1| | | |101 AA|101+10=111| | |
DIVMTQSPTFLAVTASKKVTISCTASESLYSSKHKVHYLAWYQKKPEQSPKLLIYGA...
....SNRYIGVP.DRFTGSG..SGTDFTLTISSVQVEDLTHYYCAQFYSYP
>Y15977|IGKV8-16*01|Mus musculus_C57BL/6|F|V-REGION|395..699|305 nt|1| | | |101 AA|101+10=111| | |
EIVLTQSIPSLTVSAGERVTISCKSNQNLLWSGNQRYCLVWHQWKPGQTPTPLITWT...
....SDRYSGVP.DRFIGSG..SVTDFTLTISSVQAEDVAVYFCQQHLHIP
>Y15979|IGKV8-18*01|Mus musculus_C57BL/6|ORF|V-REGION|378..687|310 nt|1| | | |103 AA|103+10=113| | |
DIVMTQSPSSLAVTAGEKVTMRCKSSQSLLWSVNQNNYLSWYQQKQGQPPKLLIYGA...
....SIRESWVP.DRFTGSG..SGTDFTLTISNVHAEDLAVYYCQHNHGSFLP
>Y15980|IGKV8-19*01|Mus musculus_C57BL/6|F|V-REGION|377..681|305 nt|1| | | |101 AA|101+10=111| | |
DIVMTQSPSSLTVTAGEKVTMSCKSSQSLLNSGNQKNYLTWYQQKPGQPPKLLIYWA...
....STRESGVP.DRFTGSG..SGTDFTLTISSVQAEDLAVYYCQNDYSYP
>Y15982|IGKV8-21*01|Mus musculus_C57BL/6|F|V-REGION|352..653|302 nt|1| | | |100 AA|100+10=110| | |
DIVMSQSPSSLAVSAGEKVTMSCKSSQSLLNSRTRKNYLAWYQQKPGQSPKLLIYWA...
....STRESGVP.DRFTGSG..SGTDFTLTISSVQAEDLAVYYCKQSYNL
>AC155332|IGKV8-23-1*01|Mus musculus_C57BL/6J|ORF|V-REGION|176913..177196|284 nt|1| | | |94 AA|94+17=111| | |
EIV.......LTVSAGERVTISCKSNQNLLWSGNQRYCLVWHQWKPGQTPTPLITWT...
....SDRYSGVP.DRFIGSG..SVTDFTLTISSVQAEDVAVYFCQQHLHIP
>AJ235944|IGKV8-24*01|Mus musculus_C57BL/6|F|V-REGION|430..734|305 nt|1| | | |101 AA|101+10=111| | |
DIVMTQSPSSLAMSVGQKVTMSCKSSQSLLNSSNQKNYLAWYQQKPGQSPKLLVYFA...
....STRESGVP.DRFIGSG..SGTDFTLTISSVQAEDLADYFCQQHYSTP
>AJ235945|IGKV8-26*01|Mus musculus_C57BL/6|ORF|V-REGION|416..726|311 nt|1| | | |103 AA|103+10=113| | |
DIVMTQSPSSLAVTAGEKVTMSCKSSQSLLWSVNQKNYLSWYQQKQRQPPKLLIYGA...
....SIRESWVP.DRFTGSG..SGTDFTLTISNVHAEDLAVYYCQHNHGSFLP
>AJ235946|IGKV8-27*01|Mus musculus_C57BL/6|F|V-REGION|332..631|300 nt|1| | | |100 AA|100+10=110| | |
NIMMTQSPSSLAVSAGEKVTMSCKSSQSVLYSSNQKNYLAWYQQKPGQSPKLLIYWA...
....STRESGVP.DRFTGSG..SGTDFTLTISSVQAEDLAVYYCHQYLSS
>AJ235947|IGKV8-28*01|Mus musculus_C57BL/6|F|V-REGION|371..675|305 nt|1| | | |101 AA|101+10=111| | |
DIVMTQSPSSLSVSAGEKVTMSCKSSQSLLNSGNQKNYLAWYQQKPGQPPKLLIYGA...
....STRESGVP.DRFTGSG..SGTDFTLTISSVQAEDLAVYYCQNDHSYP
>L17135|IGKV8-28*02|Mus musculus_MRL|F|V-REGION|1..302|302 nt|1| | | |100 AA|100+10=110| | |
DIVMTQSPSSLSVSAGDKVTMSCKSSQSLLNSRNQKNYLAWYQQKPWQPPKLLIYGA...
....STRESGVP.DRFTGSG..SGTDFTLTISSVQAEDLAVYYCQNDYSY
>AJ235948|IGKV8-30*01|Mus musculus_C3H|F|V-REGION|441..745|305 nt|1| | | |101 AA|101+10=111| | |
DIVMSQSPSSLAVSVGEKVTMSCKSSQSLLYSSNQKNYLAWYQQKPGQSPKLLIYWA...
....STRESGVP.DRFTGSG..SGTDFTLTISSVKAEDLAVYYCQQYYSYP
>AJ235958|IGKV8-34*01|Mus musculus_129/Sv|F|V-REGION|419..723|305 nt|1| | | |101 AA|101+10=111| | |
DILMTQSPSSLTVSAGEKVTMSCKSSQSLLASGNQNNYLAWHQQKPGRSPKMLIIWA...
....STRVSGVP.DRFIGSG..SGTDFTLTINSVQAEDLAVYYCQQSYSAP
>AJ231236|IGKV9-119*01|Mus musculus_C57BL/6|P|V-REGION|505..791|287 nt|1| | | |95 AA|95+16=111| | |
DIRMTQSPSTMFASLGDRVSLT*RASQSI......RVYLSWYHQKPVGTIKLLIYST...
....FKLDSGVP.SRFSGSG..SGSDYSITINSLESEDVAIYYCLQYASSP
>V00804|IGKV9-120*01|Mus musculus_C57BL/6|F|V-REGION|314..598|285 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSASLGERVSLTCRASQDI......GSSLNWLQQEPDGTIKRLIYAT...
....SSLDSGVP.KRFSGSR..SGSDYSLTISSLESEDFVDYYCLQYASSP
>IMGT000068|IGKV9-120*02|Mus musculus_C57BL/6J|F|V-REGION|504535..504821|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSASLGERVSLTCRASQDI......GSSLNWLQQEPDGTIKRLIYAT...
....SSLDSGVP.KRFSGSR..SGSDYSLTISSLESEDFVDYYCLQYASSP
>AF003295|IGKV9-123*01|Mus musculus|F|V-REGION|275..559|285 nt|1| | | |95 AA|95+16=111| | |
DIQMIQSPSSMFASLGDRVSLSCRASQGI......RGNLDWYQQKPGGTIKLLIYST...
....SNLNSGVP.SRFSGSG..SGSDYSLTISSLESEDFADYYCLQRNAYP
>AF003294|IGKV9-124*01|Mus musculus|F|V-REGION|279..563|285 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSASLGERVSLTCRASQEI......SGYLSWLQQKPDGTIKRLIYAA...
....STLDSGVP.KRFSGSR..SGSDYSLTISSLESEDFADYYCLQYASYP
>AJ231245|IGKV9-128*01|Mus musculus_C3H|P|V-REGION|289..575|287 nt|1| | | |95 AA|95+16=111| | |
DIWMTQSPSSMFGSLGDRVSLTCQASQSI......RVYLSWYQQKQVGTIKLLIYST...
....SKLDSGVP.SRFSGSG..S*SDYSLTINSLESEDVAIYYCLQHASSP
>K00880|IGKV9-129*01|Mus musculus_BALB/c|ORF|V-REGION|325..611|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQSPSSLSASLGERVSLTCRASQDI......HGYLNLFQQKPGETIKHLIYET...
....SNLDSGVP.KRFSGSR..SGSDYSLIIGSLESEDFADYYCLQYASSP
>AF441461|IGKV10-94*01|Mus spretus_SPRET/Ei|F|V-REGION|163..449|287 nt|1| | | |95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASEDI......SNYLNWYQQKPDGTVKLLIYYA...
....SSLHSGVP.SRFSGSG..SGTDYSLTISNLEPEDIATYYCQQYSKLP
>AF441451|IGKV10-96*01|Mus spretus|F|V-REGION|146..432|287 nt|1| | ||95 AA|95+16=111| | |
DIQMTQTTSSLSASLGDRVTISCRASQDI......SNYLNWYQQKPDGTVKLLIYYT...
....SRLHSGVP.SRFSGSG..SGTDYSLTISNLEQEDIATYFCQQGSTLP

</pre>
<hr />





<a href="/genedb/doc">-> IMGT/GENE-DB Documentation</a><br />
<a href="/genedb/">-> IMGT/GENE-DB Query page</a><br />
<a href="/genedb/directlinks">-> IMGT/GENE-DB direct links</a><br />
<a href="/genedb/stats">-> IMGT/GENE-DB statistics</a><br />
<a href="/IMGTgenedbdoc/programversions.html">-> IMGT/GENE-DB program versions</a><br />
<a href="/IMGTgenedbdoc/dataupdates.html">-> IMGT/GENE-DB data updates</a>

<hr />

<p class="menu">
<a href="/">IMGT Home page</a>
| <a href="/IMGTrepertoire/">IMGT Repertoire (IG and TR)</a>
| <a href="/IMGTrepertoireMH/">IMGT Repertoire (MH)</a>
| <a href="/IMGTrepertoireRPI/">IMGT Repertoire (RPI)</a>
| <a href="/IMGTindex/">IMGT Index</a>
| <a href="/IMGTScientificChart/">IMGT Scientific chart</a>
| <a href="/IMGTeducation/">IMGT Education</a>
</p>



<span class="logosFoot">
<a href="https://www.cnrs.fr"><img src="/images/cnrs.png" alt="logo CNRS" style="vertical-align:middle;" /></a>
<a href="https://www.umontpellier.fr/"><img src="/images/logos/universite_mtp.png" alt="Universit&#233; de Montpellier" style="width:60px;vertical-align:middle;" /></a>
<a href="https://ec.europa.eu/info/research-and-innovation_en"><img src="/images/UE.png" alt="logo UE" style="vertical-align:middle;" /></a>
</span>



<p>&#169;  Copyright 1995-2022 IMGT&reg;, the international ImMunoGeneTics information system&reg;
			| <a href="/about/termsofuse.php">Terms of use</a> | <a href="/about/aboutus.php">About us</a> | <a href="mailto:contact-imgt@igh.cnrs.fr">Contact us</a> | <a href="/about/CitingIMGT.php">Citing IMGT</a></p>

<!-- #echo var=$LAST_MODIFIED -->

</body>
</html>

</body>
</html>