
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<link rel="stylesheet" type="text/css" href="css/GENEDBQuery.css"/>
<link rel="stylesheet" type="text/css" href="css/IndividualResult.css"/>
<link type="text/css" href="css/GENEDBQuery_old.css" rel="stylesheet" />
<link type="text/css" href="css/styleFoot.css" rel="stylesheet" />
<title>IMGT/GENE-DB Consultation</TITLE>
</head>
<body id="GENE-DB">
<table style="width:100%;">
<tr>
<td align="left" nowrap valign="middle" width="70%">
<font size="+3" color="#000099"><b>THANK YOU</b><br/>
<b>for using <a href="/genedb/doc">IMGT/GENE-DB</a></b></font>
</td>
<td class="notshow" align=right valign=top width="30%"><a href="/"><img SRC="images/logo_IMGT2.png" ALT="WELCOME !" class="mainlogo"/></a>
</td>
</tr>
<tr>
<td class="logolegendleft">IMGT&reg;, the international ImMunoGeneTics information system&reg;</td>
<td class="logolegend">http://www.imgt.org</td>
</tr>
</table>
<hr/>
<img src="images/logoGene-DB.png" ALT="Gene-DB logo" class="mainlogo" />

<table class="citation">
<tr>
<td class="citation">
<b>Citing IMGT/GENE-DB</b>: Giudicelli V., Chaume D. and Lefranc M.-P.. Nucleic Acids Res., 33: D256 - D261 (2005).
PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/15608191/">15608191</a>
<a href="/PDF/NAR/33_D256-D261_2005.pdf"><img src="/images/pdficon.png" alt="pdf" /></a>
</td>
</tr>
</table>
<p class="programversion">IMGT/GENE-DB program version: <a href="/IMGTgenedbdoc/programversions.html">3.1.35</a> (3 March 2022)<br/>
<a href="/IMGTgenedbdoc/dataupdates.html">IMGT/GENE-DB data updates </a>
</p>

<h2 class="subtitle"><a href="/IMGTScientificChart/SequenceDescription/IMGTreferencesequences.html">IMGT/GENE-DB reference sequences</a> in FASTA format:</h2>
<h2 class="subtitle">Amino acids  sequences with gaps according to the <a href="/IMGTScientificChart/Numbering/IMGTnumberingCR.html">IMGT unique numbering</a><br /> for <span class="casimir">F+ORF+in-frame P with IMGT Gaps</span> alleles including orphons</h2>
<hr>
<pre>
The FASTA header contains 15 fields separated by '|':

1. IMGT/LIGM-DB accession number(s)
2. IMGT gene and allele name
3. species
4. IMGT allele functionality
5. exon(s), region name(s), or extracted label(s)
6. start and end positions in the IMGT/LIGM-DB accession number(s)
7. number of nucleotides in the IMGT/LIGM-DB accession number(s)
8. codon start, or 'NR' (not relevant) for non coding labels
9. +n: number of nucleotides (nt) added in 5' compared to the corresponding label extracted from IMGT/LIGM-DB
10. +n or -n: number of nucleotides (nt) added or removed in 3' compared to the corresponding label extracted from IMGT/LIGM-DB
11. +n, -n, and/or nS: number of added, deleted, and/or substituted nucleotides to correct sequencing errors, or 'not corrected' if non corrected sequencing errors
12. number of amino acids (AA): this field indicates that the sequence is in amino acids
13. number of characters in the sequence: nt (or AA)+IMGT gaps=total
14. partial (if it is)
15. reverse complementary (if it is)
</pre>
<hr/>
<b>Number of results = 406</b><br />
<pre>
>AC090843|IGHV1-11*01|Mus musculus_C57BL/6|F|V-REGION|114868..115161|294 nt|1| | ||98 AA|98+8=106| | |
QIQLQQSGA.ELASPGASVTLSCKASGYTF....TDHIMNWVKKRPGQGLEWIGRIYPV.
.SGETNYNQKFM.GKATFSVDRSSSTVYMVLNSLTSEDPAVYYCGR
>AC090843|IGHV1-12*01|Mus musculus_C57BL/6|F|V-REGION|118475..118768|294 nt|1| | | |98 AA|98+8=106| | |
QAYLQQSGA.ELVRPGASVKMSCKASGYTF....TSYNMHWVKQTPRQGLEWIGAIYPG.
.NGDTSYNQKFK.GKATLTVDKSSSTAYMQLSSLTSEDSAVYFCAR
>AC090843|IGHV1-13*01|Mus musculus_C57BL/6|P|V-REGION|133305..133598|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVIPGA*VKLSCKASGYNF....NDYEIQWVKQSLKQGLEWIGAIHPE.
.NGGITYNQKFK.GKATFTVDTSSNTAYMQLRSLTSEDTADYYCER
>AC090843|IGHV1-14*01|Mus musculus_C57BL/6|P|V-REGION|149197..149490|294 nt|1| | | |98 AA|98+8=106| | |
EFQLQQSGP.ELVKPGASVKMSCKASGYTF....TSYVMHWVKQKPGQGLEWIGYIYPY.
.NDGTKYNEKFK.GKATLTSDKSSSTAYMELSSLTSEDSAVYYCAR
>AC090843|IGHV1-15*01|Mus musculus_C57BL/6|F|V-REGION|159978..160271|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGASVTLSCKASGYTF....TDYEMHWVKQTPVHGLEWIGAIDPE.
.TGGTAYNQKFK.GKAILTADKSSSTAYMELRSLTSEDSAVYYCTR
>AC090843|IGHV1-16*01|Mus musculus_C57BL/6|ORF|V-REGION|168497..168790|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.KVVNAGASVKLSCKSSGYSF....SRYKMECVKQSHVKSLEWIEHINLF.
.NGITNYNGNFK.SKATLTVDISSSTAYMELSRLTSEDSEVYYCAR
>AC090843|IGHV1-17-1*01|Mus musculus_C57BL/6|P|V-REGION|174593..174886|294 nt|1| | ||98 AA|98+8=106| | |
EVHLQQSLP.KVVKAGPSVKISCKASGYSF....TGYYMHWVKQSHGKILQRIEYVNPY.
.NGGTGYNEKFK.DKATLTADKSFSTAYMQFSSLTSEDSLVYYCAR
>AC090843|IGHV1-18*01|Mus musculus_C57BL/6|F|V-REGION|185257..185550|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKIPCKASGYTF....TDYNMDWVKQSHGKSLEWIGDINPN.
.NGGTIYNQKFK.GKATLTVDKSSSTAYMELRSLTSEDTAVYYCAR
>L33958|IGHV1-18*02|Mus musculus_BALB/c|[F]|V-REGION|716..968|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQFGP.ELVKPGASVKISCKASGYTF....TDYNMDWVKQSHGKSLEWIGDINPN.
.NGGTIYNQKFK.GKATLTVDKSSSTAYMELR
>L33961|IGHV1-18*03|Mus musculus_BALB/c|[F]|V-REGION|715..966|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVLLQQSGP.ELVKPGASVKIPCKASGYTF....TDYNMDWVKQSHGKSLEWIGDINPN.
.NGGTIYNQKFK.GKATLTVDKSSSTAYMELR
>AC073565|IGHV1-19*01|Mus musculus_C57BL/6|F|V-REGION|177510..177803|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.VLVKPGASVKMSCKASGYTF....TDYYMNWVKQSHGKSLEWIGVINPY.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMELNSLTSEDSAVYYCAR
>BN000872|IGHV1-19-1*01|Mus musculus_C57BL/6|P|V-REGION|1366640..1366933|294 nt|1| | | |98 AA|98+8=106| | |
EVQLKQPGT.VVVKPGASVKISCQASGYSF....TGYYMHWVKQSHGKSL*WIGLIIPY.
.NGDTGYNQKFK.GKATLTVDKSSSTANMELCSLTSEDSTVYYRAR
>AC073565|IGHV1-20*01|Mus musculus_C57BL/6|F|V-REGION|162386..162679|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGDSVKISCKASGYSF....TGYFMNWVMQSHGKSLEWIGRINPY.
.NGDTFYNQKFK.GKATLTVDKSSSTAHMELRSLTSEDSAVYYCAR
>M17950|IGHV1-20*02|Mus musculus_BALB/c|F|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYFMNWVMQSHGKSLEWIGRINPY.
.NGDTFYNQKFK.GKATLTVDKSSSTAHMELRSLASEDSAVYYCAR
>AC073565|IGHV1-21*01|Mus musculus_C57BL/6|P|V-REGION|145750..146043|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCMASGYSF....SDYYMH*VKQSHGKSLEWIGYINPN.
.NGCTSYNQKFK.GKATLTVDTSSSTAYMELHSLTSEDSSVYYCAR
>AC073565|IGHV1-21-1*01|Mus musculus_C57BL/6J|P|V-REGION|149913..150206|294 nt|1| | ||98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGALVKLSCKVSGFTF....TD*YMHWVKQSHGKSLEWIGHVYPY.
.NGGTSYNQKFK.GKATLTVDNTSSTAYMELGSLTSEDSAVYYSAR
>AC073565|IGHV1-22*01|Mus musculus_C57BL/6|F|V-REGION|139885..140178|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKMSCKASGYTF....TDYNMHWVKQSHGKSLEWIGYINPN.
.NGGTSYNQKFK.GKATLTVNKSSSTAYMELRSLTSEDSAVYYCAR
>AC073565|IGHV1-23*01|Mus musculus_C57BL/6|ORF|V-REGION|121708..122001|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGASVKLSCKASGYTF....TDYEMHCVKQTPVHGLEWIGAIDPE.
.TCGTAYNQKFK.GKATLTADKSSSTAYMELRSLTSEDSAVYYCTR
>AC073565|IGHV1-24*01|Mus musculus_C57BL/6|ORF|V-REGION|113230..113523|294 nt|1| | | |98 AA|98+8=106| | |
EVRLQQSGP.KVVNAGASVKLSCKSSGYSF....SRYKMECVKQSHGKSLEWIEHINLF.
.KGVTNYNGKFK.SKATLTVDISSSTAYMELSRLTSEDSEVYYCAR
>AC073565|IGHV1-25*01|Mus musculus_C57BL/6|P|V-REGION|104749..105042|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYIMNLVKQSHGKSLEWI*EINPY.
.NGGTNYNQKFK.GKATLTVDTSSSTEYMELHSLTSEDSAVYYCAR
>AC073565|IGHV1-26*01|Mus musculus_C57BL/6|F|V-REGION|97786..98079|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYTF....TDYYMNWVKQSHGKSLEWIGDINPN.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMELRSLTSEDSAVYYCAR
>AC073565|IGHV1-27*01|Mus musculus_C57BL/6|P|V-REGION|81542..81835|294 nt|1| | | |98 AA|98+8=106| | |
KVKL*QSGP.ELVKSGASEKISCKASGYSF....TGYYMHWVKQSHGKIIE*IGLIIPY.
.NGDTGYNQKFK.GKATLTVD*SFSTAYMELHSLKS*DSVVYFCAR
>AC073565|IGHV1-28*01|Mus musculus_C57BL/6|P|V-REGION|76737..77030|294 nt|1| | | |98 AA|98+8=106| | |
EIQLQQSGP.ELVKPGASVKISFKASGYSF....TGYYMNWMK*SHGNSLEWIGYIDPY.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMHLNNLTSEDSAVYYFAR
>AC073565|IGHV1-31*01|Mus musculus_C57BL/6|F|V-REGION|56894..57187|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYYMHWVKQSHGNILDWIGYIYPY.
.NGVSSYNQKFK.GKATLTVDKSSSTAYMELRSLTSEDSAVYYCAR
>AC073565|IGHV1-32*01|Mus musculus_C57BL/6|P|V-REGION|50935..51228|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISSKTSG*TF....TGYYMHWVKQSHGKSLEWI*YIDPY.
.DGVTSYNKKFK.RKATLTVDKSSSTAYMEL*NLTSEETEVYYCVR
>AC073565|IGHV1-34*01|Mus musculus_C57BL/6|F|V-REGION|34968..35261|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKMSCKASGYTF....TDYYMHWVKQSHGKSLEWIGYIYPN.
.NGGNGYNQKFK.GKATLTVDKSSSTAYMELRSLTSEDSAVYYCAR
>L17134|IGHV1-34*02|Mus musculus_MRL/lpr|F|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGDSVKMSCKASGYTF....TDYYMDWVKQSHGKSLEWIGYIYPN.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMELHSLTSEDSAVYYCAR
>AC073565|IGHV1-35*01|Mus musculus_C57BL/6|P|V-REGION|10854..11147|296 nt|1| | | |98 AA|98+8=106| | |
EVQLKQPGT.VVVKPGASVKISCQASGYSF....TGYYMHWVKQSHEKSL*WILLIIPY.
.NGDTSNNQKFK.GKATLTVDKSSSTANMELCSLTSEDSTVYYRAR
>D13202|IGHV1-35*02|Mus musculus_BALB/c|P|V-REGION|101..394|294 nt|1| | ||98 AA|98+8=106| | |
EVQLKQPGT.VVVKPGASVKISCQASGYSF....TGYYMHWVKQSHEKSL*WIGLIIPY.
.NGDTSNNQKFK.GKATLTVDKSSSTANMELCSLTSEDSTVYYRAR
>M17696|IGHV1-35*03|Mus musculus_BALB/c|P|V-REGION|101..354|254 nt|1| | ||84 AA|84+8=92|partial in 3'| |
EVQLKQPGT.VVVKPGASVKISCQASGYSF....TGYYMHWVKQSHEKSL*WIGLIIPY.
.NGNTSNNQKFK.GKATLTVDKSSSTANMELC
>AC073565|IGHV1-36*01|Mus musculus_C57BL/6|F|V-REGION|6270..6563|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.VLVKPGPSVKISCKASGFTF....TDYYMHWVKQSHGKSLEWIGLVYPY.
.NGGTSYNQKFK.GKATLTVDTSSSTAYMELNSLTSEDSAVYYCAR
>AC079181|IGHV1-37*01|Mus musculus_C57BL/6|F|V-REGION|171648..171941|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYFMNWVKQSHGKSLEWIGRINPY.
.NGDTFYNQKFK.GKATLTVDKSSSTAHMELLSLTSEDFAVYYCAR
>AC079181|IGHV1-39*01|Mus musculus_C57BL/6|F|V-REGION|153287..153580|294 nt|1| | | |98 AA|98+8=106| | |
EFQLQQSGP.ELVKPGASVKISCKASGYSF....TDYNMNWVKQSNGKSLEWIGVINPN.
.YGTTSYNQKFK.GKATLTVDQSSSTAYMQLNSLTSEDSAVYYCAR
>AC073561|IGHV1-4*01|Mus musculus_C57BL/6|F|V-REGION|82648..82941|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELARPGASVKMSCKASGYTF....TSYTMHWVKQRPGQGLEWIGYINPS.
.SGYTKYNQKFK.DKATLTADKSSSTAYMQLSSLTSEDSAVYYCAR
>X02459|IGHV1-4*02|Mus musculus_BALB/c|F|V-REGION|101..394|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSAA.ELARPGASVKMSCKASGYTF....TSYTMHWVKQRPGQGLEWIGYINPS.
.SGYTEYNQKFK.DKTTLTADKSSSTAYMQLSSLTSEDSAVYYCAR
>AC079181|IGHV1-42*01|Mus musculus_C57BL/6|F|V-REGION|130773..131066|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYYMNWVKQSPEKSLEWIGEINPS.
.TGGTTYNQKFK.AKATLTVDKSSSTAYMQLKSLTSEDSAVYYCAR
>L33945|IGHV1-42*02|Mus musculus_C57BL/6|[F]|V-REGION|721..973|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVNPGASVKISCKASGYSF....TGYYMNWVKQSPGKSLEWIGEINPS.
.TGGTTYNQKFK.AKATLTVDKSSSTAYMQLK
>L33948|IGHV1-42*03|Mus musculus_C57BL/6|[P]|V-REGION|714..966|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYYMNWVKQSPEKSLEWIGEINPS.
.TGGTTYNQKFK.AKATLTVDKSSSTAYI*LK
>AC079181|IGHV1-43*01|Mus musculus_C57BL/6|F|V-REGION|121936..122229|294 nt|1| | | |98 AA|98+8=106| | |
EVKLQQSGP.ELVKPGASVKISCKASGYSF....TGYYMHWVKQSSEKSLEWIGEINPS.
.TGGTSYNQKFK.GKATLTVDKSSSTAYMQLKSLTSEDSAVYYCAR
>AC079181|IGHV1-46*01|Mus musculus_C57BL/6|P|V-REGION|90193..90486|294 nt|1| | | |98 AA|98+8=106| | |
QVQVQLSAA.ELVKPGSPVKLSCKASGYTV....NDNYMEQVKQRPGQSMEWIG*IHFV.
.YGGT*YNEKF*.GKSTLTVEKSSNTAYMELNSSTSEDSVVYYCAW
>AC079181|IGHV1-47*01|Mus musculus_C57BL/6|F|V-REGION|76778..77071|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVKPGASVKMSCKASGYTF....TTYPIEWMKQNHGKSLEWIGNFHPY.
.NDDTKYNEKFK.GKATLTVEKSSSTVYLELSRLTSDDSAVYYCAR
>AC079181|IGHV1-48*01|Mus musculus_C57BL/6|P|V-REGION|29659..29952|294 nt|1| | | |98 AA|98+8=106| | |
*VQWQESGT.ELVRSGASVMMSCKASGYTF....SNYTMHWVKQSHGKGHERIGYIDNY.
.YCSTDYSEKFK.IKATLTVNKSCRTAYVKLSRLTSEDSAVYYCVR
>AC079181|IGHV1-49*01|Mus musculus_C57BL/6|F|V-REGION|12663..12956|294 nt|1| | | |98 AA|98+8=106| | |
QRELQQSGA.ELVRPGSSVKLSCKDSYFAF....MASAMHWVKQRPGHGLEWIGSFTMY.
.SDATEYSENFK.GKATLTANTSSSTAYMELSSLTSEDSAVYYCAR
>AC090843|IGHV1-5*01|Mus musculus_C57BL/6|F|V-REGION|15954..16247|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGT.VLARPGASVKMSCKTSGYTF....TSYWMHWVKQRPGQGLEWIGAIYPG.
.NSDTSYNQKFK.GKAKLTAVTSASTAYMELSSLTNEDSAVYYCTR
>AC074329|IGHV1-50*01|Mus musculus_C57BL/6|F|V-REGION|87257..87550|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMQWVKQRPGQGLEWIGEIDPS.
.DSYTNYNQKFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYYCAR
>AC074329|IGHV1-51*01|Mus musculus_C57BL/6|P|V-REGION|98878..99177|300 nt|1| | | |100 AA|100+8=108| | |
QVQLPQSGS.EVGRTGASVKMFCKAPGYTF....TNYYMY*LKQSHGDSLEWI*YIYPG.
.NGLTSYAKKFK.GKATLTIDNSASTAYMQLSSMTSEASDDYCCARQV
>AC074329|IGHV1-52*01|Mus musculus_C57BL/6|F|V-REGION|112996..113289|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVRPGSSVKLSCKASGYTF....TSYWMHWVKQRPIQGLEWIGNIDPS.
.DSETHYNQKFK.DKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>AC074329|IGHV1-53*01|Mus musculus_C57BL/6|F|V-REGION|125912..126205|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGT.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGNINPS.
.NGGTNYNEKFK.SKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>L27628|IGHV1-53*02|Mus musculus_C57BL/6|[F]|V-REGION|708..959|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGT.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPAQGLEWIGNINPS.
.NGGTNYNEKFK.SKATLTVDKSSSTAYMQLS
>L26932|IGHV1-53*03|Mus musculus_C57BL/6|[F]|V-REGION|709..960|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGT.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGNINPS.
.NGGTNYNEKFK.SKATLTVDKSSSSAYMQLI
>L26927|IGHV1-53*04|Mus musculus_C57BL/6|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSSKASGYTF....TSYWMHWVKQRPGQGLEWIGNINPS.
.NGGTNYNEKFK.SKATLTVDKSSSTAYMQLS
>AC074329|IGHV1-54*01|Mus musculus_C57BL/6|F|V-REGION|161184..161477|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKVSCKASGYAF....TNYLIEWVKQRPGQGLEWIGVINPG.
.SGGTNYNEKFK.GKATLTADKSSSTAYMQLSSLTSEDSAVYFCAR
>X02064|IGHV1-54*02|Mus musculus|F|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKVSCKASGYAF....TNYLIEWVKQRPGQGLEWIGVINPG.
.SGGTNYNEKFK.GKATLTADKSSNTAYMQLSSLTSEDSAVYFCAR
>AF304553|IGHV1-54*03|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKVSCKASGYAF....TNYLIEWVKQRPGQGLEWIGVINPG.
.SGGTNYNEKFK.GKATLTADKSSSTAYMQLSSLTSDDSAVYFCAR
>AC074329|IGHV1-55*01|Mus musculus_C57BL/6|F|V-REGION|175604..175897|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWITWVKQRPGQGLEWIGDIYPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLSSLTSEDSAVYYCAR
>L26877|IGHV1-55*02|Mus musculus_BALB/c|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWMTWVKQRPGQGLEWIGDIYPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLS
>L26952|IGHV1-55*03|Mus musculus_C57BL/6|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWITWVKQRPGQGLEWIGDIYPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLI
>L26930|IGHV1-55*04|Mus musculus_C57BL/6|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWITWVKQRPGQGLEWIGDTHPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLS
>AC087166|IGHV1-56*01|Mus musculus_C57BL/6|F|V-REGION|138735..139028|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGASVKISCKAPGYTF....TSHWMQWVRQRPGQGLEWIGEIFPG.
.SGSTYYNEKFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYFCAR
>AF304551|IGHV1-56*02|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGASVKISCKAPGYTF....TSHWMQWVRQRPGQGLEWIGEIFPG.
.SGSTYYNEKFK.GKATLTLDTSSSTAYMQLSSLTSEDSAVYFCAR
>AC087166|IGHV1-58*01|Mus musculus_C57BL/6|F|V-REGION|69371..69664|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRPGSSVKMSCKTSGYTF....TSYGINWVKQRPGQGLEWIGYIYIG.
.NGYTEYNEKFK.GKATLTSDTSSSTAYMQLSSLTSEDSAIYFCAR
>M34978|IGHV1-58*02|Mus musculus_A/J|P|V-REGION|260..553|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRPGSSVKLSCKTSGYTF....TSYGINWVKQRPGQGLEWIGYIYIG.
.NGNTEYNEKFK.SKATLTSDTSSSTAYMELSSLTSEDSAIYFCAR
>AC087166|IGHV1-59*01|Mus musculus_C57BL/6|F|V-REGION|46455..46748|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVRPGTSVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGVIDPS.
.DSYTNYNQKFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYYCAR
>AC087166|IGHV1-60*01|Mus musculus_C57BL/6|P|V-REGION|37000..37299|300 nt|1| | ||100 AA|100+8=108| | |
QVQLPQSGS.EVGRTGASVKMFCKASGYTF....TNYYMH*LKQSHGDSLEWI*YIYPG.
.NGLTSYAKKFK.GKATLTIDNSASTAYMQLSSMTSEASDDYYCARQV
>AC087166|IGHV1-61*01|Mus musculus_C57BL/6|F|V-REGION|22397..22690|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVRPGSSVKLSCKASGYTF....TSYWMDWVKQRPGQGLEWIGNIYPS.
.DSETHYNQKFK.DKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>BN000872|IGHV1-62-1*01|Mus musculus_C57BL/6|F|V-REGION|684090..684381|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGASVKLSCKASGYTF....TSYWMQWVKQRPGQGLEWIGEIFPG.
.SGSTYYNEKFK.GKATLTVDTSSSTAYMQLSSLTAENSAIYLCKE
>BN000872|IGHV1-62-2*01|Mus musculus_C57BL/6|F|V-REGION|624361..624662|302 nt|1| | | |100 AA|100+8=108| | |
QVQLQQSGA.ELVKPGASVKLSCKASGYTF....TEYTIHWVKQRSGQGLEWIGWFYPG.
.SGSIKYNEKFK.DKATLTADKSSSTVYMELSRLTSEDSAVYFCARHE
>J00533|IGHV1-62-3*01|Mus musculus_C57BL/6|ORF|V-REGION|206..499|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSSKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLSSLTSEDSAVYYCAR
>L26929|IGHV1-62-3*02|Mus musculus_C57BL/6|[F]|V-REGION|715..966|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSSKASGYTF....TSYWMHWVKQRPGQGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLI
>AC073939|IGHV1-63*01|Mus musculus_C57BL/6|F|V-REGION|165321..165614|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKMSCKASGYTF....TNYWIGWAKQRPGHGLEWIGDIYPG.
.GGYTNYNEKFK.GKATLTADKSSSTAYMQFSSLTSEDSAIYYCAR
>D14633|IGHV1-63*02|Mus musculus_MRL/lpr|F|V-REGION|347..640|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKMSCKAAGYTF....TNYWIGWVKQRPGHGLEWIGDIYPG.
.GGYTNYNEKFK.GKATLTADTSSSTAYMQLSSLTSEDSAIYYCAR
>AC073939|IGHV1-64*01|Mus musculus_C57BL/6|F|V-REGION|153417..153710|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGMIHPN.
.SGSTNYNEKFK.SKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>L26853|IGHV1-64*02|Mus musculus_C57BL/6|[F]|V-REGION|717..968|252 nt|1| | ||84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGMIHPN.
.SGSTNYNEKFK.SKATLTVDKSSSTAYMQLS
>AC073939|IGHV1-66*01|Mus musculus_C57BL/6|F|V-REGION|67852..68145|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVKPGASVKISCKASGYSF....TSYYIHWVKQRPGQGLEWIGWIYPG.
.SGNTKYNEKFK.GKATLTADTSSSTAYMQLSSLTSEDSAVYYCAR
>AC073939|IGHV1-67*01|Mus musculus_C57BL/6|P|V-REGION|57022..57315|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGVSVKISCKGSGYTF....TDYAMHWVKQSHAKSLEWIGVISTY.
.YGDASYNQKFK.DKATMTVDKSSSTAYMELARLTSEDSAVYYCAR
>AC073939|IGHV1-69*01|Mus musculus_C57BL/6|F|V-REGION|37801..38094|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVMPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGEIDPS.
.DSYTNYNQKFK.GKSTLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>X00160|IGHV1-69*02|Mus musculus_BALB/c|F|V-REGION|142..435|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGEIDPS.
.DSYTNYNQKFK.GKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>L26854|IGHV1-69*03|Mus musculus_C57BL/6|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVMPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGEIDPS.
.DSYTNYNQKFK.GKATLTVDKSSSTAYMQLS
>AC090843|IGHV1-7*01|Mus musculus_C57BL/6|F|V-REGION|41120..41413|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELAKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGYINPS.
.SGYTKYNQKFK.DKATLTADKSSSTAYMQLSSLTYEDSAVYYCAR
>AC163348|IGHV1-70*01|Mus musculus_C57BL/6|P|V-REGION|80908..81201|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGTSVKMSCKASGYTF....FTYWMNWV*QRPGQGLEWIGQIFPA.
.SGSTYYNEMYK.DKAALTVDTSSSTAYMQLSSLTSEDTAVYFCAR
>AC163348|IGHV1-71*01|Mus musculus_C57BL/6|F|V-REGION|30584..30885|302 nt|1| | | |100 AA|100+8=108| | |
QVQLQQSGA.ELVKPGASVKLSCKASGYTF....TEYTIHWVKQRSGQGLEWIGWFYPG.
.SGSIKYNEKFK.DKATLTADKSSSTVYMELSRLTSEDSAVYFCARHE
>X02066|IGHV1-71*02|Mus musculus|F|V-REGION|1..293|293 nt|1| | | |97 AA|97+11=108|partial in 5'| |
...LQQSGA.GLVKPGASVKLSCKASGYTF....TEYIIHWVKQRSGQGLEWIGWFSPG.
.SGSIKYNEKFK.DKATLTADKSSSTVYMELSRLTSEDSAVYFCARHE
>AC163348|IGHV1-72*01|Mus musculus_C57BL/6|F|V-REGION|14813..15106|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLSSLTSEDSAVYYCAR
>L26851|IGHV1-72*02|Mus musculus_C57BL/6|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGRIDPN.
.SGGTKSNEKFK.SKATLTVDKPSSTAYMQLS
>L26868|IGHV1-72*03|Mus musculus_BALB/c|[F]|V-REGION|715..966|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLS
>J00532|IGHV1-72*04|Mus musculus_C57BL/6|F|V-REGION|206..499|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDTSSSTAYMQLSSLTSEDSAVHYCAR
>L26878|IGHV1-72*05|Mus musculus_BALB/c|[F]|V-REGION|709..960|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLS
>AC160473|IGHV1-74*01|Mus musculus_C57BL/6|F|V-REGION|124052..124345|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKVSCKASGYTF....TSYWMHWVKQRPGQGLEWIGRIHPS.
.DSDTNYNQKFK.GKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAI
>L26857|IGHV1-74*02|Mus musculus_C57BL/6|[F]|V-REGION|709..960|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQPGA.ELVKPGASVKVSCKASGYTF....TSYWMHWVKQRPGQGLEWIGRIHPS.
.DSDTNYNQKFK.GKATLTVDKSSSTAYMQLS
>L26933|IGHV1-74*03|Mus musculus_C57BL/6|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKVSCKASGYTF....TSYWMHWVKQRPGQGLEWIGRIHPS.
.DSDTNYNQKFK.GKATLTVDKSSSTAYMQLI
>J00537|IGHV1-74*04|Mus musculus_C57BL/6|F|V-REGION|140..433|294 nt|1| | | |98 AA|98+8=106| | |
HVQLQQPGA.ELVKPGASVKVSCKASGYTF....TSYWMHWVKQRPGQGLEWIGRIHPS.
.DSDTNYNQKFK.GKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAI
>AC160473|IGHV1-75*01|Mus musculus_C57BL/6|F|V-REGION|155354..155647|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVKPGASVKISCKASGYTF....TDYYINWVKQRPGQGLEWIGWIFPG.
.SGSTYYNEKFK.GKATLTVDKSSSTAYMLLSSLTSEDSAVYFCAR
>AC160473|IGHV1-76*01|Mus musculus_C57BL/6|F|V-REGION|169285..169578|294 nt|1| | | |98 AA|98+8=106| | |
QVQLKQSGA.ELVRPGASVKLSCKASGYTF....TDYYINWVKQRPGQGLEWIARIYPG.
.SGNTYYNEKFK.GKATLTAEKSSSTAYMQLSSLTSEDSAVYFCAR
>AC160473|IGHV1-77*01|Mus musculus_C57BL/6|F|V-REGION|183271..183564|294 nt|1| | | |98 AA|98+8=106| | |
QVQLKQSGA.ELVKPGASVKISCKASGYTF....TDYYINWVKQRPGQGLEWIGKIGPG.
.SGSTYYNEKFK.GKATLTADKSSSTAYMQLSSLTSEDSAVYFCAR
>AC160473|IGHV1-78*01|Mus musculus_C57BL/6|F|V-REGION|190177..190470|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSDA.ELVKPGASVKISCKVSGYTF....TDHTIHWMKQRPEQGLEWIGYIYPR.
.DGSTKYNEKFK.GKATLTADKSSSTAYMQLNSLTSEDSAVYFCAR
>AC160990|IGHV1-79*01|Mus musculus_C57BL/6|P|V-REGION|162565..162858|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQ*SGA.ELVKPGASVKLSCKASGYTF....TNYIINWVKEGPGHGLEWIGWISPE.
.YGHTYYNQKFK.GKATFTADTSSSTAYMELSSLTSEDSAVYFCAR
>AC090843|IGHV1-8*01|Mus musculus_C57BL/6|P|V-REGION|58234..58527|294 nt|1| | ||98 AA|98+8=106| | |
QDQLHQSEA.ELQQPGTSVKMP*KATGYTF....TKYRMCWVRQKLGQGLEWIASVDPG.
.NSNTEYNQKFK.GKATLTEHKSSSTAYIELSNLTSEDSAVYYCTR
>AC160990|IGHV1-80*01|Mus musculus_C57BL/6|F|V-REGION|138317..138610|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVKPGASVKISCKASGYAF....SSYWMNWVKQRPGKGLEWIGQIYPG.
.DGDTNYNGKFK.GKATLTADKSSSTAYMQLSSLTSEDSAVYFCAR
>AC160990|IGHV1-81*01|Mus musculus_C57BL/6|F|V-REGION|130382..130675|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELARPGASVKLSCKASGYTF....TSYGISWVKQRTGQGLEWIGEIYPR.
.SGNTYYNEKFK.GKATLTADKSSSTAYMELRSLTSEDSAVYFCAR
>AC160990|IGHV1-82*01|Mus musculus_C57BL/6|F|V-REGION|98123..98416|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVKPGASVKISCKASGYAF....SSSWMNWVKQRPGKGLEWIGRIYPG.
.DGDTNYNGKFK.GKATLTADKSSSTAYMQLSSLTSEDSAVYFCAR
>AC160990|IGHV1-83*01|Mus musculus_C57BL/6|P|V-REGION|86844..87137|294 nt|1| | | |98 AA|98+8=106| | |
*VQLQQSGP.ELVKPGASVKMSCKASGYTF....TDYYMHWVKQKPGKGLEWIGEIYPG.
.SGNTYYNEKFK.GKATLTADTSSSTAYMQLSSLTSEDSAVYFCAR
>AC160990|IGHV1-84*01|Mus musculus_C57BL/6|F|V-REGION|69959..70252|294 nt|1| | | |98 AA|98+8=106| | |
QIQLQQSGP.ELVKPGASVKISCKASGYTF....TDYYINWVKQRPGQGLEWIGWIYPG.
.SGNTKYNEKFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYFCAR
>AF305910|IGHV1-84*02|Mus musculus_BALB/c|[F]|V-REGION|315..608|294 nt|1| | | |98 AA|98+8=106| | |
QIQLQQSGP.ELVKPGASVKISCKASGYTF....TDYYINWVKQKPGQGLEWIGWIYPG.
.SGNTKYNEKFK.GKATLTVDTSSSTAYMQLSSLTSEDTAVYFCAR
>AC160990|IGHV1-85*01|Mus musculus_C57BL/6|F|V-REGION|50633..50926|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVKPGASVKLSCKASGYTF....TSYDINWVKQRPGQGLEWIGWIYPR.
.DGSTKYNEKFK.GKATLTVDTSSSTAYMELHSLTSEDSAVYFCAR
>NW_001033180|IGHV1-87*01|Mus musculus|F|V-REGION|481488..481781|294 nt|1| | ||98 AA|98+8=106| | |
QVQLQQSGA.ELARPGASVKMSCKASGYTF....TSYWMQWVKQRPGQGLEWIGAIYPG.
.DGDTRYTQKFK.GRATLTADKSSSTAYMQLSSLTSEDSAVYYCAT
>AC090843|IGHV1-9*01|Mus musculus_C57BL/6|F|V-REGION|86194..86487|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELMKPGASVKLSCKATGYTF....TGYWIEWVKQRPGHGLEWIGEILPG.
.SGSTNYNEKFK.GKATFTADTSSNTAYMQLSSLTTEDSAIYYCAR
>AC073561|IGHV10-1*01|Mus musculus_C57BL/6|F|V-REGION|74517..74818|302 nt|1| | | |100 AA|100+6=106| | |
EVQLVESGG.GLVQPKGSLKLSCAASGFSF....NTYAMNWVRQAPGKGLEWVARIRSKS
NNYATYYADSVK.DRFTISRDDSESMLYLQMNNLKTEDTAMYYCVR
>AF064445|IGHV10-1*02|Mus musculus_BALB/c|F|V-REGION|323..624|302 nt|1| | | |100 AA|100+6=106| | |
EVQLVESGG.GLVQPKGSLKLSCAASGFTF....NTYAMNWVRQAPGKGLEWVARIRSKS
NNYATYYADSVK.DRFTISRDDSQSMLYLQMNNLKTEDTAMYYCVS
>AC073561|IGHV10-3*01|Mus musculus_C57BL/6|F|V-REGION|118953..119254|302 nt|1| | ||100 AA|100+6=106| | |
EVQLVESGG.GLVQPKGSLKLSCAASGFTF....NTYAMHWVRQAPGKGLEWVARIRSKS
SNYATYYADSVK.DRFTISRDDSQSMLYLQMNNLKTEDTAMYYCVR
>AF064442|IGHV10-3*02|Mus musculus_C57BL/10|F|V-REGION|323..624|302 nt|1| | | |100 AA|100+6=106| | |
EVQLVESGG.GLVQPKGSLKLSCAASVFTF....NTYAMHWVRQAPGKGLEWVARIRSKS
SNYATYYADSVK.DRFTISRDDSQSMLYLQMNNLKTEDTAMYYCVR
>AF064444|IGHV10-3*03|Mus musculus_BALB/c|F|V-REGION|323..624|302 nt|1| | | |100 AA|100+6=106| | |
EVQLVESGG.GLVQPKGSLKLSCAASVFTF....NTYAMHWVCQAPGKGLEWVARIRSKS
NNYATYYADSVK.DRFTISRDDSQSMLYLQMNNLKTEDTAMYYCVR
>AF064446|IGHV10S3*01|Mus musculus_BALB/c|F|V-REGION|321..622|302 nt|1| | | |100 AA|100+6=106| | |
EVQLVETGG.GLVQPKGSLKLSCAASGFTF....NTNAMNWVRQAPGKGLEWVARIRSKS
NNYATYYADSVK.DRFTISRDDSQSMLYLQMNNLKTEDTAMYYCVR
>M21470|IGHV10S4*01|Mus musculus_MRL/lpr|(F)|V-REGION|58..357|300 nt|1| | | |100 AA|100+6=106| | |
EVQLVETGG.GLVQPKGSLKLSCPASGFSF....NTNAMNWVRQAPGKGLEWVARIRSKS
NNYATYYADSVK.DRFTISRDDSQSMLYLQMNNLKTEDTAMYYCVR
>AC073563|IGHV11-1*01|Mus musculus_C57BL/6|F|V-REGION|175711..176006|296 nt|1| | | |98 AA|98+8=106| | |
EVQLLETGE.GLVPPGGSRGLSCEGSGFTF....SGFWMSWVRQTPGKTLEWIGDINSD.
.GSAINYAPSIK.DRFTIFRDNDKSTLYLQMSNVRSEDTATYFCMR
>AJ851868|IGHV11-1*02|Mus musculus_129/Sv|F|V-REGION|298363..298658|296 nt|1| | | |98 AA|98+8=106| | |
EVQLLETGG.GLVQPGGSRGLSCEGSGFTF....SGFWMSWVRQTPGKTVEWIGDINSD.
.GSAINYAPSIK.DRFTIFRDNDKSTLYLQMSNVRSEDPATYFCMR
>AC073563|IGHV11-2*01|Mus musculus_C57BL/6|F|V-REGION|109349..109644|296 nt|1| | | |98 AA|98+8=106| | |
EVQLLETGG.GLVQPGGSRGLSCEGSGFTF....SGFWMSWVRQTPGKTLEWIGDINSD.
.GSAINYAPSIK.DRFTIFRDNDKSTLYLQMSNVRSEDTATYFCMR
>AJ851868|IGHV11-2*02|Mus musculus_129/Sv|F|V-REGION|238135..238430|296 nt|1| | | |98 AA|98+8=106| | |
EVQLLETGG.GLVQPGGSRGLSCEGSGFTF....SGFWMSWVRQTPGKTLEWIGDINSD.
.GSAINYAPSIK.DRFTIFRDNDKSTLYLQMSNVRSEDTATYFCMR
>AC073563|IGHV12-1*01|Mus musculus_C57BL/6|P|V-REGION|50322..50626|305 nt|1| | | |101 AA|101+7=108| | |
KI*LKESGS.ALIKPSQPLSLTCTVSGFSIT..SSSYCWHWICQPPGKGLEWMGHICYE.
..GSLNYSPSLK.SRSTISRDTSLNKFFIQLSSLTDEDTVMYYCSREN
>AJ851868|IGHV12-1*02|Mus musculus_129/Sv|P|V-REGION|190024..190328|305 nt|1| | | |101 AA|101+7=108| | |
QIQLKESGP.AVIKPS*SLSLTCTVSGFSIT..SSGFCWHWICQPPGKGLEGMGRICYE.
..GSIYYSPSLK.SPSTISRDTSLNKFFIQLSTVPDEDTDMYYYPREN
>AJ851868|IGHV12-1-1*01|Mus musculus_129/Sv|F|V-REGION|135718..136022|305 nt|1| | | |101 AA|101+7=108| | |
QIQLKESGP.AVIKPSQSLSLTCIVSGFSIT..SSSYCWHWIRQPPGKGLEWMGRICYE.
..GSIYYSPSIK.SRSTISRDTSLNKFFIQLSSVTNEDTAMYYCSREN
>AJ851868|IGHV12-1-2*01|Mus musculus_129/Sv|P|V-REGION|94389..94693|305 nt|1| | | |101 AA|101+7=108| | |
KI*LKKSGS.ALIKPSQPLSLTCTVSGFSIT..SSSYCWHWIRQSPGKWLEWMGHICYE.
..G*LNYSPSLK.RRSTISRDTSLNKFFIQLSSVTDEDTAMYYCSREN
>AC073589|IGHV12-2*01|Mus musculus_C57BL/6|P|V-REGION|161346..161650|305 nt|1| | | |101 AA|101+7=108| | |
QIQLKESGP.AVIKPS*SLSLTCTVSGFSIT..SSGFCWHWICQPPGKGLEWMGRICYE.
..GSIYYSPSLK.SPSTISRDISLNKFFIQLSSVTDEDTAMYYYSREN
>AJ851868|IGHV12-2-1*01|Mus musculus_129/Sv|ORF|V-REGION|109773..110077|305 nt|1| | | |101 AA|101+7=108| | |
QIQLKESGP.AVIKPSQSLSLTCTVSEFSIT..SSGFCWHRICQPPGKGLEWMGRICYE.
..GSIYYSPSLK.SRSTISIDTSLNKIFIQLSSVTDEDTDMYYYSREN
>AC073590|IGHV12-3*01|Mus musculus_C57BL/6|F|V-REGION|132459..132758|300 nt|1| | ||100 AA|100+8=108| | |
QMQLQESGP.GLVKPSQSLFLTCSITGFPIT...SGYYWIWIRQSPGKPLEWMGYITHS.
..GETFYNPSLQ.SPISITRETSKNQFFLQLNSVTTEDTAMYYCAGDR
>AJ972404|IGHV12-3*02|Mus musculus_129/Sv|F|V-REGION|67535..67834|300 nt|1| | | |100 AA|100+8=108| | |
QMQLQESGP.GLVKPSQSLFLACSITGFPIT...SGYYWIWIRQSPGKPLEWMGYITHS.
..GETFYNPSLQ.SPISITRETSKNQFFLQLNSVTTEDTAMYYCAGDR
>AC073589|IGHV13-1*01|Mus musculus_C57BL/6|P|V-REGION|21277..21580|304 nt|1| | | |101 AA|101+6=107| | |
EVQLVETGG.GLVQPGNSLKLSCATSGYPF....YDYWMDWVRHSPEKGLEWVARIATKT
HNYATYYAESVK.GRFIVSRDDSKSSAYMQMNSLRKEDTAVYYCARE
>AJ972403|IGHV13-1*02|Mus musculus_129/Sv|ORF|V-REGION|77460..77763|304 nt|1| | | |101 AA|101+6=107| | |
EVQLVETGG.GLVQPGNSLKLSCATSGYPF....YDYWMDWVRHFPEKGLEWVARIATKT
HNYATYYAESLK.GRFIVSRDDSKSSAYMQMNSLRKEDTAIYYWARE
>AC073590|IGHV13-2*01|Mus musculus_C57BL/6|F|V-REGION|123698..123999|302 nt|1| | ||100 AA|100+6=106| | |
QVQLVETGG.GLVRPGNSLKLSCVTSGFTF....SNYRMHWLRQPPGKRLEWIAVITVKS
DNYGANYAESVK.GRFAISRDDSKSSVYLEMNRLREEDTATYFCSR
>AJ972404|IGHV13-2*02|Mus musculus_129/Sv|F|V-REGION|58809..59110|302 nt|1| | ||100 AA|100+6=106| | |
QVQLVETGG.GLVRPGNSLKLSCVTSGFTF....SNYRMHWLRQPPGKRLEWIAVITVKS
DNYGANYAESVK.GRFTISRDDSKSSVYLQMNRLREEDTATYYCSR
>X55935|IGHV13-2*03|Mus musculus_NZB|(F)|V-REGION|1..300|300 nt|1| | | |100 AA|100+6=106| | |
QVQLVETGG.GLVRPGNSLKLSCVTSGFTF....SNYRMHWLRQPPGKRLEWIAVITVKS
DNYGANYAESVK.GRFAISRDDSKSSVYLQMDRLREEDTATYYCSR
>AC079273|IGHV14-1*01|Mus musculus_C57BL/6|F|V-REGION|118624..118917|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRPGASVKLSCTASGFNI....KDYYMHWVKQRPEQGLEWIGRIDPE.
.DGDTEYAPKFQ.GKATMTADTSSNTAYLQLSSLTSEDTAVYYCTT
>AJ851868|IGHV14-1*02|Mus musculus_129/Sv|F|V-REGION|356706..356999|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRPGALVKLSCKASGFNI....KDYYMHWVKQRPEQGLEWIGWIDPE.
.NGNTIYDPKFQ.GKASITADTSSNTAYLQLSSLTSEDTAVYYCAR
>AC073563|IGHV14-2*01|Mus musculus_C57BL/6|F|V-REGION|163123..163416|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVKPGASVKLSCTASGFNI....KDYYMHWVKQRTEQGLEWIGRIDPE.
.DGETKYAPKFQ.GKATITADTSSNTAYLQLSSLTSEDTAVYYCAR
>AJ851868|IGHV14-2*02|Mus musculus_129/Sv|P|V-REGION|289001..289294|294 nt|1| | | |98 AA|98+8=106| | |
*VKLQQSGA.ELVKPGASVKLSCKASGFNI....KDYYMH*VKQRPEQGLEWIGRIDPE.
.DGETKYAPKFQ.GKATITADTSSNTAYLQLSSLTSEDTAVYYCAR
>AC073563|IGHV14-3*01|Mus musculus_C57BL/6|F|V-REGION|97747..98040|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSVA.ELVRPGASVKLSCTASGFNI....KNTYMHWVKQRPEQGLEWIGRIDPA.
.NGNTKYAPKFQ.GKATITADTSSNTAYLQLSSLTSEDTAIYYCAR
>AJ851868|IGHV14-3*02|Mus musculus_129/Sv|F|V-REGION|226531..226824|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVKPGASVKLSCTASGFNI....KDTYMHWVKQRPEQGLEWIGRIDPA.
.NGNTKYDPKFQ.GKATITADTSSNTAYLQLSSLTSEDTAVYYCAR
>AC073589|IGHV14-4*01|Mus musculus_C57BL/6|F|V-REGION|112452..112745|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRPGASVKLSCTASGFNI....KDDYMHWVKQRPEQGLEWIGWIDPE.
.NGDTEYASKFQ.GKATITADTSSNTAYLQLSSLTSEDTAVYYCTT
>AJ851868|IGHV14-4*02|Mus musculus_129/Sv|F|V-REGION|50721..51014|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRSGASVKLSCTASGFNI....KDYYMHWVKQRPEQGLEWIGWIDPE.
.NGDTEYAPKFQ.GKATMTADTSSNTAYLQLSSLTSEDTAVYYCNA
>X55934|IGHV14S4*01|Mus musculus_MRL/lpr|(F)|V-REGION|1..291|291 nt|1| | | |97 AA|97+9=106|partial in 5'| |
.EVQLQQSG.AEVVPGASVKLSCTASGFNI....KDDYMHWAKQRPDQGLEWIGRIDPA.
.IDDTDYAPKFQ.DKATMITDTSSNIAYLQSSSLTSEDTAVYYCPY
>AC090843|IGHV15-2*01|Mus musculus_C57BL/6|F|V-REGION|67203..67499|297 nt|1| | | |99 AA|99+7=106| | |
QVHLQQSGS.ELRSPGSSVKLSCKDFDSEVF...PIAYMSWVRQKPGHGFEWIGGILPS.
.IGRTIYGEKFE.DKATLDADTLSNTAYLELNSLTSEDSAIYYCAR
>U39293|IGHV15-2*02|Mus musculus_BALB/c|(F)|V-REGION|55..351|297 nt|1| | | |99 AA|99+7=106| | |
QVHLQQSGS.ELRSPGSSVKLSCKDFDSEVF...PIAYMSWVRQKPGHGFEWIGDILPS.
.IGRTIYGEKFE.DKATLDADTVSNTAYLELNSLTSEDSAIYYCAR
>AC073563|IGHV16-1*01|Mus musculus_C57BL/6|ORF|V-REGION|88759..89057|299 nt|1| | | |99 AA|99+7=106| | |
EVQLVESGG.SLGQPGGSTKLSCEEASGFTF...SDHWMDWFRQAPGMRLEWLANTNHD.
.ESGKGYAESVK.DRFSISRDNSENLLYLQMNSLRNEDTALYYCAR
>X06864|IGHV1S10*01|Mus musculus_BALB/c|P|V-REGION|186..479|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGTSVKISCKASGYTF....LTYWMNWVK*MPGQGLEWIGQIFPA.
.SGSTNYNEMFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYFCAR
>M17574|IGHV1S10*02|Mus musculus_BALB/c|P|V-REGION|186..479|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGTSVKISCKASGYTF....LTYWMNWVK*RPAQGLEWIGQIFPA.
.SGSTNYNEMFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYFCAR
>L33934|IGHV1S100*01|Mus musculus_C57BL/6|[F]|V-REGION|718..970|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKASGYTF....TDYYMDWVKQSHGKSLEWIGDINPN.
.NGGTIYNQKFK.GKATLTVDKSSSTAYMELR
>L33936|IGHV1S101*01|Mus musculus_C57BL/6|[P]|V-REGION|720..972|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCMASGYQF....SDYYMH*VKQSHGKSLEWIGYINPN.
.NGCTSYNQKFK.GKATLTVDTSSSTAYMELH
>L33950|IGHV1S101*02|Mus musculus_C57BL/6|[P]|V-REGION|719..971|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCMASGYQF....SDYYMH*VKQSHGKSLEWIGYINPN.
.NGCTSYNQKFK.GKATLTVDTSSSTAYMELH
>L33942|IGHV1S103*01|Mus musculus_C57BL/6|[F]|V-REGION|714..966|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKASGYSF....TGYYMNWVKQSPEKSLEWIGEINPN.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMELR
>L33946|IGHV1S107*01|Mus musculus_C57BL/6|[F]|V-REGION|714..966|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKMSCKASGYKF....TDYYMHWVKQSHGKSLEWIGDINPN.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMQLN
>L33947|IGHV1S108*01|Mus musculus_C57BL/6|[F]|V-REGION|717..969|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKASGYTF....TDYYMNWVKQSHGKSLEWIGDINPN.
.NGGTSYNQKIK.GKATLTVDKSSSTAYMELR
>J00513|IGHV1S11*01|Mus musculus_BALB/c|P|V-REGION|752..1045|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKKSCKASGYTF....ANYWIGWVKQRPGHGLEWIGDIYPG.
.DGVTNYNEKFK.AKATLTADKSSSTAYMELSRLTSEDSAV*YCAR
>L33951|IGHV1S110*01|Mus musculus_C57BL/6|[P]|V-REGION|717..969|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCMASGLSF....SDYYMH*VKQSHGKSLEWIGYINPN.
.NGCTSYNQKFK.GKATLTVDTSSSTAYMELH
>L33952|IGHV1S111*01|Mus musculus_BALB/c|[F]|V-REGION|718..969|252 nt|1| | ||84 AA|84+8=92|partial in 3'| |
EVQLQQFGA.ELVKPGASVKISCKASGYTF....TDYNMDWVKQSHGKSLEWIGDINPN.
.YDSTSYNQKFK.GKATLTVDKSSSTAYMELR
>L33953|IGHV1S112*01|Mus musculus_BALB/c|[P]|V-REGION|718..970|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKAS*YTF....TDYNMHWVKQSHGKSLEWIGGINPN.
.NGATSYNQKFK.GKATLTVDKSSSTAYMELR
>L33959|IGHV1S112*02|Mus musculus_BALB/c|[F]|V-REGION|716..968|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQFGA.ELVKPGASVKISCKASGYTF....TDYNMHWVKQSHGKSLEWIGGINPN.
.NGATSYNQKFK.GKATLTVDKSSSTAYMELR
>L33954|IGHV1S113*01|Mus musculus_BALB/c|[F]|V-REGION|718..969|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKTSGYTF....TEYTMHWVKQSHGKSLEWIGGINPN.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMELR
>L33957|IGHV1S113*02|Mus musculus_BALB/c|[F]|V-REGION|717..969|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVQLQQSGP.ELVKPGALVKISCKTSGYTF....TEYTMHWVKQSHGKSLEWIGGINPN.
.NGGTSYNQKFK.GKATLTVDKSSSTAYMELR
>L33960|IGHV1S118*01|Mus musculus_BALB/c|[F]|V-REGION|715..967|253 nt|1| | | |84 AA|84+8=92|partial in 3'| |
EVLLQQSGP.ELVKPGASVKITCKASGYTF....TDYNMDWVKQSHGKSLEWIGDINPN.
.NGGTIYNQKFK.GKATLTVDKSSSTAYMELR
>J00507|IGHV1S12*01|Mus musculus_BALB/c|F|V-REGION|949..1242|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVKPGASVKISCKASGYTF....TSYYIHWVKQRPGQGLEWIGYIYPR.
.DGSTNYNEKFK.GKATLTADTSSSTAYMQLSSLTSEDSAVYFCAR
>AF025443|IGHV1S120*01|Mus musculus_BALB/c|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQSGA.ELVKPGASVKLSCKASGYTF....TSYYMYWVKQRPGQGLEWIGEINPS.
.NGGTNFNEKFK.SKATLTVDKSSSTAYMQLS
>AF025448|IGHV1S120*02|Mus musculus_BALB/c|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYYMYWVKQRPGQGLEWIGGINPS.
.NGGTNFNEKFK.SKATLTVDKSSSTAYMQLS
>AF025445|IGHV1S121*01|Mus musculus_BALB/c|[F]|V-REGION|708..959|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYNMHWVKQTPGQGLEWIGAIYPG.
.NGDTSYNQKFK.GKATLTADKSSSTAYMQLS
>AF025446|IGHV1S122*01|Mus musculus_BALB/c|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKLRPGQGFEWIGEINPS.
.NGGTNYNEKFK.RKATLTVDKSSSTAYMQLS
>AF025449|IGHV1S124*01|Mus musculus_BALB/c|[P]|V-REGION|704..955|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QAQIQQPGA.ELVRPGASVKMPCKASGYTF....TSYRMNWGKQRPGQGLEWIGYIYAG.
.SGSTDYNEKFK.SKATLTVDKSSSTAYMQLS
>AF304545|IGHV1S126*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKISCKASGYTF....TSYWMNWVKQRPGQGLEWIGEIDPS.
.DSYTNNNQKFK.DKATLTVDKSSSTAYMQLSSLTSEDSAVYYCAR
>AF304546|IGHV1S127*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWMHWVKQRPGQGLEWIGTIDPS.
.DSYTSYNQKFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVYYCTR
>AF304550|IGHV1S130*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGS.VLVRPGASVKLSCKASGYTF....TSSWMHWAKQRPGQGLEWIGEIHPN.
.SGNTNYNEKFK.GKATLTVDTSSSTAYVDLSSPTSEDSAVYYCAR
>AF304552|IGHV1S132*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVKPGASVKLSCKTSGYTF....TSYWIQWVKQRPGQGLGWIGEIFPG.
.TGTTYYNEKFK.GKATLTIDTSSSTAYMQLSSLTSEDSAVYFCAR
>AF304554|IGHV1S134*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELGRPGSSVKLSCKTSGYTF....TSYGIKWVKQRPGQGLEWIGYIYPG.
.NGYTVYNEKFQ.GKATLTSDTSSSTAYMQLRSLTSEDSAIYFCAR
>AF304556|IGHV1S135*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EIQLQQSGP.ELVKPGASVKVSCKASGYSF....TDYNMYWVKQSHGKSLEWIGYIDPY.
.NGGTSYNQKFK.GKATLTVDKSSSTAFMHLNSLTSEDSAVYYCAR
>AF304557|IGHV1S136*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKMSCKASGYTF....TDYVMHWVKQKPGQGLEWIGYIYPY.
.NDGTEYTEKFK.GKATLTLDKSSSTAYMDLSSLTSEDSTVYYCAR
>AF304558|IGHV1S137*01|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGVSVKISCKGSGYTF....TDYAMHWVKQSHAKSLEWIGVISTY.
.YGDASYNQKFK.GKATMTVDKSSSTAYMELARLTSEDSAVYYCAR
>K00707|IGHV1S14*01|Mus musculus_BALB/c|F|V-REGION|142..435|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGS.VLVRPGTSVKLSCKASGYTF....TSYWMHWAKQRPGQGLEWIGEIHPN.
.CGNINYNEKFK.GKATLTVDTSSSTAYVDLSSLTSEDSAVYYCAR
>K00603|IGHV1S15*01|Mus musculus_BALB/c|P|V-REGION|204..497|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWIT*VKQRPGRGLEWIGRIYPS.
.SGGTNYNEKFK.SKATLTVDTSSSTAYMQLSSLTSEDSAVYYCAI
>K00604|IGHV1S16*01|Mus musculus_BALB/c|P|V-REGION|205..498|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.KLVKPGASVKLSCKASGYTF....TSYWMHWVE*RPGQGLEWIGEINPS.
.NGGTNYNEKFK.RKATLTVDKSSSTAYMQLSSLTSEDSAVYYCTI
>K00605|IGHV1S17*01|Mus musculus_BALB/c|P|V-REGION|204..497|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMQWVKQRPGQGLEWIGNINPN.
.SGSTNYNEKFK.SKATLTVDKSSSTAYMQLSSLTSEDSAVYYCTR
>K00606|IGHV1S18*01|Mus musculus_BALB/c|P|V-REGION|62..355|294 nt|1| | | |98 AA|98+8=106| | |
*VQLQQPGA.ELVNTGASVKMSCKASGYTF....TSYTMHWVKQRLGQGLEWIGYINPS.
.SGYTNYNQKFK.DKATLTADKSSSTAYMQLSSLTSEDSAVYYCAR
>K00607|IGHV1S19*01|Mus musculus_BALB/c|P|V-REGION|139..432|294 nt|1| | | |98 AA|98+8=106| | |
QAQLQQSGA.ELVKPGASVKMSCKASGYSF....TSYYIHWVKQRPGQELEWIG*IFLG.
.SGNTKYNEKFK.GKATLTADTSSSTAYMQLSSLTSEDSAVHFCAR
>K02153|IGHV1S20*01|Mus musculus_A/J|F|V-REGION|1..251|251 nt|1| | ||83 AA|83+23=106|partial in 5'| |
................SSVKMSCKASGYTF....TSYGINWVKQRPGQGLEWIGYINPG.
.NGYTKYNEKFK.GKTTLTVDKSSSTAYMQLRSLTSEDSAVYFCAR
>X05746|IGHV1S20*02|Mus musculus_A/J|[F]|V-REGION|1..249|249 nt|1| | | |83 AA|83+23=106|partial in 5'| |
................SSVKMSCKASGYTF....TSYGINWVKQRPGQGLEWIGYINPG.
.NGYTKYNEKFK.GKTTLTVDKSSSTAYMQLRSLTSEDSAVYFCAR
>K02154|IGHV1S21*01|Mus musculus_A/J|F|V-REGION|1..230|230 nt|1| | ||76 AA|76+23=99|partial in 5' and in 3' | |
................SSVKLSCKTSGYTF....TSYGINWVKQRPGQGLEWIGYIYPG.
.NGYTAYNEQFK.GKATLTSDTSSSTAYMQLRSLTSEDS
>X05745|IGHV1S21*02|Mus musculus_A/J|[F]|V-REGION|1..222|222 nt|1| | ||74 AA|74+23=97|partial in 5' and in 3' | |
................SSVKLSCKTSGYTF....TSYGINWVKQRPGQGLEWIGYIYPG.
.NGYTAYNEQFK.GKATLTVDTSSSTAYMQLRSLTSE
>X02063|IGHV1S22*01|Mus musculus|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+11=106|partial in 5'| |
...LQQPGS.ELVRPGASVKLSCKASGYTF....TSYWMHWVKQRHGQGLEWIGNIYPG.
.SGSTNYDEKFK.SKGTLTVDTSSSTAYMHLSSLTSEDSAVYYCTR
>X02458|IGHV1S26*01|Mus musculus_BALB/c|F|V-REGION|101..394|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVKTGASVKMSCKASGYTF....TSYTMHWVKQRPGQGLEWIGYINPS.
.SGYTNYNQKFK.DKATLTADKSSSTAYMQLSSLTSEDSAVYYCAR
>X02460|IGHV1S28*01|Mus musculus_BALB/c|P|V-REGION|287..580|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKKSCKVSGYTF....ANYWIGWVKQRPGHGLEWIGDIYPG.
.DGVTNYNEKFK.GKATLTADKSSSTAYM*LSSLTSEDSAVYYCSR
>X02461|IGHV1S29*01|Mus musculus_BALB/c|F|V-REGION|101..377|277 nt|1| | | |92 AA|92+8=100|partial in 3'| |
EVQLQQSGP.ELVKPGASVKISCKASGYTF....TDYNMHWVKQSHGKSLEWIGYIYPY.
.NGGTGYNQKFK.SKATLTVDNSSSTAYMDVRSLTSEDSA
>J00488|IGHV1S29*02|Mus musculus_BALB/c|F|V-REGION|341..634|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGYTF....TDYNMHWVKQSHGKSLEWIGYIYPY.
.NGGTGYNQKFK.SKATLTVDNSSSTAYMELSSLTSEDSAVYYCAR
>X02462|IGHV1S30*01|Mus musculus_BALB/c|P|V-REGION|264..557|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKLGPSVKISCKASGYSF....TGYYMHWVKQSHGKSLEWIGEINPY.
.NGGTSYNQKFK.GKATLTVDTSSSTAYMELHSLTSEDSLVYYCAR
>D13201|IGHV1S30*02|Mus musculus_BALB/c|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+11=106|partial in 5'| |
...LQQSGP.ELVKPGPSVKISCKASGYSF....TGYYMHWVKQSHGKSLEWIGEINPY.
.NGGTSYNQKFK.GKATLTVDTSSSTAYMELHSLTSEDSLVYYCAR
>X02463|IGHV1S31*01|Mus musculus_BALB/c|F|V-REGION|1..208|208 nt|1| | | |69 AA|69+11=80|partial in 5' and in 3' | |
...LQQSGP.ELVKPGASVRISCKASGYTF....TSYYIHWVKQRPGQGLEWIGWIYPG.
.NVNTKYNEKFK.GKATLTA
>X02464|IGHV1S32*01|Mus musculus_BALB/c|F|V-REGION|1..208|208 nt|1| | | |69 AA|69+11=80|partial in 5' and in 3' | |
...LQQSGP.ELVKPGASVKMSCKASGYTF....TSYYIHWVKQRPGQGLEWIGWIYPG.
.DGSTKYNEKFK.GKTTLTA
>X02465|IGHV1S33*01|Mus musculus_BALB/c|F|V-REGION|1..268|268 nt|1| | | |89 AA|89+11=100|partial in 5' and in 3' | |
...LQQSGP.ELVKPGALVKISCKASGYTF....TSYDINWVKQRPGQGLEWIGWIYPG.
.DGSTKYNEKFK.GKATLTADKSSSTAYMQLSSLTSENSA
>X02467|IGHV1S34*01|Mus musculus_BALB/c|F|V-REGION|1..266|266 nt|1| | | |88 AA|88+18=106|partial in 5'| |
...........LVKTGASVKISCKASGYSF....TGYYMHWVKQSHGKSLEWIGYISCY.
.NGATSYNQKFK.GKATFTVDTSSSTAYMQFNSLTSEDSAVYYCAR
>M12376|IGHV1S35*01|Mus musculus_BALB/c|F|V-REGION|106..399|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.VLVRHGASVKLSCKASGYTF....TSSWMHWAKQRHGQGLEWIGEIHPN.
.SGNTNYNEKFK.GKATLTVDKSSSTAYVDLSSLTSEDSAVYYCAR
>M13788|IGHV1S36*01|Mus musculus|F|V-REGION|181..474|294 nt|1| | | |98 AA|98+8=106| | |
PVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGNIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLSSLTSEDSAVYYCTR
>AF304548|IGHV1S36*02|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
PVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQKPGRSLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLSSLTSEDSAVYYCTR
>M13789|IGHV1S37*01|Mus musculus|F|V-REGION|1..210|210 nt|1| | | |70 AA|70+11=81|partial in 5' and in 3' | |
...LQQSGP.QLVSPGASVKISCKASGYSF....TSYWMHWVKQRPGQGLEWIGMIDPS.
.DSETRLNQKFK.DKATLTVD
>M17575|IGHV1S40*01|Mus musculus_BALB/c|F|V-REGION|185..478|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGLSVKLSCKASGYIF....ITYWMNWVKQRPGQGLEWIGQIFPA.
.SGSTNYNEMFE.GKATLTVDTSSSTAYMQLSSLTSEDSAVYYCAR
>X06868|IGHV1S41*01|Mus musculus_BALB/c|F|V-REGION|1..267|267 nt|1| | | |89 AA|89+17=106|partial in 5'| |
..........DLVKPGASVKLSCKASGYTF....TSYWINWIKQRPGQGLEWIGAIAPG.
.SGSTYYNEMFK.GKATLTVDTSSSTAYIQLSSLASEDSAVYFCAR
>M19402|IGHV1S44*01|Mus musculus_I/St|[F]|V-REGION|101..304|204 nt|1| | | |68 AA|68+8=76|partial in 3'| |
EIQLQQSGP.ELVKPGASVKMSCKASGYTF....TDYYMHWVKQSNGKSLEWIGYINPY.
.NDYTSYNQKFN.GKA
>M19403|IGHV1S45*01|Mus musculus_I/St|F|V-REGION|385..678|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKMSCKASGYTF....TDYVMHWVKQSNGKSLEWIGYINPY.
.NDYTSYNQKFK.GKATLTVDKSSSTAYMQLNSLTSEDSAVYYCAR
>M20774|IGHV1S46*01|Mus musculus_I/St|[F]|V-REGION|385..679|295 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKISCKASGFTF....TDYVMHWVKQSHGKSLEWIGLIIPY.
.NGDTFYNQKFK.GKATLTVDTSSSTAYMELNSLTSEDFAVYYCAR
>M34977|IGHV1S47*01|Mus musculus_A/J|ORF|V-REGION|259..552|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELGRPGSSVKLSCKTSGYTF....TSYGINWVKQRPGQDLEWIGYIYPG.
.NGYTAYNEKFQ.GEATLTSDTSSSTAYMQLRSLTSEDSAIYFCAR
>M34979|IGHV1S49*01|Mus musculus_A/J|F|V-REGION|259..552|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGA.ELVRPGSSVKLSCKTSGYTF....TSYGINWVKQRPGQGLEWIGYIYLG.
.NGYTAYNEKFK.GKATLTSDTSSSTAYMQLRSLTSEDSVIKFCAR
>J00535|IGHV1S5*01|Mus musculus_C57BL/6|F|V-REGION|222..515|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGEGLEWIGNIYPG.
.SSSTNYNEKFK.SKATLTVDTPSSTAYMQLSSLTSEDSAVYYCAR
>M34980|IGHV1S50*01|Mus musculus_A/J|F|V-REGION|261..554|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVKPGASVRISCKTSGYTF....TSYNIHWVKERPGQGLEWIGWIYPG.
.DGNTKYNEKFK.GKTTLTADKSSSTAYMQLSSLTSEDSAVYFCAR
>M34981|IGHV1S51*01|Mus musculus_A/J|P|V-REGION|261..554|294 nt|1| | ||98 AA|98+8=106| | |
QVQLQQSGP.ELVRPGTSVKISCKASGYTF....ITYWMNWVK*RPGQGLEWIGQIFPA.
.SGSTNYNEMFK.GKATLTVDTSSSTAYMQLSSLTSEDSAVHFCAR
>M34982|IGHV1S52*01|Mus musculus_A/J|F|V-REGION|256..549|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGA.ELVRPGTSVKVSCKASGYVF....TNYLIEWVKQRPGQGLEWIGVINPG.
.SGGTNYNEKFK.GKATLTADKSSSTAYMQLSSLTSDDSAVYFCAR
>M34983|IGHV1S53*01|Mus musculus_A/J|F|V-REGION|254..547|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSDA.ELVKPGASVKISCKASGYTF....TDHAIHWVKQKPEQGLEWIGYISPG.
.NGDIKYNEKFK.GKATLTADKSSSTAYMQLNSLTSEDSAVYFCKR
>L14547|IGHV1S53*02|Mus musculus_BALB/c|F|V-REGION|520..813|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSDA.ELVKPGASVKISCKASGYTF....TDHAIHWVKQKPEQGLEWIGYISPG.
.NGDIKYNEKFK.GKATLTADKSSSTAYMQLNSLTSEDSAVYFCKR
>L14548|IGHV1S53*03|Mus musculus_BALB/c|F|V-REGION|366..659|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSDT.ELVKPGASVKISCKASGYTF....TDHAIHWVKQRPEQGLEWIGYISPG.
.NGDIKYNEKFK.GKATLTADKSSSTAYMQLNSLTSEDSAVYFCKR
>M34985|IGHV1S55*01|Mus musculus_A/J|F|V-REGION|254..547|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGS.VLVRPGASVKLSCKASGYTF....TSYWMNWVKQRPGQGLEWIGGIYPN.
.SGSTDYNEKFK.GKATLTVDTSSSTTYMDLSSLTSKDSAVYYCAR
>M34987|IGHV1S56*01|Mus musculus_A/J|F|V-REGION|125..418|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQSGP.ELVKPGASVRISCKASGYTF....TSYNIHWVKQRPGQGLEWIGWIYPG.
.DGNTKYNEKFK.GKTTLTADKSSSTAYMQLSSLTSEDSAVYFCAR
>D14634|IGHV1S61*01|Mus musculus_MRL/lpr|F|V-REGION|284..577|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWINWVKQRPGQGLEWIGNIYPG.
.SSSTNYNEKFK.SKATLTVDTSSSTAYMQLSSLTSDDSAVYYCAR
>L26856|IGHV1S65*01|Mus musculus_C57BL/6|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
PVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLS
>L26873|IGHV1S65*02|Mus musculus_BALB/c|[P]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
PVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWI*RIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLS
>L26886|IGHV1S65*03|Mus musculus_BALB/c|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
PVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPN.
.SGGTKYNEKFK.SKATLTVDKPSSTAYMQLS
>L26858|IGHV1S67*01|Mus musculus_C57BL/6|[F]|V-REGION|726..977|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMYWVKQRPGQGLERIGEINPG.
.NGGTNYNEKFK.SKATLTVDKSSSTAYMQLS
>AF025447|IGHV1S67*02|Mus musculus_BALB/c|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMYWVKQRPGQGLERIGEINPS.
.NGGTNYNEKFK.SKATLTVDKSSSTAYMQLS
>L26860|IGHV1S68*01|Mus musculus_C57BL/6|[F]|V-REGION|718..969|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGTSVKLSCKASGYTF....TSYWINWVKLRPGQGLEWIGDIYPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLS
>L26874|IGHV1S68*02|Mus musculus_BALB/c|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGTSVKLSCKASGYNF....TSYWINWVKLRPGQGLEWIGDIYPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLS
>L26867|IGHV1S70*01|Mus musculus_BALB/c|[F]|V-REGION|714..965|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVRPGASVKLSCKASGYTF....TSYWINWVKQRPGQGLEWIGNIYPL.
.DSNTNYNQKFK.DKATLTVDKSSSTAYMQLS
>L26869|IGHV1S72*01|Mus musculus_BALB/c|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGAPVKLSCKASGYTF....TSYWMNWVKQRPGRGLEWIGRIDPS.
.DSETHYNQKFK.DKATLTVDKSSSTAYIQLS
>L26870|IGHV1S73*01|Mus musculus_BALB/c|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGFEWIGRIHPS.
.NGGTNYNEKFK.SKATLTVDTSSSTAYMHLS
>L26871|IGHV1S74*01|Mus musculus_BALB/c|[P]|V-REGION|711..962|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVELQQSGP.QLVKPGASVKISCKASGYSF....TSYWMHWVKQRPGQGLEWIAMIDPS.
.DSETK*NQKFK.SKATLTVDKSSSTAYMQLS
>L26872|IGHV1S75*01|Mus musculus_BALB/c|[F]|V-REGION|709..960|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWINWVKQRPGQGLEWIGDIYPG.
.RGITNYNEKFK.SKATLTLDTSSSTAYMQLS
>L26885|IGHV1S75*02|Mus musculus_BALB/c|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKMSCKASGYTF....TSYWINWVKQRPGQGLEWIGDIYPG.
.TGITNYNEKFK.SKATLTLDTSSSTAYMQLS
>L26875|IGHV1S78*01|Mus musculus_BALB/c|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVMPGASVKMSCKASGYTF....TDYWMHWVKQRPGQGLEWIGTIDTS.
.DSYTSYNQKFK.GKATLTVDESSSTAYMQLS
>L26880|IGHV1S81*01|Mus musculus_BALB/c|[F]|V-REGION|715..966|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGEINPS.
.NGRTNYNEKFK.SKATLTVDKSSSTAYMQLS
>AF304547|IGHV1S81*02|Mus musculus_BALB/c|[F]|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGEINPS.
.NGRTNYNEKFK.SKATLTVDKSSSTAYMQLSSPTSEDSAVYYCAR
>L26881|IGHV1S82*01|Mus musculus_BALB/c|[F]|V-REGION|703..954|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVRPGASVKLSCKASGYSF....TSYWMNWVKQRPGQGLEWIGMIHPS.
.DSETRLNQKFK.DKATLTVDKSSSTAYMQLS
>L26882|IGHV1S83*01|Mus musculus_BALB/c|[F]|V-REGION|706..957|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGS.ELVRPGASVKLSCKASGYTF....TSSWMHWAKQRPGQGLEWIGRIDPN.
.SGNTNYNEKFK.GKATLTVDTSSSTAYVDLS
>L26883|IGHV1S84*01|Mus musculus_BALB/c|[P]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGSSVKLSCKASGYNF....TSYWINWVKLRPYQGIEWI*DIYPG.
.SGSTNYNEKFK.SKATLTVDTSSSTAYMQLS
>L26925|IGHV1S87*01|Mus musculus_C57BL/6|[F]|V-REGION|710..961|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGT.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGEGLEWIGNIYPG.
.SSSTNYNEKFK.SKATLTVDTSSSTAYMQLS
>J00511|IGHV1S9*01|Mus musculus_BALB/c|P|V-REGION|817..1110|294 nt|1| | | |98 AA|98+8=106| | |
EVQLQQSGP.ELVKPGASVKITCKASDYSF....TGYIMNWVKQSHGKSLEWIGEINPY.
.NGGTSYNQKFK.GKATLTVDTSSSTAYMELHSLTSEDSLVYYCAR
>L26931|IGHV1S92*01|Mus musculus_C57BL/6|[F]|V-REGION|717..968|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGMIHPN.
.SGSTNYNEKFK.SKATLTVDKSSSSAYMKLI
>L26934|IGHV1S95*01|Mus musculus_C57BL/6|[F]|V-REGION|707..958|252 nt|1| | | |84 AA|84+8=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGQGLEWIGRIDPG.
.SSSTNYNEKFK.SKATLTVDTSSSTAYMQLI
>L26935|IGHV1S96*01|Mus musculus_C57BL/6|[F]|V-REGION|707..955|249 nt|1| | ||83 AA|83+9=92|partial in 3'| |
QVQLQQPGA.ELVKPGASVKLSCKASGYTF....TSYWMHWVKQRPGRGLEWIGRIDPY.
..SDIKYSEKFK.NKATLTVDKPSNTAYMQLS
>AC090887|IGHV2-2*01|Mus musculus_C57BL/6|F|V-REGION|46221..46513|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQADDTAIYYCAR
>AJ851868|IGHV2-2*02|Mus musculus_129/Sv|F|V-REGION|1033754..1034046|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQANDTAIYYCAR
>J00502|IGHV2-2*03|Mus musculus_BALB/c|F|V-REGION|408..700|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQSNDTAIYYCAR
>AJ851868|IGHV2-2-1*01|Mus musculus_129/Sv|P|V-REGION|821838..822130|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTFSGF*L....TSYGVHWERHSPGKGLEWLGVIWSG.
..VHTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQADDTAIYYCAR
>AJ851868|IGHV2-2-2*01|Mus musculus_129/Sv|F|V-REGION|565106..565398|293 nt|1| | | |97 AA|97+9=106| | |
QVQMKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQADDTAIYYCVR
>AC090887|IGHV2-3*01|Mus musculus_C57BL/6|F|V-REGION|69138..69430|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVSWVRQPPGKGLEWLGVIWGD.
..GSTNYHSALI.SRLSISKDNSKSQVFLKLNSLQTDDTATYYCAK
>AJ851868|IGHV2-3-1*01|Mus musculus_129/Sv|F|V-REGION|764425..764717|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWSG.
..GSTDYNAPFI.SRLSISKDNSKSQIFFKMNSLQADDTAIYYCAR
>AC090887|IGHV2-4*01|Mus musculus_C57BL/6|F|V-REGION|111245..111537|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQPPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQADDTAIYYCAK
>AJ851868|IGHV2-4*02|Mus musculus_129/Sv|F|V-REGION|968711..969003|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQPPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQADDTAIYYCAR
>AJ851868|IGHV2-4-1*01|Mus musculus_129/Sv|F|V-REGION|787959..788251|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWSG.
..GSTDYNAAFI.SRLSISKDNSKSQVFFKMNSLQADDTAIYYCAR
>AC090887|IGHV2-5*01|Mus musculus_C57BL/6|F|V-REGION|143436..143728|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWRG.
..GSTDYNAAFM.SRLSITKDNSKSQVFFKMNSLQADDTAIYYCAK
>AJ851868|IGHV2-5-1*01|Mus musculus_129/Sv|F|V-REGION|715517..715809|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.SLVQPSQSLSITCTVSGFSL....TSYGVHWVRQSPGKGLEWLGVIWRG.
..GSTDYNAAFM.SRLSITKDNSKSQVFFKMNSLQADDTAIYYCAK
>CAAA01078923|IGHV2-6*01|Mus musculus_C57BL/6|F|V-REGION|5653..5945|293 nt|1| | ||97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVDWVRQSPGKGLEWLGVIWGV.
..GSTNYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAS
>AJ851868|IGHV2-6*02|Mus musculus_129/Sv|F|V-REGION|885012..885304|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVHWVRQPPGKGLEWLVVIWSD.
..GSTTYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAR
>BN000872|IGHV2-6*03|Mus musculus_C57BL/6|F|V-REGION|0..0|291 nt|1| | ||97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVHWVRQPPGKGLEWLVVIWSD.
..GSTTYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAR
>AJ851868|IGHV2-6-1*01|Mus musculus_129/Sv|F|V-REGION|854274..854566|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTISGFSL....TSYGVHWVRQPPGKGLEWLVVIWSD.
..GSTTYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAR
>AJ851868|IGHV2-6-2*01|Mus musculus_129/Sv|F|V-REGION|747982..748274|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.DLVAPSQSLSITCTVSGFSL....TSYGVHWVRQPPGKGLEWLVVIWSD.
..GSTTYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAR
>AJ851868|IGHV2-6-3*01|Mus musculus_129/Sv|F|V-REGION|672560..672849|290 nt|1| | | |96 AA|96+10=106| | |
QVQ.NKSGP.GLVEPSQSLSITCTVYWFSL....TSYGVSWVRQPPGKGLKWLGVIWAG.
..GSTNYNSALI.SRLSISKDNSKSQVFLKMNSLQTDDTAIYYCVR
>AJ851868|IGHV2-6-4*01|Mus musculus_129/Sv|F|V-REGION|586956..587248|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....SRYSVHWVRQPPGKGLEWLGMIWGG.
..GSTDYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAR
>AJ851868|IGHV2-6-5*01|Mus musculus_129/Sv|F|V-REGION|552420..552712|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TDYGVSWIRQPPGKGLEWLGVIWGG.
..GSTYYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAK
>AJ851868|IGHV2-6-6*01|Mus musculus_129/Sv|F|V-REGION|535840..536132|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TNSGVHWVRQSPGKGLEWLGVIWGD.
..GSTNYNSAFK.SRLSISKDNSKSQVFLKMNSLQTDDTARYYCAK
>AJ851868|IGHV2-6-7*01|Mus musculus_129/Sv|F|V-REGION|494384..494676|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TGYGVNWVRQPPGKGLEWLGMIWGD.
..GSTDYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTARYYCAR
>V00767|IGHV2-6-7*02|Mus musculus|F|V-REGION|269..561|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TGYGVNWVRQPPGKGLEWLGMIWGD.
..GSTDYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTARYYCAR
>BN000872|IGHV2-6-8*01|Mus musculus_C57BL/6|F|V-REGION|2274719..2275011|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVDWVRQSPGKGLEWLGVIWGV.
..GSTNYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAS
>CAAA01073483|IGHV2-7*01|Mus musculus_C57BL/6|F|V-REGION|19354..19646|293 nt|1| | | |97 AA|97+9=106| | |
QVQMKESGP.DLVQPSQTLSLTCTVSGFSL....SSYGVHWFRKPPRKGLEWLGGIWSG.
..GSIYYTPALS.SRLSVSRDTSKSQVFFKMSSLQSEDTAVYHCAR
>AC079273|IGHV2-9*01|Mus musculus_C57BL/6|F|V-REGION|65767..66059|293 nt|1| | ||97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVDWVRQPPGKGLEWLGVIWGG.
..GSTNYNSALM.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAK
>AJ851868|IGHV2-9*02|Mus musculus_129/Sv|F|V-REGION|405504..405796|293 nt|1| | ||97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYGVHWVRQPPGKGLEWLGVIWAG.
..GSTNYNSALM.SRLSISKDNSKSQVFLKMNSLQTDDTAMYYCAR
>AC160985|IGHV2-9-1*01|Mus musculus_C57BL/6|F|V-REGION|107139..107431|293 nt|1| | ||97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYAISWVRQPPGKGLEWLGVIWTG.
..GGTNYNSALK.SRLSISKDNSKSQVFLKMNSLQTDDTARYYCAR
>AJ851868|IGHV2-9-2*01|Mus musculus_129/Sv|F|V-REGION|770242..770534|293 nt|1| | ||97 AA|97+9=106| | |
QVQLKESGP.GLVAPSQSLSITCTVSGFSL....TSYDISWIRQPPGKGLEWLGVIWTG.
..GGTNYNSAFM.SRLSISKDNSKSQVFLKMNSLQTDDTAIYYCVR
>M27021|IGHV2S3*01|Mus musculus_NFS|F|V-REGION|367..659|293 nt|1| | | |97 AA|97+9=106| | |
QVQLKQSGP.GLVAPSQSLFITCTVYGFSL....TSYEINWVRQPPGKGLEWLGVIWTG.
..GSTNYNSALI.SRLSISKDNSKSLVFLKMNSLQTDDTAIYYCVR
>AC073563|IGHV3-1*01|Mus musculus_C57BL/6|F|V-REGION|193242..193537|296 nt|1| | ||98 AA|98+8=106| | |
DVQLQESGP.GMVKPSQSLSLTCTVTGYSIT...SGYDWHWIRHFPGNKLEWMGYISYS.
..GSTNYNPSLK.SRISITHDTSKNHFFLKLNSVTTEDTATYYCAR
>AJ851868|IGHV3-1*02|Mus musculus_129/Sv|F|V-REGION|314932..315225|294 nt|1| | | |98 AA|98+8=106| | |
DVQLQESGP.DLVKPSQSLSLTCTVTGYSIT...SGYSWHWIRQFPGNKLEWMGYIHYS.
..GSTNYNPSLK.SRISITRDTSKNQFFLQLNSVTTEDTATYYCAR
>AJ851868|IGHV3-2*02|Mus musculus_129/Sv|F|V-REGION|252920..253213|294 nt|1| | | |98 AA|98+8=106| | |
DVQLQESGP.GLVKPSQSLSLTCTVTGYSIT...SDYAWNWIRQFPGNKLEWMGYISYS.
..GSTSYNPSLK.SRISITRDTSKNQFFLQLNSVTTEDTATYYCAR
>AC073589|IGHV3-3*01|Mus musculus_C57BL/6|F|V-REGION|92449..92744|296 nt|1| | | |98 AA|98+8=106| | |
DVQLQESGP.SLVRPSQTLSLTCTVTGFSIN...SDCYWIWIRQFPGNKLEYIGYTFYS.
..GITYYNPSLE.SRTYITRDTSKNQFSLKLSSVTTEDTATYYCAR
>AJ851868|IGHV3-3*02|Mus musculus_129/Sv|P|V-REGION|12959..13254|296 nt|1| | | |98 AA|98+8=106| | |
DVQLLESGP.SLVRPSQTLSLTCTVTGFSIN...SDCYWI*IRQFPGNKLEYIGNTFYS.
..GITYYNPSLE.SRTYITRDTSKNQFSLKLSSVTTEDTATYYCAR
>D13203|IGHV3-3*03|Mus musculus_BALB/c|F|V-REGION|1..148|148 nt|1| | | |48 AA|48+58=106|partial in 5'| |
......................................................DTFYS.
..GITYYNPSLE.SRTYITRDTSKNQFSLKLSSVTTEDTATYYCAR
>AC073590|IGHV3-4*01|Mus musculus_C57BL/6|F|V-REGION|19554..19852|299 nt|1| | | |99 AA|99+7=106| | |
DVQLQESGP.ALVKPSQTVSLTCTVTGYSIT..NGNHWWNWIRQVSGSKLEWIGYISSS.
..GSTDSNPSLK.SRISITRDTSKNQLFLQLNSVTTEDIATYYCAR
>AJ972403|IGHV3-4*02|Mus musculus_129/Sv|F|V-REGION|63447..63745|299 nt|1| | | |99 AA|99+7=106| | |
DVQLQESGP.GLVKPSQTVSLTCTVTGYSIT..NGNHWWNWIRQVSGNKLEWMGYISSS.
..GSTDSNPSLK.SQISITRDTSKNQLFLQLNSVTIEDIATYYCAR
>AC073590|IGHV3-5*01|Mus musculus_C57BL/6|F|V-REGION|28589..28887|299 nt|1| | | |99 AA|99+7=106| | |
DVQLQESGP.GLVKPSQTVFLTCTVTGISIT..TGNYRWSWIRQFPGNKLEWIGYIYYS.
..GTITYNPSLT.SRTTITRDTPKNQFFLEMNSLTAEDTATYYCAR
>AJ972403|IGHV3-5*02|Mus musculus_129/Sv|F|V-REGION|72462..72760|299 nt|1| | | |99 AA|99+7=106| | |
DVQLQESGP.GLVKPSQTVSLTCTVTGISIT..TGNYRWSWIRQFPGNKLEWIGYIYYS.
..GTITYNPSLT.SRTTITRDTSKNQFFLEMNSLTAEDTATYYCAR
>AC073590|IGHV3-6*01|Mus musculus_C57BL/6|F|V-REGION|54090..54385|296 nt|1| | | |98 AA|98+8=106| | |
DVQLQESGP.GLVKPSQSLSLTCSVTGYSIT...SGYYWNWIRQFPGNKLEWMGYISYD.
..GSNNYNPSLK.NRISITRDTSKNQFFLKLNSVTTEDTATYYCAR
>AJ972403|IGHV3-6*02|Mus musculus_129/Sv|F|V-REGION|94066..94361|296 nt|1| | | |98 AA|98+8=106| | |
DVQLQESGP.GLVKPSQSLSLTCSVTGYSIT...SGYYWNWIRQFPGNKLEWMGYISYD.
..GSNNYNPSLK.NRISITRDTSKNQFFLKLNSVTTEDTATYYCAR
>AJ223544|IGHV3-6*03|Mus musculus_BALB/c|(F)|V-REGION|55..348|294 nt|1| | | |98 AA|98+8=106| | |
DVQLQESGP.GLVKPSQSLSLTCSVTGYSIT...SGYYWNWIRQFPGNKLEWMGQISYD.
..GSNNYNPSLK.NRISITRDTSKNQFFLKLNSVTTEDTATYYCAR
>AC073590|IGHV3-7*01|Mus musculus_C57BL/6|P|V-REGION|80123..80418|296 nt|1| | ||98 AA|98+8=106| | |
EVQLQESRP.SLVKPSQSLSFTCSVTGYSIT...SGYDWNWIRQFPGNKL*WMRYIHKS.
..GSTNYNPSLK.SQVSITRDTSKNQFYLQLNSVTTEDTAIYYCAN
>AJ972403|IGHV3-7*02|Mus musculus_129/Sv|P|V-REGION|120098..120393|296 nt|1| | | |98 AA|98+8=106| | |
EVQLQESGP.DLVKTSQSLSFTCSVTGYSIT...SGYDWNCIWQFPGNKL*CMRYIHKS.
..GSTNYNPSLK.SQVSITGDTSKNQFYLQLNSVTTEDTAIYYCAN
>AC073590|IGHV3-8*01|Mus musculus_C57BL/6|F|V-REGION|88313..88605|293 nt|1| | ||97 AA|97+9=106| | |
EVQLQESGP.GLAKPSQTLSLTCSVTGYSI....TSDYWNWIRKFPGNKLEYMGYISYS.
..GSTYYNPSLK.SRISITRDTSKNQYYLQLNSVTTEDTATYYCAR
>AJ972403|IGHV3-8*02|Mus musculus_129/Sv|F|V-REGION|127596..127888|293 nt|1| | | |97 AA|97+9=106| | |
EVQLQESGP.SLVKPSQTLSLTCSVTGDSI....TSGYWNWIRKFPGNKLEYMGYISYS.
..GSTYYNPSLK.SRISITRDTSKNQYYLQLNSVTTEDTATYYCAR
>K01569|IGHV3S1*01|Mus musculus_A/J|F|V-REGION|1..293|293 nt|1| | | |97 AA|97+9=106| | |
EVQLQESGP.SLVKPSQTLSLTCSVTGDSI....TSDYWNWIRKFPGNKLEYMGYISYS.
..GSTYYNPSLK.SRISITRDTSKNQYYLQLNSVTSEDTATYYCAR
>K02790|IGHV3S1*02|Mus musculus_BALB/c|F|V-REGION|1..293|293 nt|1| | | |97 AA|97+9=106| | |
EVQLQESGP.SLVKPSQTLSLTCSVTGDSI....TSGYWNWIRKFPGNKLEYMGYISYS.
..GSTYYNPSLK.SRISITRDTSKNQYYLQLNSVTTEDTPTYYCAR
>AJ223545|IGHV3S7*01|Mus musculus_C57BL/6|(F)|V-REGION|55..347|293 nt|1| | | |97 AA|97+9=106| | |
EVQLQESGP.GLAKPSQTLSLTCSATGYSI....TSDYWNWIRKFPGNKLEYMGFISGS.
..GSTYYNPSLK.SRISITRDTSKNQYYLQLNSVTTEDAATYYCAR
>AC079273|IGHV4-1*01|Mus musculus_C57BL/6|F|V-REGION|134953..135248|296 nt|1| | ||98 AA|98+8=106| | |
EVKLLQSGG.GLVQPGGSLKLSCAASGIDF....SRYWMSWVRRAPGKGLEWIGEINPD.
.SSTINYAPSLK.DKFIISRDNAKNTLYLQMSKVRSEDTALYYCAR
>AJ851868|IGHV4-1*02|Mus musculus_129/Sv|F|V-REGION|334879..335174|296 nt|1| | | |98 AA|98+8=106| | |
EVKLLESGG.GLVQPGGSLKLSCAASGFDF....SRYWMSWVRQAPGKGLEWIGEINPD.
.SSTINYTPSLK.DKFIISRDNAKNTLYLQMSKVRSEDTALYYCAR
>AC073563|IGHV4-2*01|Mus musculus_C57BL/6|P|V-REGION|144446..144741|296 nt|1| | | |98 AA|98+8=106| | |
EVKLLESGG.GLVQPGGSLKLSCAASGFDF....SKDWMSWVRQATGKGLE*IGEINPG.
.SSTINYTPSLK.DKFIISRDNAKNTLYLQMSKVRSEDTALYYCAR
>AJ851868|IGHV4-2*02|Mus musculus_129/Sv|F|V-REGION|272931..273226|296 nt|1| | | |98 AA|98+8=106| | |
EVKLLESGG.GLVQPGGSLNLSCAASGFDF....SRYWMSWARQAPGKGQEWIGEINPG.
.SSTINYTPSLK.DKFIISRDNAKNTLYLQMSKVRSEDTALYYCAR
>AC090887|IGHV5-1*01|Mus musculus_C57BL/6|P|V-REGION|30881..31176|296 nt|1| | ||98 AA|98+8=106| | |
EVQLVESGG.GLVQPGGSRKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVAAISTD.
.GSFIY*PDTVK.GRFTISRDNAKNTLFLQMSSLRYEDTAMYYCLR
>AJ851868|IGHV5-1*02|Mus musculus_129/Sv|P|V-REGION|1049088..1049383|296 nt|1| | | |98 AA|98+8=106| | |
ELQLVESGG.GLVQPGGSRKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVAAISTD.
.GSFIY*PDTVK.GRFTISRDNAKNTLFLQMSSLRYEDTAMYYCLR
>AC090887|IGHV5-12*01|Mus musculus_C57BL/6|F|V-REGION|160078..160373|296 nt|1| | ||98 AA|98+8=106| | |
EVKLVESGG.GLVQPGGSLKLSCAASGFTF....SDYYMYWVRQTPEKRLEWVAYISNG.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSRLKSEDTAMYYCAR
>AJ851868|IGHV5-12*02|Mus musculus_129/Sv|F|V-REGION|899557..899852|296 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVQPGGSLKLSCATSGFTF....SDYYMYWVRQTPEKRLEWVAYISNG.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSRLKSEDTAMYYCAR
>AF120465|IGHV5-12*03|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVQPGGSLKLSCAASGFTF....SDYYMYWVRQTPEKRLEWVAYISNG.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLEMSSLRSEDTAMYYCARR
>AJ851868|IGHV5-12-1*01|Mus musculus_129/Sv|F|V-REGION|698825..699120|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVKPGGSLKLSCAASGFAF....SSYDMSWVRQTPEKRLEWVAYISSG.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>AJ851868|IGHV5-12-2*01|Mus musculus_129/Sv|F|V-REGION|618131..618426|296 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVQPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVAYISNG.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>AF290963|IGHV5-12-2*02|Mus musculus_129/Sv|F|V-REGION|1..263|263 nt|1| | | |87 AA|87+19=106|partial in 5'| |
............VKPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVAYISNG.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>BN000872|IGHV5-12-4*01|Mus musculus_C57BL/6|F|V-REGION|2308604..2308898|294 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVKPGQSLKLSCAASGFTF....SNYYMSWVHQTPEKRLEWVAYISSS.
.GVSTYYPDNVK.GRFAISRDNAKNTLYLQMTSLKSEDTALYYCAR
>AC079273|IGHV5-15*01|Mus musculus_C57BL/6|F|V-REGION|13317..13612|296 nt|1| | ||98 AA|98+8=106| | |
EVKLVESGG.GLVQPGGSLKLSCAASGFTF....SDYGMAWVRQAPRKGPEWVAFISNL.
.AYSIYYADTVT.GRFTISRENAKNTLYLEMSSLRSEDTAMYYCAR
>AJ851868|IGHV5-15*02|Mus musculus_129/Sv|F|V-REGION|455690..455985|296 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVQPGGSRKLSCAASGFTF....SDYGMAWVRQAPGKGPEWVAFISNL.
.AYSIYYADTVT.GRFTISRENAKNTLYLEMSSLRSEDTAMYYCAR
>X67409|IGHV5-15*03|Mus musculus_BALB/c|[F]|V-REGION|1..261|261 nt|1| | | |87 AA|87+19=106|partial in 5'| |
............VKPGGSRKLSCAASGFTF....SDYGMAWVRQAPGKGPEWVAFISNL.
.AYSIYYADTVT.GRFTISRENAKNTLYLEMSSLRSEDTAMYYCAR
>AF120459|IGHV5-15*04|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVQPGGSLKLSCAASGFTF....SDYGMAWVRQAPRKGPEWVAFISNL.
.AYSIYYADTVT.GRFTISRENAKNTLYLEMSSLRSEDTAMYYCARR
>CAAA01073483|IGHV5-15*05|Mus musculus|F|V-REGION|17..312|296 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.ALVQPGGSLKLSCAASGFTF....SDYGMAWVRQAPRKGPEWVAFISNL.
.AYSIYYADTVT.GRFTISRENAKNTLYLEMSSLRSEDTAMYYCAR
>AC079273|IGHV5-16*01|Mus musculus_C57BL/6|F|V-REGION|25197..25492|296 nt|1| | ||98 AA|98+8=106| | |
EVKLVESEG.GLVQPGSSMKLSCTASGFTF....SDYYMAWVRQVPEKGLEWVANINYD.
.GSSTYYLDSLK.SRFIISRDNAKNILYLQMSSLKSEDTATYYCAR
>AF120471|IGHV5-16*02|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESEG.GLVQPGSSMKLSCTASGFTF....SDYYMAWVRQVPEKGLEWVANINYD.
.GSSTYYLDSLK.SRFIISRDNAKNILYLQMSSLKSEDTAMYYCARR
>AC079273|IGHV5-17*01|Mus musculus_C57BL/6|F|V-REGION|45820..46113|294 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVKPGGSLKLSCAASGFTF....SDYGMHWVRQAPEKGLEWVAYISSG.
.SSTIYYADTVK.GRFTISRDNAKNTLFLQMTSLRSEDTAMYYCAR
>AJ851868|IGHV5-17*02|Mus musculus_129/Sv|F|V-REGION|430902..431195|294 nt|1| | | |98 AA|98+8=106| | |
DVQLVESGG.GLVQPGGSRKLSCAASGFTF....SSFGMHWVRQAPEKGLEWVAYISSG.
.SSTIYYADTVK.GRFTISRDNPKNTLFLQMTSLRSEDTAMYYCAR
>AF120470|IGHV5-17*03|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SDYGMHWVRQAPEKGLEWVAYISSG.
.SSTIYYADTVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCARR
>AC090887|IGHV5-2*01|Mus musculus_C57BL/6|F|V-REGION|36458..36753|296 nt|1| | ||98 AA|98+8=106| | |
EVQLVESGG.GLVQPGESLKLSCESNEYEF....PSHDMSWVRKTPEKRLELVAAINSD.
.GGSTYYPDTME.RRFIISRDNTKKTLYLQMSSLRSEDTALYYCAR
>X01113|IGHV5-2*02|Mus musculus_BALB/c|F|V-REGION|395..690|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVQPRESLKLSCESNEYEF....PSHDMSWVRKTPEKRLELVAAINSD.
.GGSTYYPDTME.RRFIISRDNTKKTLYLQMSSLRSEDTALYYCAR
>AF120460|IGHV5-2*03|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVQPGESLKLSCESNEYEF....PSHDMSWVRKTPEKRLELVAAINSD.
.GGSTYYPDTME.RRFIISRDNTKKTLYLQMSSLRSEDTAMYYCARR
>AC073590|IGHV5-21*01|Mus musculus_C57BL/6|P|V-REGION|85968..86263|296 nt|1| | ||98 AA|98+8=106| | |
EVRLVESGG.SLIEHEGYIQLFCQASGFTL....SGYWMHWICQAPGKGPEWLANIKYD.
.GSEKYYAVSMK.G*FAISRDLPKNFLYLQMSNLRNEDTAMYYCAR
>AC090887|IGHV5-4*01|Mus musculus_C57BL/6|F|V-REGION|55400..55695|296 nt|1| | ||98 AA|98+8=106| | |
EVQLVESGG.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVATISDG.
.GSYTYYPDNVK.GRFTISRDNAKNNLYLQMSHLKSEDTAMYYCAR
>AJ851868|IGHV5-4*02|Mus musculus_129/Sv|F|V-REGION|1024532..1024827|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVKPGGSLKLSCAASGFTF....SDYYMYWVRQTPEKRLEWVATISDG.
.GSYTYYPDSVK.GRFTISRDNAKNNLYLQMSSLKSEDTAMYYCAR
>AF120461|IGHV5-4*03|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVATISDG.
.GSYTYYPDNVK.GRFTISRDNAKNNLYLQMSHLKSEDTAMYYCARR
>AC090887|IGHV5-6*01|Mus musculus_C57BL/6|F|V-REGION|83460..83755|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.DLVKPGGSLKLSCAASGFTF....SSYGMSWVRQTPDKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>AF120474|IGHV5-6*02|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.DLVKPGGSLKLSCAASGFTF....SSYGMSWVRQTPDKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCARR
>AF120466|IGHV5-6*03|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYGMSWVRQTPDKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCARR
>AJ851868|IGHV5-6-1*01|Mus musculus_129/Sv|F|V-REGION|941789..942084|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.DLVKPGGSLKLSCAASGFTF....SSYGMSWVRQTPDKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>AJ851868|IGHV5-6-2*01|Mus musculus_129/Sv|F|V-REGION|756123..756418|296 nt|1| | | |98 AA|98+8=106| | |
DVKLVESGG.GLVKLGGSLKLSCAASGFTF....SSYYMSWVRQTPEKRLELVAAINSN.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSSLKSEDTALYYCAR
>X67408|IGHV5-6-2*02|Mus musculus_BALB/c|F|V-REGION|1..261|261 nt|1| | | |87 AA|87+19=106|partial in 5'| |
............VKPGGSLKLSCAASGFTF....SSYYMSWVRQTPEKRLELVAAINSN.
.GGSTYYPDTVK.GRFTISRDNAKNTLYLQMSSLKSEDTALYYCAR
>AJ851868|IGHV5-6-3*01|Mus musculus_129/Sv|F|V-REGION|679323..679618|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVQPGGSLKLSCAASGFTF....SSYGMSWVRQTPDKRLELVATINSN.
.GGSTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>AF290966|IGHV5-6-3*02|Mus musculus_129/Sv|F|V-REGION|1..230|230 nt|1| | | |76 AA|76+30=106|partial in 5'| |
.......................AASGFTF....SSYGMSWVRQTPDKRLELVATINSN.
.GGSTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCAR
>AJ851868|IGHV5-6-4*01|Mus musculus_129/Sv|F|V-REGION|644016..644311|296 nt|1| | | |98 AA|98+8=106| | |
DVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCTR
>U04227|IGHV5-6-4*02|Mus musculus_BALB/c|F|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYTMSWVRQSPEKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLKSEDTAMYYCTR
>AJ851868|IGHV5-6-5*01|Mus musculus_129/Sv|F|V-REGION|595231..595523|293 nt|1| | ||97 AA|97+9=106| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVASISSG.
..GSTYYPDSVK.GRFTISRDNARNILYLQMSSLRSEDTAMYYCAR
>AJ851868|IGHV5-6-6*01|Mus musculus_129/Sv|F|V-REGION|514664..514957|294 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVQPGGSLKLSCAASGFTF....SSYAMSWIRQTPDKRLEWVASISSG.
.GSYTYYPDSVK.GRFTIPRDNTKNTLYLQMSSLSSKDTALYYCAR
>AC090887|IGHV5-9*01|Mus musculus_C57BL/6|F|V-REGION|119723..120018|296 nt|1| | ||98 AA|98+8=106| | |
EVMLVESGG.GLVKPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVATISGG.
.GGNTYYPDSVK.GRFTISRDNAKNTLYLQMSSLRSEDTALYYCAR
>AJ851868|IGHV5-9*02|Mus musculus_129/Sv|F|V-REGION|960245..960540|296 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFAF....SSYDMSWVRQTPEKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNARNTLYLQMSSLRSEDTALYYCAR
>X00163|IGHV5-9*03|Mus musculus_BALB/c|F|V-REGION|248..543|296 nt|1| | | |98 AA|98+8=106| | |
EVMLVESGG.GLVKPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVATISSG.
.GGNTYYPDSVK.GRFTISRDNAKNNLYLQMSSLRSEDTALYYCAR
>AF120472|IGHV5-9*04|Mus musculus_C57BL/6|[F]|V-REGION|1..295|295 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVATISGG.
.GGNTYYPDSVK.GRFTISRDNAKNTLYLQMSSLRSEDTAMYYCAR
>AJ851868|IGHV5-9-1*01|Mus musculus_129/Sv|F|V-REGION|862415..862708|294 nt|1| | | |98 AA|98+8=106| | |
EVMLVESGG.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLRSEDTAMYYCAR
>BN000872|IGHV5-9-1*02|Mus musculus_C57BL/6|F|V-REGION|2334745..2335041|294 nt|1| | | |98 AA|98+8=106| | |
DVKLVESGE.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVAYISSG.
.GDYIYYADTVK.GRFTISRDNARNTLYLQMSSLKSEDTAMYYCTR
>AJ851868|IGHV5-9-2*01|Mus musculus_129/Sv|F|V-REGION|845537..845832|296 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVKPGGSLKLSCAASGFTF....SSYGMSWVRQTPEKRLEWVATISGG.
.GSYTYYPDSVK.GRFTISRDNAKNNLYLQMSSLRSEDTALYYCAR
>AJ851868|IGHV5-9-3*01|Mus musculus_129/Sv|F|V-REGION|805213..805508|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVATISSG.
.GSYTYYPDSVK.GRFTISRDNAKNTLYLQMSSLRSEDTAMYYCAR
>AJ851868|IGHV5-9-4*01|Mus musculus_129/Sv|F|V-REGION|778352..778647|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQSPEKRLEWVAEISSG.
.GSYTYYPDTVT.GRFTISRDNAKNTLYLEMSSLRSEDTAMYYCAR
>AJ851868|IGHV5-9-5*01|Mus musculus_129/Sv|F|V-REGION|739246..739541|296 nt|1| | | |98 AA|98+8=106| | |
EVMLVESGG.GLVKPGGSLKLSCAASGFTF....SSYTMSWVRQTPEKRLEWVATISSG.
.GGNTYYPDSVK.GRFTISRDNAKNNLYLQMSSLRSEDTALYYCAR
>U04228|IGHV5S12*01|Mus musculus_BALB/c|[F]|V-REGION|1..264|264 nt|1| | | |88 AA|88+18=106|partial in 5'| |
..........GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVASISSG.
..GSTYYPDSVK.GRFTISRDNARNILYLQMSSLRSEDTAMYYCAR
>AF120463|IGHV5S21*01|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGE.GLVKPGGSLKLSCAASGFTF....SSYAMSWVRQTPEKRLEWVAYISSG.
.GDYIYYADTVK.GRFTISRDNARNTLYLQMSSLKSEDTAMYYCARR
>AF120469|IGHV5S24*01|Mus musculus_C57BL/6|[F]|V-REGION|1..297|297 nt|1| | | |99 AA|99+8=107| | |
EVKLVESGG.GLVKPGQSLKLSCAASGFTF....SNYYMSWVHQTPEKRLEWVAYISSS.
.GVSTYYPDNVK.GRFAISRDNAKNTLYLQMTSLKSEDTAMYYCARR
>X03399|IGHV5S4*01|Mus musculus_BALB/c|F|V-REGION|1..294|294 nt|1| | | |98 AA|98+8=106| | |
EVKLVESGG.GLVKPGGSLKLSCATSGFTF....SSYGMSWVRQTPEKRLEWVATISGG.
.GSYTYYPDSVK.GRFTISRDNAKNNLYLQMSSLRSEDTALYYCAR
>AC073590|IGHV6-3*01|Mus musculus_C57BL/6|F|V-REGION|157651..157949|299 nt|1| | | |99 AA|99+6=105| | |
EVKLEESGG.GLVQPGGSMKLSCVASGFTF....SNYWMNWVRQSPEKGLEWVAQIRLKS
DNYATHYAESVK.GRFTISRDDSKSSVYLQMNNLRAEDTGIYYCT
>AJ972404|IGHV6-3*02|Mus musculus_129/Sv|F|V-REGION|90404..90702|299 nt|1| | | |99 AA|99+6=105| | |
EVKLEESGG.GLVQPGGSMKLSCVASGFTF....SNYWMSWVRQSPEKGLEWVAQIRLKS
DNYATHYAESVK.GRFTISRDDSKSSVYLQMNNLRAEDTGIYYCT
>X03398|IGHV6-3*03|Mus musculus_BALB/c|F|V-REGION|1..300|300 nt|1| | | |100 AA|100+6=106| | |
EVKLEESGG.GLVQPGGSMKLSCVASGFTF....SNYWMSWVRQSPEKGLEWVAQIRLKS
DNYATHYAESVK.GRFTISRDDSKSSVYLQMNNLRAEDTGIYYCTG
>AC073561|IGHV6-4*01|Mus musculus_C57BL/6|ORF|V-REGION|1987..2286|300 nt|1| | | |100 AA|100+6=106| | |
DVNLEVSGG.GLVKPGGSMQLFCVASGFTF....VDGWMDWVRQSPEKGLEWVAEIANKA
NNYATYYPESVK.GRFTISRDDFKSSVYLHMNSLRAEDTGIYYCTR
>AJ972404|IGHV6-4*02|Mus musculus_129/Sv|ORF|V-REGION|106233..106532|300 nt|1| | ||100 AA|100+6=106| | |
DVNLEVSGG.GLVKPGGSMQRSCVDSGFTF....VDGWMDWVCQSPEKGLEWVAEIANKA
NNYATYYPESVK.GRFTISRDDFKSRVYLQKNSLRAEDTGIYYCTR
>AC073561|IGHV6-5*01|Mus musculus_C57BL/6|ORF|V-REGION|12109..12408|300 nt|1| | | |100 AA|100+6=106| | |
EVKIEESGG.GLVQPGGSMKLSCAASGFTF....SDYRMDWVHHSTENGLEWVAEIRNKA
SNYATYYVESVN.GRFTISRDDSKSSVYLQMNSLRAEDTGIYYCTR
>AJ972404|IGHV6-5*02|Mus musculus_129/Sv|ORF|V-REGION|115902..116201|300 nt|1| | ||100 AA|100+6=106| | |
EVKIEESGG.GLVQPGGSMKLSCAASGFTF....SDYRMDWVHHSTENGLEWVAEIRNKA
SNYATYYAESVK.GRFTISRDDSKSSVYLQMNSLRAEDTGIYYCKR
>AC073561|IGHV6-6*01|Mus musculus_C57BL/6|F|V-REGION|30302..30601|300 nt|1| | | |100 AA|100+6=106| | |
EVKLEESGG.GLVQPGGSMKLSCAASGFTF....SDAWMDWVRQSPEKGLEWVAEIRNKA
NNHATYYAESVK.GRFTISRDDSKSSVYLQMNSLRAEDTGIYYCTR
>AJ972404|IGHV6-6*02|Mus musculus_129/Sv|F|V-REGION|133286..133585|300 nt|1| | | |100 AA|100+6=106| | |
EVKLEESGG.GLVQPGGSMKLSCVASGFTF....SNYWMNWVRQSPEKGLEWVAEIRLKS
NNYATHYAESVK.GRFTISRDDSKSSVYLQMNNLRAEDTGIYYCTR
>AC073561|IGHV6-7*01|Mus musculus_C57BL/6|F|V-REGION|51138..51437|300 nt|1| | | |100 AA|100+6=106| | |
EEKLDESGG.GLVQPGRSMKLSCVASGFTF....TNSWMNWFCQSPEKGLEWVAQIKSKP
YNYETYYSDSVK.GRFTISRDDSKSSVYLQMNNLRAEDTGIYYCTW
>AJ972404|IGHV6-7*02|Mus musculus_129/Sv|F|V-REGION|149200..149499|300 nt|1| | | |100 AA|100+6=106| | |
EVKLDETGG.GLVQPGRPMKLSCVASGFTF....SDYWMNWVRQSPEKGLEWVAQIRNKP
YNYETYYSDSVK.GRFTISRDDSKSSVYLQMNNLRAEDMGIYYCTW
>K00692|IGHV6S2*01|Mus musculus_BALB/c|[F]|V-REGION|15..179|165 nt|1| | | |55 AA|55+5=60|partial in 3'| |
DVNLEVSGG.GLVGLGGSMQRSCVDSGFTF....VDGWMDWVCQSPEKGLEWVAEIANKL
>K00693|IGHV6S3*01|Mus musculus_BALB/c|[F]|V-REGION|14..178|165 nt|1| | | |55 AA|55+5=60|partial in 3'| |
DVNLEVSGG.GLVGLGGSMQHSCVDSGFTF....VDGWMDWVRQSPEKGLEWVAEIANKL
>K00694|IGHV6S4*01|Mus musculus_BALB/c|[F]|V-REGION|1..165|165 nt|1| | ||55 AA|55+10=65|partial in 5' and in 3' | |
.....ESGG.GLVQPGGSMKLSCVASGFTF....SNYWMNWVRQSPEKGLEWVAEIRLKS
GYATH
>AC079273|IGHV7-1*01|Mus musculus_C57BL/6|F|V-REGION|83079..83384|306 nt|1| | ||102 AA|102+6=108| | |
EVKLVESGG.GLVQSGRSLRLSCATSGFTF....SDFYMEWVRQAPGKGLEWIAASRNKA
NDYTTEYSASVK.GRFIVSRDTSQSILYLQMNALRAEDTAIYYCARDA
>AJ851868|IGHV7-1*02|Mus musculus_129/Sv|F|V-REGION|388422..388727|306 nt|1| | | |102 AA|102+6=108| | |
EVKLVESGG.GLVQPGGSLRLSCATSGFTF....SDFYMEWVRQPPGKRLEWIAASRNKA
NDYTTEYSASVK.GRFIVSRDTSQSILYLQMNALRAEDTAIYYCARDA
>X03253|IGHV7-1*03|Mus musculus_C57BL/10|F|V-REGION|261..566|306 nt|1| | | |102 AA|102+6=108| | |
EVKLVESGG.GLVQSGRSLRLSCATSGFTF....SDFYMEWVRQAPGKGLEWIAASRNKA
NDYTTEYSASVK.GRFIVSRDTSQSILYLQMNALRAEDTAIYYCARDA
>AC079273|IGHV7-2*01|Mus musculus_C57BL/6|ORF|V-REGION|98696..99001|306 nt|1| | | |102 AA|102+6=108| | |
EVKLVESEG.GLVQPGGSLRLSCATSGFTF....TDFYMNWVCQPPRKALEWLGFIRNKA
NGYTTDYSASMK.GRFTISRDNSQSILYLQMNTLRTEDSATYYCARDT
>AC073563|IGHV7-3*01|Mus musculus_C57BL/6|F|V-REGION|4402..4705|304 nt|1| | | |101 AA|101+6=107| | |
EVKLVESGG.GLVQPGGSLSLSCAASGFTF....TDYYMSWVRQPPGKALEWLGFIRNKA
NGYTTEYSASVK.GRFTISRDNSQSILYLQMNALRAEDSATYYCARY
>AJ851868|IGHV7-3*02|Mus musculus_129/Sv|F|V-REGION|73268..73571|304 nt|1| | | |101 AA|101+6=107| | |
EVKLVESGG.GLVQPGGSLRLSCATSGFTF....TDYYMSWVRQPPGKALEWLGFIRNKA
NGYTTEYSASVK.GRFTISRDNSQSILYLQMNTLRAEDSATYYCARD
>X03256|IGHV7-3*03|Mus musculus_C57BL/10|F|V-REGION|190..493|304 nt|1| | | |101 AA|101+6=107| | |
EVKLVESGG.GLVQPGGSLSLSCAASGFTF....TDYYMSWVRQLPGKALEWLGFIRNKA
NGYTTEYSASVK.GRFTISRDNSQSILYLQMNALRAEDSATYYCAKD
>M20457|IGHV7-3*04|Mus musculus_NZB|F|V-REGION|1..304|304 nt|1| | | |101 AA|101+6=107| | |
EVKLVESGG.GLVQPGGSLSLSCAASGFTF....TDYYMSWVRQPPGKALEWLALIRNKA
NGYTTEYSASVK.GRFTISRDNSQSILYLQMNALRAEDSATYYCARD
>AC073589|IGHV7-4*01|Mus musculus_C57BL/6|F|V-REGION|66090..66395|306 nt|1| | | |102 AA|102+6=108| | |
EVKLMESGG.GLVQPGASLRLSCAASGFTF....TDYYMSWVRQPPGKAPEWLALIRNKA
NGYTTEYTASVK.GRFTISRDNSQNILYLQMNTLRAEDSATYYCVKAV
>AJ972403|IGHV7-4*02|Mus musculus_129/Sv|F|V-REGION|47209..47514|306 nt|1| | | |102 AA|102+6=108| | |
EVKLMESGG.GLVQPGASLRLSCEASGFTF....TDYYMSWVRQPPGKSPEWLALIRNKA
NGYTTEYSASVK.GRFTISRDNSQNILYLQMNTLRAEASATYYCAKDV
>M16726|IGHV7-4*03|Mus musculus_BALB/c|F|V-REGION|459..764|306 nt|1| | ||102 AA|102+6=108| | |
EVKLMESGG.GLVQPGASLRLSCEASGFTF....TDYYMSWVRQPPGKSPEWLALIRNKA
NGYTTEYSASVK.GRFTISRDNSQNILYLQMNTLRAEASATYYCAKDV
>X03255|IGHV7-4*04|Mus musculus_C57BL/10|F|V-REGION|196..501|306 nt|1| | | |102 AA|102+6=108| | |
EVKLVESGG.GLVQPGGSLRLSCAASGFTF....TDYYMSWVRQPPGKAPEWLALIRNKA
NGYTTEYTASVK.GRFTISRDNSQNILYLQMNTLRAEDSATYYCVKAV
>AC073939|IGHV8-10*01|Mus musculus_C57BL/6|P|V-REGION|141615..141915|301 nt|1| | | |100 AA|100+7=107| | |
QVTL*QSGP.GKLQPSPTFSLTCSFSGFSLS..TYGMMVSWMCQPSGKGVEWLAHIWWN.
..DDKGYNPSLR.SRLTISKDTPNTQVFLKISSVVTADTTTYYCA*G
>AC073939|IGHV8-11*01|Mus musculus_C57BL/6|F|V-REGION|93806..94106|301 nt|1| | | |100 AA|100+7=107| | |
QITQKESGP.GILQPSQTLSLTCSFSGFSLS..TSGMGVGWIHQPSGNGLEWLAHIWWN.
..DNKYYNTALK.SRLTISKDTSNNQVFLKIASVDTADTATYYCARI
>AC073939|IGHV8-12*01|Mus musculus_C57BL/6|F|V-REGION|13010..13310|301 nt|1| | | |100 AA|100+7=107| | |
QVTLKESGP.GILQSSQTLSLTCSFSGFSLS..TSGMGVSWIRQPSGKGLEWLAHIYWD.
..DDKRYNPSLK.SRLTISKDTSRNQVFLKITSVDTADTATYYCARR
>AC163348|IGHV8-13*01|Mus musculus_C57BL/6|P|V-REGION|7455..7755|301 nt|1| | ||100 AA|100+7=107| | |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TFGMGVSWIRQPSGKGLEWLAHIYWD.
..DDKHYNPSLK.SRLTISKDTSNNQVFLKITTVDTADTATYYCARR
>AC160473|IGHV8-14*01|Mus musculus_C57BL/6|P|V-REGION|129831..130131|301 nt|1| | | |100 AA|100+7=107| | |
QVTLKEYGP.GKL*PSQTFSLTCTFSGFSLS..TYGMMVSCMCQPSGKGLVWLALIWWN.
..NDKGYNPFLR.SQLTISKDTSNNQVFLKITSVDPADTATYYCA*G
>AC073561|IGHV8-2*01|Mus musculus_C57BL/6|ORF|V-REGION|57802..58101|300 nt|1| | | |100 AA|100+8=108| | |
QVTLKVSGP.GILQPSQTLSLACTFSGISLS..TSGMGLSWLRKPSGKALEWLASIWN..
..NDNYYNPSLK.SRLTISKETSNYQVFLKLTSVDTADSATYYGAWRE
>AC079181|IGHV8-4*01|Mus musculus_C57BL/6|ORF|V-REGION|43860..44160|301 nt|1| | ||100 AA|100+7=107| | |
QITLKQSGP.GIVQPSQPVRLTCTFSGFSLS..TSGIGVTWIRQPSGKGLEWLATIWWD.
..DDNRYNPSLK.SRLAVSKDTSNNQAFLNIITVETADTAIYYCAQS
>AC074329|IGHV8-5*01|Mus musculus_C57BL/6|ORF|V-REGION|35069..35369|301 nt|1| | | |100 AA|100+7=107| | |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TSNMGIGWIRQPSGKGLEWLAHIWWN.
..DDKYYNPSLK.SRLTISKDTSNNQVFLKITSVDTADTATYYCAQI
>U23020|IGHV8-5*02|Mus musculus_C57BL/6|[F]|V-REGION|2..286|285 nt|1| | | |95 AA|95+7=102|partial in 3'| |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TSNMGIGWIRQPSGKGLEWLAHIWWN.
..DDKYYNPSLK.SRLTISKDTSNNQVFLKITTVDTADTATY
>AC074329|IGHV8-6*01|Mus musculus_C57BL/6|F|V-REGION|133286..133586|301 nt|1| | ||100 AA|100+7=107| | |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TFGMGVSWIRQPSGKDLEWLAHIYWD.
..DDKHYNPSLK.SQLRISKDTSNNQVFLKITTVDTVDTATYYCARR
>AC087166|IGHV8-7*01|Mus musculus_C57BL/6|P|V-REGION|123398..123691|294 nt|1| | | |98 AA|98+7=105| | |
QVILKESGP.GILQPSQTLSLTCSFSGFSFR..T*GMAVSWMRQPLGKGLEWLAQIGSD.
..DSKLYNPSLK.SRTTISKDTSNNHVFLKITSEDTEDSATYYCA
>AC087166|IGHV8-8*01|Mus musculus_C57BL/6|F|V-REGION|87468..87768|301 nt|1| | ||100 AA|100+7=107| | |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TFGMGVGWIRQPSGKGLEWLAHIWWD.
..DDKYYNPALK.SRLTISKDTSKNQVFLKIANVDTADTATYYCARI
>U23022|IGHV8-8*02|Mus musculus_C57BL/6|[F]|V-REGION|2..286|285 nt|1| | | |95 AA|95+7=102|partial in 3'| |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TFGMGIGWIRQPSGKGLEWLAHIWWD.
..DDKYYNPALK.SRLTISKDTSNNQVFLKITSVDTADTATY
>BN000872|IGHV8-8-1*01|Mus musculus_C57BL/6|ORF|V-REGION|666356..666661|301 nt|1| | | |100 AA|100+7=107| | |
QVILKESGP.GILQPSQTLSLTCSFSGFSLS..TYGMTVSWMRQPSRKGLEWLAQIGSD.
..DSKLYNPSLK.SRITISKETSNNHVFLKIICVDTEVSVTYYCAHR
>AC073939|IGHV8-9*01|Mus musculus_C57BL/6|ORF|V-REGION|192608..192908|301 nt|1| | | |100 AA|100+7=107| | |
QVTLKESGP.GILQPSQTLSLTCSFSVFSLS..TFGMGVSWIRQPSGKGLEWLAHIYWD.
..EDKHYKPSLK.SRLTISKDTSNNQVFLKITTVDTADTATYYSARR
>U23025|IGHV8-9*02|Mus musculus_C57BL/6|[P]|V-REGION|2..289|288 nt|1| | ||96 AA|96+7=103|partial in 5' and in 3' | |
QVTLKESGP.GILQPSQTLSLTCSFSVFSLN..TFGMGVSWIRQPSGKGLEWLAHIYWD.
..EDKHYKPSLK.SRLTISKDTSNNQVFLKITTVDTADTATY*
>U23023|IGHV8-9*03|Mus musculus_C57BL/6|[F]|V-REGION|2..286|285 nt|1| | | |95 AA|95+7=102|partial in 3'| |
QVTLKESGP.GILQSSQTLSLTCSFSVFSLS..TFGMGVSWIRQPSGKGLEWLAHIYWD.
..EDKHYKPSLK.SRLTISKDTSNNQVFLKITSVDTADTATY
>U14945|IGHV8S2*01|Mus musculus|P|V-REGION|144..444|301 nt|1| | | |100 AA|100+7=107| | |
QDTLKEYGP.GKL*PSQTFSLTCTFSGFSLS..TYGMMVSWMCQPSGKGLVWLALIWCN.
..NDKGYNPFLR.SQLTISKDTSNNQVFLKITSVDPADTATYYCA*G
>U23021|IGHV8S6*01|Mus musculus_C57BL/6|[P]|V-REGION|2..286|285 nt|1| | ||95 AA|95+7=102|partial in 3'| |
QVTLKESGP.GILQPSQTLSLTCSFSGFSLS..TFGMGVSWIRQPLGKGLE*LAHIYWD.
..DDKRYNPSLK.SRLTISKDTSRNQVFLKITTVDTADTATY
>U23024|IGHV8S9*01|Mus musculus_C57BL/6|[F]|V-REGION|2..286|285 nt|1| | | |95 AA|95+7=102|partial in 3'| |
QVTLKESGP.GILQSSQTLSLTCSFSGFSLN..TSGMGVSWIRQPSGKGLEWLAHIYWD.
..DDKRYNPSLK.SRLTISKDTSRNQVFLKITSVDTADTATY
>AC073563|IGHV9-1*01|Mus musculus_C57BL/6|F|V-REGION|63664..63957|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TEYPMHWVKQAPGKGFKWMGMIYTD.
.TGEPTYAEEFK.GRFAFSLETSASTAYLQINNLKNEDTATYFCVR
>AJ851868|IGHV9-1*02|Mus musculus_129/Sv|F|V-REGION|202425..202718|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTY.
.TGEPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDMATYFCAR
>Z15022|IGHV9-1*03|Mus musculus_BALB/c|F|V-REGION|117..396|280 nt|1| | | |93 AA|93+8=101|partial in 3'| |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTY.
.TGEPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDTAT
>Z15020|IGHV9-1*04|Mus musculus_BALB/c|F|V-REGION|116..395|280 nt|1| | | |93 AA|93+8=101|partial in 3'| |
QIQLAQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTY.
.TGEPTYADDFK.GRFAFSLETCASTAYLQINNLKNQDTAT
>AC073563|IGHV9-2*01|Mus musculus_C57BL/6|F|V-REGION|48591..48884|294 nt|1| | | |98 AA|98+8=106| | |
QIQFVQSGP.ELKKPGETVKISCKASVYTF....TEYPMHWVKQAPGKGFKWMGWINTY.
.SGEPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDTATYFCAR
>AJ851868|IGHV9-2*02|Mus musculus_129/Sv|F|V-REGION|185169..185462|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYAMHWVKQAPGKGLKWMGWKYTN.
.TGEPTYGDDFK.GRFAFSLETSASTAYLQINNLKNEDMATYFCAR
>AJ851868|IGHV9-2-1*01|Mus musculus_129/Sv|F|V-REGION|92637..92930|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TDYSMHWVKQAPGKGLKWMGWINTE.
.TGEPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDTATYFCAR
>AC073563|IGHV9-3*01|Mus musculus_C57BL/6|F|V-REGION|16900..17193|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TTYGMSWVKQAPGKGLKWMGWINTY.
.SGVPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDTATYFCAR
>AJ851868|IGHV9-3*02|Mus musculus_129/Sv|F|V-REGION|146596..146889|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTN.
.TGEPTYAEEFK.GRFAFSLETSASTAYLQINNLKNEDTATYFCAR
>L14366|IGHV9-3*03|Mus musculus_BALB.K|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTN.
.TGEPTYAEEFK.GRFAFSLETSASTAYLQINNLKNEDTATYFC
>AJ851868|IGHV9-3-1*01|Mus musculus_129/Sv|F|V-REGION|122148..122441|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTY.
.TGEPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDTATYFCAR
>AC073590|IGHV9-4*01|Mus musculus_C57BL/6|F|V-REGION|65900..66193|294 nt|1| | ||98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TTAGMQWVQKMPGKGFKWIGWINTH.
.SGEPKYAEDFK.GRFAFSLETSASTAYLQISNLKNEDTATYFCAR
>AJ972403|IGHV9-4*02|Mus musculus_129/Sv|F|V-REGION|105877..106170|294 nt|1| | | |98 AA|98+8=106| | |
QIQLVQSGP.ELKKPGETVRISCKASGYTF....TTAGMQWVQKMPGKGLKWIGWINTH.
.SGVPKYAEDFK.GRFAFSLETSASTAYLQISNLKNEDTATYFCAR
>L14364|IGHV9-4*03|Mus musculus_BALB.K|F|V-REGION|1..280|280 nt|1| | | |93 AA|93+11=104|partial in 5'| |
...LVQSGP.ELKKPGETVRISCKASGYTF....TTAGMQWVQKMPGKGLKWIGWINTH.
.SGVPKYAEDFK.GRFAFSLETSASTAYLQISNLKNEDTATYFC
>L14367|IGHV9S7*01|Mus musculus_BALB.K|F|V-REGION|1..284|284 nt|1| | ||94 AA|94+10=104|partial in 5' and in 3' | |
..QLVQSGP.ELKKPGETVKISCKASGYTF....TTAGMQWVQKMPGKGLKWIGWINTH.
.SGVPKYAEDFK.GRFAFSLETSASTAYLQINNLKNEDMATYFC
>L14368|IGHV9S8*01|Mus musculus_BALB.K|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QIQLVQSGP.ELKKPGETVKISCKASGYTF....TNYGMNWVKQAPGKGLKWMGWINTE.
.TGEPTYADDFK.GRFAFSLETSASTAYLQINNLKNEDTATYFC

</pre>
<hr />





<a href="/genedb/doc">-> IMGT/GENE-DB Documentation</a><br />
<a href="/genedb/">-> IMGT/GENE-DB Query page</a><br />
<a href="/genedb/directlinks">-> IMGT/GENE-DB direct links</a><br />
<a href="/genedb/stats">-> IMGT/GENE-DB statistics</a><br />
<a href="/IMGTgenedbdoc/programversions.html">-> IMGT/GENE-DB program versions</a><br />
<a href="/IMGTgenedbdoc/dataupdates.html">-> IMGT/GENE-DB data updates</a>

<hr />

<p class="menu">
<a href="/">IMGT Home page</a>
| <a href="/IMGTrepertoire/">IMGT Repertoire (IG and TR)</a>
| <a href="/IMGTrepertoireMH/">IMGT Repertoire (MH)</a>
| <a href="/IMGTrepertoireRPI/">IMGT Repertoire (RPI)</a>
| <a href="/IMGTindex/">IMGT Index</a>
| <a href="/IMGTScientificChart/">IMGT Scientific chart</a>
| <a href="/IMGTeducation/">IMGT Education</a>
</p>



<span class="logosFoot">
<a href="https://www.cnrs.fr"><img src="/images/cnrs.png" alt="logo CNRS" style="vertical-align:middle;" /></a>
<a href="https://www.umontpellier.fr/"><img src="/images/logos/universite_mtp.png" alt="Universit&#233; de Montpellier" style="width:60px;vertical-align:middle;" /></a>
<a href="https://ec.europa.eu/info/research-and-innovation_en"><img src="/images/UE.png" alt="logo UE" style="vertical-align:middle;" /></a>
</span>



<p>&#169;  Copyright 1995-2022 IMGT&reg;, the international ImMunoGeneTics information system&reg;
			| <a href="/about/termsofuse.php">Terms of use</a> | <a href="/about/aboutus.php">About us</a> | <a href="mailto:contact-imgt@igh.cnrs.fr">Contact us</a> | <a href="/about/CitingIMGT.php">Citing IMGT</a></p>

<!-- #echo var=$LAST_MODIFIED -->

</body>
</html>

</body>
</html>