
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<link rel="stylesheet" type="text/css" href="css/GENEDBQuery.css"/>
<link rel="stylesheet" type="text/css" href="css/IndividualResult.css"/>
<link type="text/css" href="css/GENEDBQuery_old.css" rel="stylesheet" />
<link type="text/css" href="css/styleFoot.css" rel="stylesheet" />
<title>IMGT/GENE-DB Consultation</TITLE>
</head>
<body id="GENE-DB">
<table style="width:100%;">
<tr>
<td align="left" nowrap valign="middle" width="70%">
<font size="+3" color="#000099"><b>THANK YOU</b><br/>
<b>for using <a href="/genedb/doc">IMGT/GENE-DB</a></b></font>
</td>
<td class="notshow" align=right valign=top width="30%"><a href="/"><img SRC="images/logo_IMGT2.png" ALT="WELCOME !" class="mainlogo"/></a>
</td>
</tr>
<tr>
<td class="logolegendleft">IMGT&reg;, the international ImMunoGeneTics information system&reg;</td>
<td class="logolegend">http://www.imgt.org</td>
</tr>
</table>
<hr/>
<img src="images/logoGene-DB.png" ALT="Gene-DB logo" class="mainlogo" />

<table class="citation">
<tr>
<td class="citation">
<b>Citing IMGT/GENE-DB</b>: Giudicelli, V. et al. Nucleic Acids Res., 33: D256 - D261 (2005).
PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/15608191/">15608191</a>
<a href="/PDF/NAR/33_D256-D261_2005.pdf"><img src="/images/pdficon.png" alt="pdf" /></a>
</td>
</tr>
</table>
<p class="programversion">IMGT/GENE-DB program version: <a href="/IMGTgenedbdoc/programversions.html">3.1.30</a> (18 September 2020)<br/>
<a href="/IMGTgenedbdoc/dataupdates.html">IMGT/GENE-DB data updates </a>
</p>

<h2 class="subtitle"><a href="/IMGTScientificChart/SequenceDescription/IMGTreferencesequences.html" >IMGT/GENE-DB reference sequences</a> in FASTA format:</h2>
<h2 class="subtitle">Amino acids  sequences with gaps according to the <a href="/IMGTScientificChart/Numbering/IMGTnumberingCR.html">IMGT unique numbering</a><br /> for <span class="casimir">F+ORF+in-frame P with IMGT Gaps</span> alleles including orphons</h2>
<hr>
<pre>
The FASTA header contains 15 fields separated by '|':

1. IMGT/LIGM-DB accession number(s)
2. IMGT gene and allele name
3. species
4. IMGT allele functionality
5. exon(s), region name(s), or extracted label(s)
6. start and end positions in the IMGT/LIGM-DB accession number(s)
7. number of nucleotides in the IMGT/LIGM-DB accession number(s)
8. codon start, or 'NR' (not relevant) for non coding labels
9. +n: number of nucleotides (nt) added in 5' compared to the corresponding label extracted from IMGT/LIGM-DB
10. +n or -n: number of nucleotides (nt) added or removed in 3' compared to the corresponding label extracted from IMGT/LIGM-DB
11. +n, -n, and/or nS: number of added, deleted, and/or substituted nucleotides to correct sequencing errors, or 'not corrected' if non corrected sequencing errors
12. number of amino acids (AA): this field indicates that the sequence is in amino acids
13. number of characters in the sequence: nt (or AA)+IMGT gaps=total
14. partial (if it is)
15. reverse complementary (if it is)
</pre>
<hr/>
<b>Number of results = 84</b><br />
<pre>
>AM773729|IGHV1-1*01|Vicugna pacos|F|V-REGION|32643..32938|296 nt|1| | | |98 AA|98+8=106| |rev-compl|
QVQLVQPGA.ELRKPGALLKVSCKASGYTF....TSYYIDWVRQAPGQGLGWVGRIDPE.
.DGGTNYAQKFQ.GRVTLTADTSTSTAYVELSSLRSEDTAVCYCVR
>AM939697|IGHV1S2*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVQPGA.ELRKPGASVKVSCKASGYTF....TSYYIDWVRQAPGQGLEWMGRIDPE.
.DGGTKYAQKFQ.GRVTFTADTSTSTAYVELSSLRSEGTPVYYC
>AM939698|IGHV1S3*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVQPGA.ELRKPGASLKVSCKASGYTF....TSYYIDWVRQAPGQGLEWMGRIDPE.
.DGGTKYAQKFQ.GRVTFTADTSTRTAYVELSSLRSEGTAVYYC
>AM939699|IGHV1S4*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVQPGA.ELRKPGASLKVSCKASGYTF....TSYYIDWVRQAPGQGLEWMGRIDPE.
.DGGTKYAQKFQ.GRVTFTADTSTSTAYVELSSLRSEGTAVYYC
>AM939700|IGHV1S5*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVQPGA.ELRKPGASVKVSCKASGYTF....TSYYIDWV*QAPGQGLEWMGRIDPE.
.DGGTNYAQKFQ.GRVTFTADTSTSTAYVELSSLRSEGTAVYYC
>AM939701|IGHV1S6*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVQPGA.ELRNPGASVKVSCKASGYTF....TSYYIDWVRQAPGQGLEWMGRIDPE.
.DGGTKYAQKFQ.GRVTFTADTSTSTAYVELSSLRSEGTAVYYC
>AM773729|IGHV3-1*01|Vicugna pacos|F|V-REGION|12791..13086|296 nt|1| | | |98 AA|98+8=106| | |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....DDYAMSWVRQAPGKGLEWVSAISWN.
.GGSTYYAESMK.GRFTISRDNAKNTLYLQMNSLKSEDTAVYYCAK
>AM773729|IGHV3-2*01|Vicugna pacos|F|V-REGION|27864..28159|296 nt|1| | | |98 AA|98+8=106| |rev-compl|
QVQLVESGG.GLVQPGGSLRLSCAASGLTF....SSYYMSWVRQAPGKGLESVSSIYSY.
.SSNTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEDTAVYYCAT
>AM773729|IGHV3-3*01|Vicugna pacos|F|V-REGION|685..980|296 nt|1| | | |98 AA|98+8=106| | |
QVQLVESGG.GLVQAGGSLRLSCAASGRTF....SSYAMGWFRQAPGKEREFVAAISWS.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEDTAVYYCAA
>AM773548|IGHV3S1*01|Vicugna pacos|F|V-REGION|6852..7147|296 nt|1| | | |98 AA|98+8=106| | |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYWMYWVRQAPGKGLEWVSAINTG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKSEDTAVYYCAK
>AM939768|IGHV3S10*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYDMSWVRQAPGKGLEWVSGIYSD.
..GSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939707|IGHV3S11*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLESVSSIYSY.
.SSNTYYADSVK.GRFTISRDNAKNTLYLQMNSLKSEDMAVYYC
>AM939708|IGHV3S12*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLESVSSIYSY.
.SSNTYYADSVK.GRFTISRDNAKNTLYLQMNSLKSEGTAVYYC
>AM939709|IGHV3S13*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGLTF....SSYYMSWVRQAPGKGLESVSTIYSY.
.GGNTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939710|IGHV3S14*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLRLSCAASGLTF....SSYAMSWVRQAPGKGLESVSTINSD.
.GSNTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPDGTAVYYC
>AM939711|IGHV3S15*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....DDYAMSWVRQAPGKGLEWVSAISWN.
.GGSTYYAESMK.GRFTISRDNAKNTLYLQMNSLKSEGTAVYYC
>AM939712|IGHV3S16*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQVVESGG.GLVQPGGSLRLSCAASGFTF....SDYAMSWVRQAPGKGLEWVSAISWN.
.GGSTYYAESMK.GRFTISRDNAKNTLYLQMNSLKSEGTAVYYC
>AM939713|IGHV3S17*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | ||96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTF....DDYGMSWVRHSPGKGLEWVSAISWN.
.GGSTYYAESMK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939714|IGHV3S18*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGLTF....GSYWMYWVRQAPGKGLEWVSSINTG.
.GDSSYYADSMK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939715|IGHV3S19*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRRSCATSGFTF....GSYWMYWVRQAPGKGLEWVSSINTG.
.GDSSYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEDTAVY*C
>AM939716|IGHV3S2*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSSAMSWVRQAPGKGLEWVSSIYSY.
.SSNTYYADSVK.GRFTISTDNAKNTLYLQMNSLKSEDTAVYYC
>AM939718|IGHV3S20*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGVSLRLSCAASGFTF....SSHYMSWVHQAPGKGLEWVSSINTG.
.GDSSFYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939719|IGHV3S21*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGVSLRLSCAASGFTF....SSYYMSWVRHSPGKGLE*VSAINSG.
.GGSTSYADSMK.GQFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939720|IGHV3S22*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGVSLRLSCAASGFTF....SSYYMSWVRHSPGKGLE*VSAINSG.
.GGSTSYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939721|IGHV3S23*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYYMSWVRHSPGKGLE*VSAINSG.
.GGSTSYADSMK.GQFTISRDNAKNTLYLQMNSLKSEGTAVYYC
>AM939722|IGHV3S24*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYYMSWVRHAPGKGLE*VSTINSG.
.GGSTSYADSMK.GQFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939723|IGHV3S25*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYWMNWVRQAPGKGLEWVSAINSG.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939724|IGHV3S26*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYWMNWVRQAPGKGLEWVSAINSG.
.GGSTSYADSMK.GQFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939725|IGHV3S27*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTF....GSYWMYWVRQAPGKGLEWVSAINSG.
.GGSTSYADSMK.GQFTISSDNARNTLYLQMNSLKPEGTAVYYC
>AM939726|IGHV3S28*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRVSCAASGFTF....SSYYMSWVRQAPGKGLEWVSAINTG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939727|IGHV3S29*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRVSCAASGFTF....SSYYMSWVRQAPGKGLEWVSTINTG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939728|IGHV3S3*01|Vicugna pacos|ORF|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVRLVESGG.GLVQPGGSLRLSCAASGFTF....SSSAMSRVRQVPGKGLEWVSSIYSY.
.SSNTYYADSVK.GRFTISTDNAKNTLYLQMNSLKSEGTAVYYC
>AM939729|IGHV3S30*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGPEWVSAINTG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939730|IGHV3S31*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGPEWVSAINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939731|IGHV3S32*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....GSYDMSWVRQAPGKGPEWVSAINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939732|IGHV3S33*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYDMSWVRQAPGKGPEWVSDINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939733|IGHV3S34*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTL....GSYDMSWVRQAPGKGPEWVSCINSG.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939734|IGHV3S35*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQVVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLEWVSYINSG.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEDTAVYYC
>AM939735|IGHV3S36*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....DNYAMSWVRQAPGKGLEWVSYINSG.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939736|IGHV3S37*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRRAPGKGLEWVSYINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939717|IGHV3S38*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....GSYDMSWVRRAPGKGLEWVSYINSG.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939737|IGHV3S39*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLEWVSDINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939738|IGHV3S4*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESVG.GLVQDGGSLRLSCAASGRTF....SSSAMSWVRQAPGKGLEWVSSIYSY.
.SSNTYNADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939739|IGHV3S40*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYDMSWVRQAPGKGLEWVSAINSG.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEDTAVYYC
>AM939740|IGHV3S41*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLEWVSAINSG.
.GGSTSYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939741|IGHV3S42*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLEWVSAINSG.
.GGSTSYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939742|IGHV3S43*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRHSPGKGLEWVSAINSG.
.GGSTSYADSMK.GQFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939743|IGHV3S44*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKGLEWVSAINGG.
.DGSTSYADSVK.GRSTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939744|IGHV3S45*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLKHSCAASGLTF....GSYDMSWVRQAPGKGPEWVSAINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939745|IGHV3S46*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....GSYVLSWVCHSPGKGLEWVSAINSC.
.GGSTSYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939746|IGHV3S47*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLPRAASGFTF....GSYDVSWVRQAPGKGPE*VSAINSG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939747|IGHV3S48*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVHPGGSLKLSCAASGFTF....RSYDMSWVPHAP*KGPEWVSYINSG.
.GGSTYYAHSVK.GRFTISRDNAKNTLYL*MNSLKPEGTAVFYC
>AM939748|IGHV3S5*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGLTF....SSYYMSWVRQAPGKGLEWVSGIYSD.
.GSDTYYADSVE.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM773548|IGHV3S53*01|Vicugna pacos|F|V-REGION|33845..34137|293 nt|1| | | |97 AA|97+9=106| | |
QVQLVESGG.GLVQPGGSLRLSCAASGSIF....SINAMGWYRQAPGKQRELVAAITSG.
..GSTNYADSVK.GRFTISRDNAKNTVYLQMNSLKPEDTAVYYCNA
>AM939756|IGHV3S54*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGSIF....SINAMGWYRQAPGKQRELVAAINTG.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKSEGTAVYYC
>AM939763|IGHV3S55*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLRHSCAASGLTF....GSYAMGWYRQAPGKERELVAAISSG.
..GSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939764|IGHV3S56*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLRHSCAASGLTF....GSYAMGWYRQAPGKERELVAAISSG.
..GSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVSYC
>AM939765|IGHV3S57*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLRLSCAASGRTF....SSYAMGWYRQAPGKERELVAAISSG.
..GSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939752|IGHV3S58*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLRLSCAASGRTF....SSYAMGWFRQAPGKEREFVAAISWI.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939753|IGHV3S59*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQVVESGG.GLVQAGGSLRLSCAASGFTF....SSYAMGWFRQAPGKEREFVAAISWS.
.GGSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939749|IGHV3S6*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYWMNWVRQAPGKGLEWVSGIYSD.
.GSDTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939754|IGHV3S60*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTL....DNYAMGWFRQAPGKEREFVSCISWS.
.GGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939757|IGHV3S61*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTL....DYYAIGWFRQAPGKEREGVSCISSS.
.DGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939758|IGHV3S62*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTL....DYYAIGWFRQAPGKEREGVSCISSS.
.DGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGPAVYYC
>AM939759|IGHV3S63*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESGG.GLVQAGGSLRLSCAASGFTL....DYYAIGWFRQAPGKEREGVSCISSS.
.DGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939760|IGHV3S64*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQVVESGG.GLVQPGGSLRLSCAASGFTL....DYYAISWFRQAPGKEREGVSCINSS.
.DGSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM939761|IGHV3S65*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTL....DYYAIGWFRQAPGKEREGVSCISSS.
.GGSTNYADSVK.GRFTISRDNAKNTVYLQMNSLKPEDTAVYYC
>AM939762|IGHV3S66*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLVESGG.GLVQAGGSLRLSCAASGFTF....DDYAIGWFRQAPGKEREGVSCISSS.
.DGSTYYADSVK.GRFTISSDNAKNTVYLQMNSLKPEGTAVYYC
>AM939766|IGHV3S67*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
EVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMSWVRQAPGKERELVAAISSG.
..GSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKSEGTAEYYC
>AM939755|IGHV3S68*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVESVG.GLVQDGGSLRLSCAASGRTF....SRSAMRWFRQAPGKEREWVSCISSS.
.DGSTNYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939750|IGHV3S7*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLVETGG.GLVQPGGSLRLSCAASGFTF....SSSAMSWVRQAPGKGLEWVSGIYSD.
.GSDTYYADSVK.GRFTISRDNAKNTLYLQMNSLKPEGTAVYYC
>AM939751|IGHV3S8*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
QLQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMGWARQVPGKGLEWVSGIYSD.
..GSTYYADSVK.GRFTISRDNAKNTLYLQMNSLKSEGTAVYYC
>AM939767|IGHV3S9*01|Vicugna pacos|F|V-REGION|1..285|285 nt|1| | | |95 AA|95+9=104|partial in 3'| |
QVQLVESGG.GLVQPGGSLRLSCAASGFTF....SSYAMGWARQVPGKGLEWVSGIYSD.
..GSTYYADSVK.GRFTISRDNAKNTVYLQMNSLKPEGTAVYYC
>AM773548|IGHV4S1*01|Vicugna pacos|F|V-REGION|23907..24205|299 nt|1| | | |99 AA|99+7=106| | |
QVQLQESGP.GLVKPSQTLSLTCTVSGGSIT..TSYYAWSWIRQPPGKGLEWMGVIAYD.
..GSTYYSPSLK.SRTSISRDTSKNQFSLQLSSVTPEDTAVYYCAR
>AM939702|IGHV4S10*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
EVQLQESGP.GLVKPSQMLSLTCTLSGDSIT..TSCYAWSWIRQPPGKGLEWMGAIYS..
..GSTYYSPSLK.SRTSISRDTSKNQFSLQLSSVTPEGTAVYYC
>AM939703|IGHV4S11*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLQESGP.DLVKPSQMLSLTCTVSGGSNT..TSYYAWSWIRQPPGKGLE*MGAIYS..
..GSTYYSPSLK.SRTSISRDTSKNQFSLQLSSVTPEGTAVYYC
>AM939769|IGHV4S2*01|Vicugna pacos|F|V-REGION|1..291|291 nt|1| | | |97 AA|97+7=104|partial in 3'| |
EVQLQESGP.GLVKPSQTLSLTCTVSGGSIT..TSYYAWSWIRQPPGKGLEWMGAIAYS.
..GSTYYSPSLK.SRTSISRDTSNNQFSLQLSSVTPEGTAVYYC
>AM939770|IGHV4S3*01|Vicugna pacos|F|V-REGION|1..291|291 nt|1| | | |97 AA|97+7=104|partial in 3'| |
EVQVQESGP.GLVKPSQALSLTCTASGGSNT..TSYYAWSWIRQPPGKGLEWMGAIAYS.
..GSTYYSPSLK.SRTSISRDTSNNQFSLQLSSVTPEGTAVYYC
>AM939771|IGHV4S4*01|Vicugna pacos|F|V-REGION|1..291|291 nt|1| | | |97 AA|97+7=104|partial in 3'| |
QVQRQESGP.GLVKPSQMLSLTCTASGGSNT..TSYYAWSWIRQPPGKGLEWMGAIAYD.
..GSTYYSPSLK.SHTSISRDTSKNQFSLQLSSVTPEGTAVYYC
>AM939772|IGHV4S5*01|Vicugna pacos|F|V-REGION|1..291|291 nt|1| | | |97 AA|97+7=104|partial in 3'| |
QVQLQESGP.GLVKPSQTLSLTCTVSGGSIT..TSYYYWSWIRQPPGKGLEWMGAIAYD.
..GSTYYSPSLK.SRTSISRDTSKNQFSLQLSSVTPEGTAVYYC
>AM939773|IGHV4S6*01|Vicugna pacos|F|V-REGION|1..291|291 nt|1| | | |97 AA|97+7=104|partial in 3'| |
EVQLQESGP.GLLKPSQTLSLTCAVYGGSIT..TSYYYWSWIRQPPGKGLEWMGAIAYD.
..GSTYYSPSLK.SHTSISRDTSKNQFSLQLSSVTPEGTAVYYC
>AM939704|IGHV4S7*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLQESGP.GLVKPSQTLSLTCAVYGGSIT..TSCYAWSCICQPPEKGLEWMAAIYS..
..GSTYYSPSLK.SHTSISRDTSKNQFSLQLSSVTPEGTAVYYC
>AM939705|IGHV4S8*01|Vicugna pacos|F|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQLQESGP.GLVKPSQTLSLTCTASGGSNT..TSYYAWSWIRQPPGKGLEWMGAIYS..
..GSTYYSPSLK.SRTSISRDTSNNQFSLQLSSVTPEGTCVYYC
>AM939706|IGHV4S9*01|Vicugna pacos|P|V-REGION|1..288|288 nt|1| | | |96 AA|96+8=104|partial in 3'| |
QVQL*ESGP.GLVKPSQTLSLTCTVSGGSIT..TSCYAWSWIHQPPGKGLE*MGAIYS..
..GSTYYSPSLK.SHTSISRDTSKNQFSLQLSSVTPEGTAVYYC

</pre>
<hr />





<a href="/genedb/doc">-> IMGT/GENE-DB Documentation</a><br />
<a href="/genedb/">-> IMGT/GENE-DB Query page</a><br />
<a href="/genedb/directlinks">-> IMGT/GENE-DB direct links</a><br />
<a href="/genedb/stats">-> IMGT/GENE-DB statistics</a><br />
<a href="/IMGTgenedbdoc/programversions.html">-> IMGT/GENE-DB program versions</a><br />
<a href="/IMGTgenedbdoc/dataupdates.html">-> IMGT/GENE-DB data updates</a>

<hr />

<p class="menu">
<a href="/">IMGT Home page</a>
| <a href="/IMGTrepertoire/">IMGT Repertoire (IG and TR)</a>
| <a href="/IMGTrepertoireMH/">IMGT Repertoire (MH)</a>
| <a href="/IMGTrepertoireRPI/">IMGT Repertoire (RPI)</a>
| <a href="/IMGTindex/">IMGT Index</a>
| <a href="/IMGTScientificChart/">IMGT Scientific chart</a>
| <a href="/IMGTeducation/">IMGT Education</a>
</p>



<span class="logosFoot">
<a href="https://www.cnrs.fr"><img src="/images/cnrs.png" alt="logo CNRS" style="vertical-align:middle;" /></a>
<a href="https://www.umontpellier.fr/"><img src="/images/logos/universite_mtp.png" alt="Universit&#233; de Montpellier" style="width:60px;vertical-align:middle;" /></a>
<a href="https://ec.europa.eu/info/research-and-innovation_en"><img src="/images/UE.png" alt="logo UE" style="vertical-align:middle;" /></a>
</span>



<p>&#169;  Copyright 1995-2020 IMGT&reg;, the international ImMunoGeneTics information system&reg;
			| <a href="/about/termsofuse.php">Terms of use</a> | <a href="/about/aboutus.php">About us</a> | <a href="mailto:contact-imgt@igh.cnrs.fr">Contact us</a> | <a href="/about/CitingIMGT.php">Citing IMGT</a></p>

<!-- #echo var=$LAST_MODIFIED -->

</body>
</html>

</body>
</html>